---
title: Analysis of Microtiter Plate Phenotyping of Temperature Response of the major plant pathogen *Zymoseptoria tritici*
author: "Silvia Minana-Posada"
date: "2023-11-06"
output: 
  rmdformats::robobook:
    self_contained: true
    number_sections: true
    code_folding: hide
    highlight: haddock
    bibliography: references.bib
    use_bookdown : true
bibliography: references.bib
link-citations: true
---

> This is the analysis from the phenotypic information obtained through the microtiter-plate five-temperature phenotyping experiments with the fungal wheat pathogen <ins>Zymoseptoria tritici</ins> conducted by me during 2022. The raw results are the measurements of the optical density at 405nm [@boixel2019] of the wells through 2 daily measurements on a 7 day period. The intention of these analysis is to extract different growth parameters that can be used to explore the diversity of the response of a world-wide collection of <ins>Z. tritici</ins> to temperature. For this purpose, different steps were conducted and will be presented here in that order: optical density to cell density transformation and filtering, growth variables calculations, ratios and sensitivities calculated, final filtering. After this, the data was explored in order to obtain information about thermal adaptation in this pathogen.

```{r setup, message=FALSE, error=FALSE, warning=FALSE, eval=TRUE}

##### Packages  
library(bookdown)
library(mgcv)
library(genio)
library(MetBrewer)
library(scales)
library(rstatix) 
library(grid)
library(ggpubr)
library(readxl)
library(RColorBrewer)
library(wesanderson)
library(tidyverse)
library(dplyr)
library(lubridate)
library(MetBrewer)
library(npreg)
library(tidygraph)
library(ggraph)
library(igraph)
library(corrplot)
library(tidygraph)
library(pheatmap)
library(raster)
#library(dismo)
#library(rgdal)
library(maps)
library(ggrepel)
library(viridis)
library(broom)
library(car)
library(factoextra)
library(kableExtra)
library(GGally)
library(dbscan)
library(scatterpie)
library(magrittr)
library(metR)
library(ggnewscale)
library(cowplot)


##### Knitr options

knitr::opts_chunk$set(message = F)
knitr::opts_chunk$set(warning = F)
knitr::opts_chunk$set(results = T)


palette_alice = c("#DA4167",
                  "#ffba0a",
                  "#A20106",
                  "#3F6E0C",
                  "#129eba",
                  "#8fa253")
palette_infinite = colorRampPalette(palette_alice)

K_colors = c("#FF6319", #Argentina & Uruguay
             "#FCCC0A", #Australia
             "#8E258D", #Canada
             "#C60C30", #Chile
             "#00A1DE", #Europe
             "#1877F2", #Middle East
             "#012169", #Africa
             "#6E3219" )#USA


clim_colors=c("#eeb94f","#b15e12","#8fb314","#586e0c","#00cba6","#00513b", "#04b256","#e7ade7")



palette_temperature = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E")

palette_infinitetemp = colorRampPalette(palette_temperature)

##### Directories

analysis_dir = "./"
plates_dir = paste0(analysis_dir, "03_Plate_maps/")


Condition_all_temperatures = c("12C", "15C", "18C", "22C", "25C")
Condition_all_temperatures_without_control = c("12C")



All_Conditions_list <-
  list(
    Condition_all_temperatures,
    Condition_all_temperatures_without_control
  )




```

```{r readdata, eval=FALSE}

##### Read data files
#These files were obtained from the python script made by Alice Feurtey (2021) called Convert_tecan_file.py that creates .txt files from the excel files that the Tecan machine produces. 

OD_input_old = paste0(analysis_dir, "02_OD_Data/OD_results_up_to_20220721.txt")
OD_input_updated = paste0(analysis_dir, "02_OD_Data/OD_results_up_to_20221004.txt")



##### Make tables readable 

OD_raw_old =  read_delim(OD_input_old,
                     col_names = c("Date", "Time", "An_name", "Row", "Col", "Absorbance"),
                     col_types = list( Date = col_character(), Time = col_time(), An_name = col_character(),
                                       Row = col_character(), Col = col_double(), Absorbance = col_character())) %>%
  mutate(Absorbance = as.numeric(str_replace(Absorbance, ",", "."))) %>% separate( col = An_name, sep = "_",
                                                                                   into = c( "Remove", "Plate", "Timepoint", "Temperature", "Condition", "Repet", "Repet2" ))%>%
  dplyr::select(-c(Remove, Repet2)) %>%
  mutate(Condition = ifelse(Condition == "temperature", Temperature, Condition)) 


OD_raw_updated =  read_delim(OD_input_updated,
                     col_names = c("Date", "Time", "An_name", "Row", "Col", "Absorbance"),
                     col_types = list( Date = col_character(), Time = col_time(), An_name = col_character(),
                                       Row = col_character(), Col = col_double(), Absorbance = col_character())) %>%
  mutate(Absorbance = as.numeric(str_replace(Absorbance, ",", "."))) %>% separate( col = An_name, sep = "_",
                                                                                   into = c( "Remove", "Plate", "Timepoint", "Temperature", "Condition", "Repet", "Repet2" ))%>%
  dplyr::select(-c(Remove, Repet2)) %>%
  mutate(Condition = ifelse(Condition == "temperature", Temperature, Condition)) 


##### Link to Plate maps that have the information of the samples that are in each well. 
#In the folder 03_Platemaps/ in the R Project I only used the Plate maps from Plate020 and not the plates before, I, additionally, removed the plates that were done without re scaling which are Plates from Plate035 to Plate043. That is why those plates are not there. 
plates_dir="./03_Plate_maps/"

list_maps = list.files(path = plates_dir,
                       pattern = "Plate",
                       full.names = TRUE)

plate_maps = data.frame()
for (name in list_maps) {
  temp = read_excel(name) %>%
    mutate(Plate = str_replace_all(name, "./03_Plate_maps/", "") %>% str_replace_all(".xlsx", ""))
  plate_maps = bind_rows(plate_maps, temp)
}

OD_raw_old=OD_raw_old %>%
  right_join(., plate_maps)

OD_raw_updated=OD_raw_updated %>%
  right_join(., plate_maps)


### Correspondence files to add the ID_genome name, which is how the strains are named in the vcf file

#Isolates with more than 20% missing data: 

missing20rem<-c("Texas_1994_ST94TX_PF4a_1", "Tunisia_2008_ST08TN_33_1_2", "ST90ORE_a12_3B_13", "ST92YE_1")  

Correspondencefile <-read_csv("01_Source_Tables/CorrespondencebtwnIDvcfsandexactnames_SMP_20230424_v1.csv") 

isolateswrongstructure<-c("Ukraine_1995_ST95UR_F1c_1", "Israel_1992_ISZE2a_2", "Texas_1994_ST94TXPG7a_1", "Kenya_1994_STKE94ML1a", "Kenya_1994_STKE94MF1a")

k8admixture<-read_tsv("./01_Source_Tables/newgenofile.high_anc_coef_snmf.tsv")
k8admixture$Year<-as.character(k8admixture$Year)

k8admixture<- k8admixture %>%
  mutate(Adm_Cluster = case_when(endsWith(Cluster, "V1") ~ "Argentina_Uruguay",
                            endsWith(Cluster, "V2") ~ "Australia",
                            endsWith(Cluster, "V3") ~ "Europe",
                            endsWith(Cluster, "V4") ~ "Tunisia",
                            endsWith(Cluster, "V5") ~ "USA",
                            endsWith(Cluster, "V6") ~ "Middle_East",
                            endsWith(Cluster, "V7") ~ "Chile",
                            endsWith(Cluster, "V8") ~ "Canada"))

climateclass<-read.csv("climateclass.csv")
newcoords<-read.csv("01_Source_Tables/WP1_SMP_Coordandvariety_contloc_20240826_v1.csv", header=TRUE) %>% rename(Strain=Isolate_full_name_phenotyping)

##### Remove is a typo in t12 Plate50 where instead of Plate, each cell has "Plate

OD_raw_old$Plate =gsub('"', "", OD_raw_old$Plate)
OD_raw_updated$Plate =gsub('"', "", OD_raw_updated$Plate)



##### The conditions in some Plates might be typed differently, so remove those differences.

OD_raw_old=OD_raw_old %>%
  mutate(Condition = ifelse(Condition == "KCl1M", "KCl", Condition)) %>%
  mutate(Condition = ifelse(Condition == "pH4.5", "pH", Condition))

OD_raw_updated=OD_raw_updated %>% 
  mutate(Condition = ifelse(Condition == "KCl1M", "KCl", Condition)) %>%
  mutate(Condition = ifelse(Condition == "pH4.5", "pH", Condition))


##### Remove space in Plate050 t12 25C Rep 3 as it was written "Plate050_t12_25C_temperature _3" 

OD_raw_old[OD_raw_old$Plate == "Plate050" & OD_raw_old$Timepoint == "t12" & OD_raw_old$Temperature == "25C" & OD_raw_old$Condition == 'temperature ' , "Repet"] <- "3"  
OD_raw_old[OD_raw_old$Plate == "Plate050" & OD_raw_old$Timepoint == "t12" & OD_raw_old$Temperature == "25C" & OD_raw_old$Condition == 'temperature ' , "Condition"] <- "25C"  

OD_raw_updated[OD_raw_updated$Plate == "Plate050" & OD_raw_updated$Timepoint == "t12" & OD_raw_updated$Temperature == "25C" & OD_raw_updated$Condition == 'temperature ' , "Repet"] <- "3"  
OD_raw_updated[OD_raw_updated$Plate == "Plate050" & OD_raw_updated$Timepoint == "t12" & OD_raw_updated$Temperature == "25C" & OD_raw_updated$Condition == 'temperature ' , "Condition"] <- "25C"  


##### Merge the 2 dataframes 

OD_raw<- full_join(OD_raw_old, OD_raw_updated)

##### Remove information that comes from FAILED files. First group and then filter the data with NAs then, confirm with examples visually looking at the original files. Remove FAILED plates and also missing data (I did not manage to always take all the replicate measurements because of time limitation in the evening). 

OD_raw<- subset(OD_raw, !is.na(Date)) #Just like this there are way too many NAs, because in OD_raw_old has no information for the last plates.  

OD_rawwithna<- subset(OD_raw, is.na(Absorbance)) %>% dplyr::select(Plate, Date, Time, Condition, Timepoint) %>% distinct()

OD_rawwithna

#Therefore, only 3 measurements have NAs which means these plates were the ones interrupted. So, I will remove them


OD_raw_nona1<-OD_raw %>% filter(Date == "2022-03-27") %>% filter( Timepoint == "t11") %>% filter( Condition == "pH") %>% subset(is.na(Absorbance)) ### Checked file manually and there was a run that did not even begin and another one that is the correct one all in the same minute, so I just have to remove NAs.

OD_raw01<-anti_join(OD_raw, OD_raw_nona1)

OD_raw_nona2<-OD_raw01 %>% filter(Date == "2022-04-07") %>% filter(Timepoint == "t6")%>% filter( Condition == "15C") %>% subset(is.na(Absorbance)) ### Checked file manually and there was a run that did not even begin and another one that is the correct, so I just have to remove NAs.
                                 
OD_raw02<-anti_join(OD_raw01, OD_raw_nona2)


OD_raw_nona3<-OD_raw02%>% filter(Date == "2022-07-10") %>% filter( Timepoint == "t12") %>% filter(Condition =="18C") %>% subset(is.na(Absorbance))

OD_raw03<-anti_join(OD_raw02, OD_raw_nona3)  ### Checked file manually and there was a run that did not even begin and another one that is the correct one all in the same minute, so I just have to remove NAs.

##### Check if all the wells with Absorbance are being assigned to a Strain:

#I had to do this and not it is removed, but before the changes in the Plate files I found in which Plate maps I had errors like this:
#OD_rawNAooo<- subset(OD_raw03, is.na(Strain)) %>% select(Plate) %>% distinct() ### In the end it was from Plate068 to Plate073. Now this problem is fixed. 


##### Check if there are repetitions of the repetitions.

OD_raw04<-OD_raw03%>% 
  group_by(Plate, Date, Condition, Timepoint, Repet, Cell) %>% 
  filter(n()>1) %>% summarize(n=n()) %>% dplyr::select(Plate, Date, Timepoint, Repet, Condition, n) %>% distinct()
OD_raw04 ### Empty table because there are no un-necessary Repeated measurements 
#The plate Plate065_t3_18C_SA is repeated because I made a mistake actually, the file p723_Silvia_9.8.21_ODline_20220713_115046.xlsx actually is GB of Plate062 t11, so I need to change that. 
#Because of plate maps this has to be done in the original file so this issue will be solved when this script runs. Plate071_t3_12C_temperature is repeated because I only changed the first repeat of the file and not the rest, so it is actually, Plate068_t11_12C_temperature in file p723_Silvia_9.8.21_ODline_20220720_121429.xlsx


#### Removing particular wells with noted contamination or with noted mistakes

####This includes the cases in which the control wells are reduced because of lack of material on that day's experiment due to lack of foresight on my side (did not prepare enough quantity for being able to fill all the wells the first day i did the experiment). 


##### Remove the wells of Plate020 (no repetitions) for which 4 samples have no H2O2 results (because of lack of material that day). They are located in Line 10 and 11. 


OD_raw05<- OD_raw03 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "H2O2" & Reduced_strainname == "ST08TN_33_9_6"))
OD_raw06<- OD_raw05 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "H2O2" & Reduced_strainname == "STDN94Gw8"))
OD_raw07<- OD_raw06 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "H2O2" & Reduced_strainname == "a12_4A_7"))
OD_raw08<- OD_raw07 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "H2O2" & Reduced_strainname == "BOO35_D4a1"))

##### Remove control wells in Plate020 and Plate023 that have missing wells for the controls for the chemical stresses

#### Plate020

### GB:

GBcellsPlate020<-c("E1", "F1", "G1", "H1")

OD_raw09<- OD_raw08 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "GB" & Cell %in% GBcellsPlate020)) 
OD_raw10<- OD_raw09 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "GB" & Col == 12)) 

### pH:

OD_raw11<- OD_raw10 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "pH" & Cell == "C1")) 
OD_raw12<- OD_raw11 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "pH" & Col == 12)) 

### H2O2:

H2O2cellsPlate020<-c("C1", "D1", "E1", "F1", "G1", "H1")

OD_raw13<- OD_raw12 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "H2O2" & Cell %in% H2O2cellsPlate020)) 
OD_raw14<- OD_raw13 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "H2O2" & Col == 12))

### SA:

SAcellsPlate020<-c("F1", "G1", "H1")

OD_raw15<- OD_raw14 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "SA" & Cell %in% SAcellsPlate020)) 
OD_raw16<- OD_raw15 %>% dplyr::filter(!(Plate == "Plate020" & Condition == "SA" & Col == 12))

#### Plate023

### H2O2: 

OD_raw17<- OD_raw16 %>% dplyr::filter(!(Plate == "Plate023" & Condition == "H2O2" & Col == 12))


###### Remove the noted contaminated wells. 

contPlate047<-c("C1", "D1")

OD_raw18<- OD_raw17 %>% dplyr::filter(!(Plate == "Plate047" & Condition == "25C" & Cell %in% contPlate047))

Plati<-c("Plate053", "Plate054", "Plate055")
OD_raw19<- OD_raw18 %>% dplyr::filter(!(Plate %in% Plati & Reduced_strainname == "a12_3B_14"))


##### Remove plates that had the weird media problem (Alexandra (HIWI))
#These plates can be used as positive control of contamination 

contPlates<-c("Plate083", "Plate084", "Plate085")

OD_raw20<- OD_raw19 %>% dplyr::filter(!(Plate %in% contPlates))

##### Remove replicate of Israel_1992_ISY_Ar_4f in Plate097
Plati2<-c("Plate097", "Plate095", "Plate096")
OD_raw20<- OD_raw20 %>% dplyr::filter(!(Plate %in% Plati2 & Reduced_strainname == "ISY_Ar_4f"))

#### Remove high temperature of 12C climate chamber error

Plati3<-c("Plate092", "Plate095", "Plate098")
OD_raw20<- OD_raw20 %>% dplyr::filter(!(Plate %in% Plati3 & Condition == "12C"))




#### Removal of possibly contaminated wells of the Negative_control and Normalization of the Absorbance

####To do this, my strategy is to calculate the standard deviation of the Absorbance at the last time point (t12) for the Negative_control labelled wells in all the plates. But first, I noticed that I did a mistake and for some Plates I mistook the time point when I manually noted the timepoint. After that, the Absorbance correction and normalization.

OD_raw20$Col<-as.character(OD_raw20$Col)

##### Time point relabeling 

timepoint_relabeling <- data_frame(paste("t", 0:12, sep=""), c(paste("t0", 0:9, sep=""),paste("t", 10:12, sep="")),paste(1:13))%>%
  `colnames<-`(c("original", "adjusted", "clustering_output_timepoints")) 


for (i in 1: dim(timepoint_relabeling)[1]){
      OD_raw20$Timepoint[OD_raw20$Timepoint %in% timepoint_relabeling[i,"original"]%>%
      unlist() %>%
      as.vector()] <-
    timepoint_relabeling[i,"adjusted"]%>%
      unlist() %>%
      as.vector()
}


##### Correct the wrongly time point assessment (these are typos that were done when naming the files to extract the information of the plates)

error1<-OD_raw20 %>% filter(Strain=="Australia_1979to1991_WAC8444_57t" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_1979to1991_WAC8444_57t" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)


error1<-OD_raw20 %>% filter(Strain=="Australia_1979to1991_WAC8448_61t" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_1979to1991_WAC8448_61t" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Australia_2001_STAus01_1A5" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_2001_STAus01_1A5" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Australia_2001_STAus01_1B7" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_2001_STAus01_1B7" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Australia_2001_STAus01_1E1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_2001_STAus01_1E1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Australia_2001_STAus01_1F3" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_2001_STAus01_1F3" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Australia_2001_STAus01_1G2" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Australia_2001_STAus01_1G2" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="California_1989_CF11A_ST44" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="California_1989_CF11A_ST44" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="California_1989_CG22B_ST69" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="California_1989_CG22B_ST69" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Canada_1992_CNRD_4a_2" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Canada_1992_CNRD_4a_2" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Chile_1995_STCH95_1F2a" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Chile_1995_STCH95_1F2a" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Chile_1995_STCH95_1H1a" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Chile_1995_STCH95_1H1a" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Czech_Republic_2010_ST10CRI_348" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Czech_Republic_2010_ST10CRI_348" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Denmark_1994_STDN94C3a" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Denmark_1994_STDN94C3a" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Denmark_1994_STDN94Ee2a" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Denmark_1994_STDN94Ee2a" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Germany_1992_GEB_3a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Germany_1992_GEB_3a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Indiana_1993_I1a_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Indiana_1993_I1a_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Indiana_1993_I33a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Indiana_1993_I33a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Iran_2001_STIR01B_76a" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Iran_2001_STIR01B_76a" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Israel_1992_ISYF3a_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Israel_1992_ISYF3a_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Israel_1992_ISY_Ar_17e" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Israel_1992_ISY_Ar_17e" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Israel_1992_ISY_Ar_22f" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Israel_1992_ISY_Ar_22f" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Missouri_1994_ST94MO12a_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Missouri_1994_ST94MO12a_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Missouri_1994_ST94MO17a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Missouri_1994_ST94MO17a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_1990_a12_3B_3" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_1990_a12_3B_3" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_1990_a15_2A_8" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_1990_a15_2A_8" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_1990_a15_4A_13" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_1990_a15_4A_13" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_1990_a15_4A_4" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_1990_a15_4A_4" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_2015_ST15ORE_Mad_C5" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_2015_ST15ORE_Mad_C5" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_2015_ST15ORE_Mad_D6" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_2015_ST15ORE_Mad_D6" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Oregon_2015_ST15ORE_Ste_A17" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Oregon_2015_ST15ORE_Ste_A17" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Switzerland_1999_ST99CH_3A8" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Switzerland_1999_ST99CH_3A8" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Switzerland_1999_ST99CH_3B4" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Switzerland_1999_ST99CH_3B4" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Switzerland_1999_ST99CH_3C4" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Switzerland_1999_ST99CH_3C4" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Switzerland_1999_ST99CH_3F1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Switzerland_1999_ST99CH_3F1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Switzerland_1999_ST99CH_3H4" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Switzerland_1999_ST99CH_3H4" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Texas_1994_ST94TXPD2a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Texas_1994_ST94TXPD2a_1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Uruguay_1993_STUR93_EG4a.1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Uruguay_1993_STUR93_EG4a.1" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Strain=="Uruguay_1993_STUR93_LF2b_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Strain=="Uruguay_1993_STUR93_LF2b_1" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Plate == "Plate047" & Strain=="Negative_control" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Plate == "Plate047" & Strain=="Negative_control" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Plate == "Plate050" & Strain=="Negative_control" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08") %>% mutate(Timepoint="t04")
OD_raw20<-OD_raw20 %>% filter(!(Plate == "Plate050" & Strain=="Negative_control" & Condition =="18C" & Timepoint=="t12" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)

error1<-OD_raw20 %>% filter(Plate == "Plate047" & Strain=="Oregon_2015_ST15ORE_Ste_A11" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08") %>% mutate(Timepoint="t12")
OD_raw20<-OD_raw20 %>% filter(!(Plate == "Plate047" & Strain=="Oregon_2015_ST15ORE_Ste_A11" & Condition =="18C" & Timepoint=="t04" & Date=="2022-06-08")) 
OD_raw20<-full_join(OD_raw20, error1)


corradmx<-Correspondencefile %>% left_join(k8admixture) %>% ungroup() %>% dplyr::select(-Latitude, -Longitude, -Variety, -Year, -Geographic_region, -Location, -Location_year, -Country, -Trust_in_location) %>% left_join(newcoords) %>% left_join(climateclass)
corradmx$Year<-as.character(corradmx$Year)
condalltempplusnegcont<-c(Condition_all_temperatures, "ODLine", "ODLines")

#Add the ID_genome column and only keep temperature variables

OD_raw20<- OD_raw20 %>% dplyr::select(-Year, -Location, -Location_year, -Country) %>% filter(Condition %in% condalltempplusnegcont) %>% left_join(corradmx) 

OD_raw20a<- OD_raw20 %>% dplyr::select(-Year,  -Location, -Location_year, -Country) %>% left_join(corradmx) %>% filter(!(ID_genome_file %in% missing20rem))  %>% filter(!(ID_genome_file %in% isolateswrongstructure))


write.csv(OD_raw20a ,  "./01_Source_Tables/OD_results_nonegativecontrolfilter.csv", row.names = FALSE)

forpubOD_rawa<-anti_join(OD_raw20, OD_raw20a) %>% mutate(Reason_removed="Wrong_genetic_cluster_or_missing_data")

forpubOD_raw<-left_join(OD_raw20, forpubOD_rawa)


write.csv(forpubOD_raw, "./01_Source_Tables/OD_results_rawtemperatures.csv", row.names = FALSE)

```

```{r negcont, eval=FALSE}


##### Negative_control analysis 

negcont <- OD_raw20a %>% dplyr::filter(Reduced_strainname == "Negative_control")

p00<-ggplot(negcont, aes(x=Timepoint, y=Absorbance)) +
  geom_point() + 
  geom_label(data=negcont %>% filter(Absorbance > 0.2), aes(label=Cell)) + ggtitle("Absorbance of Negative_control wells per Timepoint") + geom_hline(yintercept = mean(negcont$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont$Absorbance)+sd(negcont$Absorbance)), color='red', lty='dashed')

p00

coli<-c("1", "12")
negcont$Col<-as.character(negcont$Col)
negcont1and12 <- negcont %>% dplyr::filter(Col %in% coli)

p01<-ggplot(negcont1and12, aes(x=Timepoint, y=Absorbance)) +
  geom_point() + 
  geom_label(data=negcont1and12 %>% filter(Absorbance > 0.2), aes(label=Cell)) + ggtitle("Absorbance of Negative_control wells per Timepoint columns 1 and 12 only")+ geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')

p01

negcontt12<-OD_raw20a %>% dplyr::filter(Timepoint == "t12" & Reduced_strainname == "Negative_control")

p02<-ggplot(negcontt12, aes(x=Plate, y=Absorbance)) +
  geom_point() + 
  geom_label(data=negcontt12 %>% filter(Absorbance > 0.2), aes(label=Cell)) + ggtitle("Absorbance of Negative_control wells per Plate") + geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')


p03<-ggplot(negcontt12, aes(x=Absorbance)) + 
 geom_histogram(aes(y=..density..), colour="black", fill="white")+
 geom_density(alpha=.2, fill="#FF6666") + ggtitle("Density Plot of the Absorbance of Negative_control wells") 


min.mean.sd.max <- function(x) {
  r <- c(min(x), mean(x) - sd(x), mean(x), mean(x) + sd(x), max(x))
  names(r) <- c("ymin", "lower", "middle", "upper", "ymax")
  r
}

# Boxplot with Standard deviation and Mean modification to boxplots
p04 <- ggplot(aes(y = Absorbance, x = Plate), data = negcontt12)
p04 <- p04 + stat_summary(fun.data = min.mean.sd.max, geom = "boxplot") + geom_jitter(position=position_jitter(width=.2), size=3) + ggtitle("Boxplots of absorbance per plate based delimited by the mean and standard deviation") + geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')


p05 <- ggplot(aes(y = Absorbance, x=factor(Col)), data = negcontt12)
p05 <- p04 + stat_summary(fun.data = min.mean.sd.max, geom = "boxplot") + geom_jitter(position=position_jitter(width=.2), size=3) + ggtitle("Boxplots of absorbance per column based delimited by the mean and standard deviation")+ geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')


#I will remove when I had negative control in other lines than 1 and 12

negcontt12$Col<- as.character(negcontt12$Col)

negcontt12col1and12<- negcontt12 %>% dplyr::filter(Col %in% coli)

p06 <- ggplot(aes(y = Absorbance, x=factor(Col)), data = negcontt12col1and12)
p06 <- p06 + stat_summary(fun.data = min.mean.sd.max, geom = "boxplot") + geom_jitter(position=position_jitter(width=.2), size=3) + ggtitle("Boxplots of absorbance per column based delimited by the mean and standard deviation with only columns 1 and 12")+ geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')


negcontt12col1and12sd<- negcontt12col1and12 %>% dplyr::filter(Absorbance< (mean(Absorbance)+sd(Absorbance)))

p07 <- ggplot(aes(y = Absorbance, x=factor(Col)), data = negcontt12col1and12sd)
p07 <- p07 + stat_summary(data=negcontt12col1and12, fun.data = min.mean.sd.max, geom = "boxplot") + geom_jitter(position=position_jitter(width=.2), size=3) + ggtitle("Boxplots of absorbance per column based delimited by the mean and standard deviation with only columns 1 and 12 after cutting the data by >mean+sd") + geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')

 
p08 <- ggplot(aes(y = Absorbance, x=factor(Plate)), data = negcontt12col1and12)
p08 <- p08 + stat_summary(fun.data = min.mean.sd.max, geom = "boxplot") + geom_jitter(position=position_jitter(width=.2), size=3) + ggtitle("Boxplots of absorbance per Plate based delimited by the mean and standard deviation with only columns 1 and 12") + geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')


p09 <- ggplot(aes(y = Absorbance, x=factor(Plate)), data = negcontt12col1and12sd)
p09 <- p09 + stat_summary(fun.data = min.mean.sd.max, geom = "boxplot", data=negcontt12col1and12) + geom_jitter(position=position_jitter(width=.2), size=3) + ggtitle("Boxplots of absorbance per Plate based delimited by the mean and standard deviation with only columns 1 and 12 after cutting the data by >mean+sd") + geom_hline(yintercept = mean(negcont1and12$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(negcont1and12$Absorbance)+sd(negcont1and12$Absorbance)), color='red', lty='dashed')


##### Cut off of the Negative_control

#It seems that there is plenty of data points of Negative_control wells in all plates, so I should get the limit value and apply it to the rest of the dataset. 

cutoff_negative_control<- mean(negcontt12col1and12$Absorbance) + sd(negcontt12col1and12$Absorbance)
cutoff_negative_control

cutoff_negcont_alltimepoints<-mean(negcont1and12$Absorbance) + (6*sd(negcont1and12$Absorbance))
cutoff_negcont_alltimepoints # I think I will stay with this one. 

##### Removal of Negative_control "contaminated" wells


# Remove negative control values out of lines 1 and 12
colo<-c("2", "3", "4", "5", "6", "7", "8", "9", "10", "11")
OD_raw21<- OD_raw20a %>% dplyr::filter(!(Reduced_strainname =="Negative_control" & Col %in% colo ))
OD_raw22 <- OD_raw21 %>% dplyr::filter(!(Reduced_strainname =="Negative_control" & Absorbance > cutoff_negcont_alltimepoints ))




negcontt12col1and12plussd<- negcontt12col1and12 %>% dplyr::filter(Absorbance > cutoff_negcont_alltimepoints) %>% ungroup() # 25 wells!
OD_raw_nonegcont<- OD_raw21 %>% dplyr::filter(!(Reduced_strainname == "Negative_control"))


OD_raw_negcont<- anti_join(negcont1and12, negcontt12col1and12plussd, by=c("Cell", "Plate", "Condition")) 

#Removed the wells that in t12 are above the cut-off of the standard dev. of the mean (possibly contaminated)

OD_raw23<- full_join(OD_raw_negcont, OD_raw_nonegcont) 


##### Take the average values of the Negative_control per time point and plate and subtract that of the rest of the wells. 

p10<-ggplot(OD_raw_negcont, aes(x=Timepoint, y=Absorbance)) +
  geom_point() + 
  geom_label(data=OD_raw_negcont %>% filter(Absorbance > 0.2), aes(label=Cell)) + ggtitle("Absorbance of Negative_control wells per Timepoint") + geom_hline(yintercept = mean(OD_raw_negcont$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(OD_raw_negcont$Absorbance)+sd(OD_raw_negcont$Absorbance)), color='red', lty='dashed')


p10

#Still t3 has this H12 well that is so crazy, also t09 B12. Does not make sense. Maybe just run the cut-off everywhere. 

negcontcol1and12plussd<- negcont1and12 %>% dplyr::filter(Absorbance > cutoff_negcont_alltimepoints)


OD_raw_negcontall<- anti_join(negcont1and12, negcontcol1and12plussd, by=c("Cell", "Plate", "Condition")) 



OD_raw24<- full_join(OD_raw_negcontall, OD_raw_nonegcont) 



p11<-ggplot(OD_raw_negcontall, aes(x=Timepoint, y=Absorbance)) +
  geom_point() + 
  geom_label(data=OD_raw_negcontall %>% filter(Absorbance > 0.2), aes(label=Cell)) + ggtitle("Absorbance of Negative_control wells per Timepoint") + geom_hline(yintercept = mean(OD_raw_negcontall$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(OD_raw_negcontall$Absorbance)+sd(OD_raw_negcontall$Absorbance)), color='red', lty='dashed')

p11


p12<-ggplot(OD_raw_negcontall, aes(x=Plate, y=Absorbance, color=Timepoint)) +
  geom_point() + 
  geom_label(data=OD_raw_negcontall %>% filter(Absorbance > 0.2), aes(label=Cell)) + ggtitle("Absorbance of Negative_control wells per Plate") + geom_hline(yintercept = mean(OD_raw_negcontall$Absorbance, na.rm=TRUE), color='blue', lty='dashed') + geom_hline(yintercept = (mean(OD_raw_negcontall$Absorbance)+sd(OD_raw_negcontall$Absorbance)), color='red', lty='dashed')

### enough data points per plate and time points 



write.csv(OD_raw24, "./01_Source_Tables/OD_results_negativecontrolfilter.csv", row.names = FALSE)

forpubOD_raw1a<-anti_join(forpubOD_raw, OD_raw24) %>% mutate(Reason_removed="Negative_control_contamination")

forpubOD_raw1<-full_join(forpubOD_raw, forpubOD_raw1a)


write.csv(forpubOD_raw1, "./01_Source_Tables/OD_results_rawtemperatures.csv", row.names = FALSE)


```

# Optical Density (OD/Absorbance) normalization with calibration curves

The first step is to remove the glucose peptone liquid (GPL) media (negative control) wells that were contaminated (Standard Deviation based filtering) and remove the strains that either have more than 20% of missing genetic data or that correspond to the wrong genetic cluster (most likely due to mislabeling errors).

Here are the first 10 lines of the data table with the date, time, condition, well information, absorbance, strain information.

Then, the absorbance normalization was performed by removing from the absorbance values the blank (media only wells) per plate.


```{r start, error=FALSE, message=FALSE, warning=FALSE, eval=FALSE}

forpubOD_raw1.1<-read.csv("./01_Source_Tables/OD_results_rawtemperatures.csv") 

forpubOD_raw1<-forpubOD_raw1.1%>% filter(Strain != "Negative_control") 

forpubOD_raw1.2<-forpubOD_raw1.1%>% filter(Strain == "Negative_control") 

forpubOD_raw1$ID_genome_file<- forpubOD_raw1$ID_genome_file %>% replace_na("Non")

forpubOD_raw1a<-forpubOD_raw1 %>% filter(ID_genome_file == "Non") %>% mutate(Reason_removed="No_genome_file")


forpubOD_raw1b<-forpubOD_raw1 %>% filter(ID_genome_file != "Non")

forpubOD_raw2<-full_join(forpubOD_raw1b, forpubOD_raw1a)

forpubOD_raw2.1<-full_join(forpubOD_raw2, forpubOD_raw1.2)


write.csv(forpubOD_raw2.1, "./01_Source_Tables/OD_results_rawtemperatures.csv", row.names = FALSE)


```

```{r start1, error=FALSE, message=FALSE, warning=FALSE}
  
remove<-c("Negative_control_contamination", "Wrong_genetic_cluster_or_missing_data", "No_genome_file")

OD_raw24<-read.csv("./01_Source_Tables/OD_results_rawtemperatures.csv") %>% filter(!(Reason_removed %in% remove))



OD_raw24 %>% head(10) %>% kable(caption="First 10 data points")%>% kable_styling()

forpubOD_raw1<-read.csv("./01_Source_Tables/OD_results_rawtemperatures.csv")%>% unite(col = "Date", Date, Time, sep = " ") %>% mutate(Date = as.POSIXct(Date, format = "%Y-%m-%d %H:%M:%S"))

```

```{r ODnorm, error=FALSE, message=FALSE, warning=FALSE}

##### Absorbance Normalization

#Separate the OD_lines from the actual measurements
OD_lines = OD_raw24 %>% filter(Condition %in% c("ODLine", "ODLines"))
OD_results = OD_raw24 %>% filter(!(Condition %in% c("ODLine", "ODLines")))

#As negative control, the average of the absorbance of the 16 cells (or less) with just the media per plate per time point was used. Each plate corresponds to one treatment and it has 20 strains tested simultaneously for the same condition with 4 well replicates. 

average_neg_control_OD_lines = OD_lines %>%
  filter(Strain == "Negative_control") %>%
  group_by(Plate, Condition, Timepoint, Date) %>%
  dplyr::summarize(Absorbance_control = mean(Absorbance))

average_neg_control = OD_results %>%
  filter(Strain == "Negative_control") %>%
  group_by(Plate, Condition, Timepoint, Date) %>%
  dplyr::summarize(Absorbance_control = mean(Absorbance))


OD_results  = left_join(OD_results, average_neg_control)%>%
  mutate(Norm_absorbance = Absorbance - Absorbance_control) %>% unite(col = "Date", Date, Time, sep = " ") %>% mutate(Date = as.POSIXct(Date, format = "%Y-%m-%d %H:%M:%S"))


OD_results = ungroup(OD_results) %>%
  filter(Strain != "Negative_control") 
#OD_results[OD_results=='NA'] <- NA
#new_DF <- OD_results[is.na(OD_results$Absorbance),]
#lala<-as.data.frame(!complete.cases(OD_results))
#sum(!complete.cases(OD_results))
#sum(!complete.cases(OD_lines))


#negativedodresults<-OD_results %>% filter(Norm_absorbance <= 0)
#OD_results[OD_results$Norm_absorbance < 0, "Norm_absorbance"] = 0 # Not possible to have negative OD values



```

## Calibration curves {.tabset}

After normalizing the absorbance, which means remove the negative control (blank) value from the absorbance value of the wells with the strains, the calibration curves were obtained for all the strains and the differences between the calibration curves (which I also refer to as OD lines) explored. The calibration curves were obtained through a linear regression of 8 different concentrations' absorbance per each strain. In the different tabs are found different plots to test if there were biases by plate (time of the year when the experiment was done), year of collection of the strains or the geographical region they come from.

```{r ODlines, error=FALSE, message=FALSE, warning=FALSE}

### Taken and modified from Dani Osoko's work in July 2022
#Linking OD_line Rows to Concentration

OD_lines = OD_raw24 %>% filter(Condition %in% c("ODLine", "ODLines"))
concentration_OD = data_frame(
  c("A", "B", "C", "D", "E", "F", "G", "H"),
  c(2000000, 1750000, 1500000, 1000000, 750000, 500000, 250000, 100000))%>% #spores per mL
  `colnames<-`(c("Row", "Concentration"))

#OD with concentration for OD_lines -> OD_results get overwritten
OD_lines <- full_join(OD_lines, concentration_OD, by ="Row")


OD_lines  = left_join(OD_lines, average_neg_control_OD_lines)%>%
  mutate(Norm_absorbance = Absorbance - Absorbance_control) %>%
  unite(col = "Date", Date, Time, sep = " ") %>%
  mutate(Date = as.POSIXct(Date, format = "%Y-%m-%d %H:%M:%S"))

OD_lines = ungroup(OD_lines) %>%
  filter(Strain != "Negative_control")


OD_lines[OD_lines$Norm_absorbance <= 0, "Norm_absorbance"] = 0

zero_point_for_regression <- OD_lines
zero_point_for_regression$Absorbance = 0
zero_point_for_regression$Concentration = 0
zero_point_for_regression$Absorbance_control = 0
zero_point_for_regression$Norm_absorbance = 0
zero_point_for_regression = zero_point_for_regression %>% distinct()

OD_lines = rbind(OD_lines, zero_point_for_regression)%>%
  dplyr::arrange(Strain, Plate)%>%
  dplyr::select(-c(Row, Cell, Col, Date, Timepoint))%>% distinct()



temp_df <- OD_lines %>% 
  dplyr::select(c(Strain, Plate))
temp_df <- bind_cols(temp_df,as.data.frame(paste(temp_df$Plate,
                                                 temp_df$Strain,
                                                 sep = "_")))%>%
  `colnames<-`(c("Strain", "Plate", "Strain_Plate"))%>%
  distinct()%>%
  dplyr::arrange(desc(Strain_Plate))

#Prename List for easier analysis
plot_list_regression <-vector(mode = "list",
                   length = dim(temp_df)[1])
names(plot_list_regression) <- as.character(temp_df$Strain_Plate)
stat_list_regression=plot_list_regression


regression_collection_data_frame <- data_frame()

#Loop for each Strain (counting each replicate as one (e.g. strain on plate 32 is seen as 3 different strains))
for (i in 1:dim(temp_df)[1]){
  
  temp <- OD_lines %>%
    dplyr::arrange(Strain, Plate) %>%
    filter(Strain == temp_df$Strain[i])%>%
    filter(Plate == temp_df$Plate[i])
  
  #Linear Regression
  temp_lm <-
    lm(I(Concentration-0) ~ (0 + Norm_absorbance), temp)
  
  #Taking out Slope and Intercept
  needed_info = summary(temp_lm)$coefficients
  adj_r_squared = summary(temp_lm)$adj.r.squared
  
  #temp_df$Strain_Plate[i] is the name of the current strain and plate --> assigns the found data to the corresponding spot in the list
  stat_list_regression[[temp_df$Strain_Plate[i]]]=summary(temp_lm)
  
  #appends new data to the collection
  regression_collection_data_frame <-rbind(regression_collection_data_frame,
                                           data.frame(temp_df[i,c(1,2)], needed_info[1, 1], 0, adj_r_squared) %>%
                                             `colnames<-`(c("Strain", "Plate","slope", "intercept", "Adjusted_R_squared")))
  
}

regression_collection_data_frame=regression_collection_data_frame%>%
  dplyr::arrange(desc(Plate))

```

### Strains with non-optimal linear regressions

The adjusted R-squared of these strains calibration in the table is lower than 0.8. There is no particular pattern of which population's strains have non-optimal linear regressions. From the 411 strains, 30 strains in total have a non-optimal calibration curves, but for most the adjusted R-squared is still above 0.7.

```{r ODLinesbadreg, error=FALSE, message=FALSE, warning=FALSE}


badreg_ODLines<-regression_collection_data_frame%>% dplyr::filter(Adjusted_R_squared <=0.8)

knitr::kable(badreg_ODLines%>%arrange(desc(Plate)), caption="Calibration curves with suboptimal fit.")%>% kable_styling()

odbadrsqstrains<-unique(badreg_ODLines$Strain) 

numberstrainsODlines<-OD_lines %>% group_by(Strain) %>% summarize(numb = n()) %>% dplyr::select(-numb) 
numberstrainsODresults<-OD_results %>% group_by(Strain) %>% summarize(numb = n()) %>% dplyr::select(-numb)

```

### Calibration curves' slope density per plate

The plates that were done at the beginning of the experiment tend to have higher slope values, but it could be because of the random selection of samples of the later plates were the samples with higher slope values. Non of the instruments or materials were changed during the time of the experiments.

```{r ODLinesdenslop, fig.cap="Calibration curves' slope density per plate.", error=FALSE, message=FALSE, warning=FALSE, fig.width=10}

#Ploting slopes of regressions by the plates

ggplot(regression_collection_data_frame, aes(x = slope,color = Plate, fill = Plate)) + 
  geom_density(alpha = 0.25, 
               position = "identity") +
  scale_fill_manual(values = palette_infinite(length(unique(regression_collection_data_frame$Plate))))+
  scale_color_manual(values = palette_infinite(length(unique(regression_collection_data_frame$Plate))))+
  theme(legend.position = "bottom")+
  labs(title = "Plates' Density of Strain Slopes") +theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))

```

### Calibration curves' slope density per year of strain collection

There is no clear pattern that shows that the calibration curves' slope is higher or lower for a particular year, or that a pattern exist showing a higher or lower slope from the oldest collections to the newest.

```{r ODLinesdenyear, fig.cap="Calibration curves' slope density per year of strain collection.", error=FALSE, message=FALSE, warning=FALSE, fig.width=10}

OD_lines1<- full_join(OD_lines, regression_collection_data_frame)
OD_lines1$Year<-as.character(OD_lines1$Year)

Year_denODLinesplot<-ggplot(OD_lines1, aes(x = slope, color = Year, fill = Year)) + 
  geom_density(alpha = 0.25, 
               position = "identity") +
  scale_fill_manual(values = palette_infinite(length(unique(OD_lines1$Year))))+
  scale_color_manual(values = palette_infinite(length(unique(OD_lines1$Year))))+
  theme(legend.position = "bottom")+
  labs(title = "Year's Density of Strain Slopes")+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))

Year_denODLinesplot

```

### Calibration curves' slope density per geographic region of origin

There is no clear pattern that shows that the calibration curves' slope is higher or lower for a particular region, or at least not very evident, it seems there are outliers in all geographic regions. There is specially a big group for Oceania (Australia) in the higher end of the slopes.

```{r ODLinesdencont, fig.cap="Calibration curves' slope density per genetic cluster.", error=FALSE, message=FALSE, warning=FALSE, fig.width=10}


Continent_denODLinesplot<-ggplot(OD_lines1, aes(x = slope,color = Adm_Cluster, fill = Adm_Cluster)) + 
  geom_density(alpha = 0.15, 
               position = "identity") +
  scale_fill_manual(values = K_colors)+
  scale_color_manual(values =  K_colors)+
  theme(legend.position = "bottom")+
  labs(title = "Genetic Clusters' Density of Strain Slopes")+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))

Continent_denODLinesplot

```

### Average Calibration Curve per Geographic Region

Here we can observe that the absorbance values for Middle-eastern and North American strains calibration curves are much higher than for other geographical regions, and that the African strains are the ones with lower absorbance values.

```{r ODLinescontaver, fig.cap="Average Calibration Curve per Geographic Region.", error=FALSE, message=FALSE, warning=FALSE, fig.width=10}

##### OD Line as Phenotype

temp = OD_lines%>%
  group_by(Geographic_region, Concentration) %>%
  dplyr::summarize(Norm_absorbance = mean(Norm_absorbance))%>%
  filter(Concentration > 0)

temp$Concentration_label <- as.character(temp$Concentration) #needed for grouping in plots as concentration is numeric

OD_as_Phenotype = temp %>%
  ggplot(aes(x = reorder(Concentration_label,Concentration), y = Norm_absorbance, group = Concentration, color = Geographic_region)) + 
  geom_point(size = 5,
             alpha = .6) +
  geom_line(aes(group = Geographic_region), alpha = .3, size=5) +theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 270),
    legend.position = "right"
  ) +
  labs(y = "Normalised Absorbance", title = "Average Calibration Curve per Geographic Region", x="Spore Concentration (spores/mL)") +
  scale_y_continuous(labels = scientific,
                     limits = c(min(temp$Norm_absorbance),
                                max(temp$Norm_absorbance))) + theme(axis.line = element_line(colour = "grey50"))

OD_as_Phenotype





```

## Spore concentration calculation from calibration curves IQR Filtering to remove Tecan machine artifacts between the 3 machine runs

First, using the linear regression formula, the calibration curves were then used to obtain the spore concentration (cell density). The values of cell densities are very large so they were transformed by now using the units 10\^6 spores/mL for making the next calculations easier. If there are some wells in the microtiter plate (there are 4 wells per treatment per strain) that might be have a very disperse values between the three runs per time point, we performed a per time point IQR filtering step to remove the machine artifacts.   

```{r calibration, fig.cap="Density plots of maximum absolute difference and the logarithm base 10 + 0.001 of the maximum absolute difference between the three machine runs per time point.", message=FALSE, warning=FALSE, error=FALSE}

##### It takes Regression Table and Changes Line plate to the corresponding results plate

Key_Lines_to_Results = read_excel(paste0(analysis_dir, "01_Source_Tables/Key_Plate_Lines.xlsx"))

converted_regression_collection_data_frame = regression_collection_data_frame

for (i in 1:dim(converted_regression_collection_data_frame)[1]) {
  for (j in 1:length(Key_Lines_to_Results$Line_Plate_1)){
   
     if(regression_collection_data_frame$Plate[i] %in% Key_Lines_to_Results$Line_Plate_1[j]| regression_collection_data_frame$Plate[i] %in% Key_Lines_to_Results$Line_Plate_2[j])
       {
  converted_regression_collection_data_frame$Plate[i] = Key_Lines_to_Results$Result_Plate[j]
     }
    else{}
  }
}
  
OD_resultsconv = OD_results %>%
  left_join(converted_regression_collection_data_frame, by = c("Strain", "Plate"))%>%
  ungroup()


# Following the linear regression formula

OD_resultsconv$Concentration <- OD_resultsconv$Norm_absorbance * OD_resultsconv$slope + OD_resultsconv$intercept 



#nacasesodresultsconv<- OD_resultsconv %>% subset(is.na(OD_resultsconv$Concentration))
#numberstrainsODresultsconv<-OD_resultsconv %>% group_by(Strain) %>% summarize(numb = n()) %>% dplyr::select(-numb) # control in case I am losing data

# Transformation of concentration units from value to value*10^6.

OD_resultsconv <- OD_resultsconv %>% ungroup() %>% mutate(Concentration=Concentration/1000000)


OD_resultsconv$Concentration[OD_resultsconv$Concentration < 0]<- 0 #Make negative concentrations equal to 0.
#OD_resultsconv %>% filter(Concentration<0) # Checkpoint


#write_csv(OD_resultsconv, "OD_resultsconv.csv")

#OD_resultsconv<-read_csv("OD_resultsconv.csv")

sd_od_resultsconv<-ungroup(OD_resultsconv) %>% 
    dplyr::select(Strain, Condition, Cell, Timepoint, Concentration, Repet) %>% 
      pivot_wider(names_from = Repet, names_prefix="Repet_", values_from = Concentration) 


sd_od_resultsconv1<-sd_od_resultsconv %>% 
  rowwise() %>%  
  mutate(Compare1and2=abs(Repet_1 - Repet_2), 
         Compare2and3=abs(Repet_2 - Repet_3), 
         Compare1and3=abs(Repet_1 - Repet_3)) %>% 
  rowwise() %>% 
  mutate(Max_diff=max(c(Compare1and2, Compare2and3, Compare1and3), na.rm = T), 
         Min_diff=min(c(Compare1and2, Compare2and3, Compare1and3), na.rm = T)) %>% 
  drop_na(Repet_2) #This drop_na removes the measurements where I only have one run of the machine.


per_timepoint_sumstat = sd_od_resultsconv1 %>%
  group_by(Timepoint) %>%
  summarize(IQR_Max_diff = IQR(Max_diff, na.rm = T),
            Q3_IQR = quantile(Max_diff, na.rm = T, probs = 0.75),
            per_timepoint_threshold = IQR_Max_diff *3 + Q3_IQR)
ggplot() +
  geom_density(data = sd_od_resultsconv1 %>% filter(Max_diff>0), aes(Max_diff), fill = "#F3A712", alpha=0.8) +
  geom_vline(data = per_timepoint_sumstat, aes(xintercept = per_timepoint_threshold))+ 
  facet_grid(rows = vars(Timepoint), scales = "free_y")

ggplot() +
  geom_density(data = sd_od_resultsconv1%>% filter(Max_diff>0), aes(Max_diff + 0.001), fill = "#F3A712", alpha=0.8) + 
  geom_vline(data = per_timepoint_sumstat, aes(xintercept = per_timepoint_threshold))+ 
  scale_x_log10() +
  facet_grid(rows = vars(Timepoint), scales = "free_y") +
  theme_minimal() +
  labs(x = "log10(Max_diff)") 

```


```{r iqrfilmachineruns, fig.cap="Number of measurements that were filtered or kept After Machine Run IQR filtering per condition, Frequency Distribution of Concentration Observations per Temperature Before Machine Run IQR filtering and Frequency Distribution of Concentration Observations per Temperature After Machine Run IQR filtering.", message=FALSE, warning=FALSE, error=FALSE}

sd_od_result_filtered <-sd_od_resultsconv1  %>% 
  left_join(per_timepoint_sumstat) %>% 
  mutate(RemoveRepet_1=ifelse( (Compare1and2 > per_timepoint_threshold) & (Compare1and3 > per_timepoint_threshold), "Yes", "No")) %>% 
  mutate(RemoveRepet_2=ifelse( (Compare1and2 > per_timepoint_threshold) & (Compare2and3 > per_timepoint_threshold), "Yes", "No")) %>% 
  mutate(RemoveRepet_3=ifelse( (Compare1and3 > per_timepoint_threshold) & (Compare2and3 > per_timepoint_threshold), "Yes", "No")) %>% 
  pivot_longer(cols=c("Repet_1", "Repet_2", "Repet_3"), names_to =c("Repet"), 
               values_to =c("Concentration"), names_prefix = c("Repet_"  )) %>% 
  pivot_longer(cols=c("RemoveRepet_1", "RemoveRepet_2", "RemoveRepet_3"), names_to =c("RRepet"), 
               values_to =c("RemoveIQR"), names_prefix = c("RemoveRepet_"  ))  %>%
  filter(Repet == RRepet)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
count_filtering =  sd_od_result_filtered %>% 
  count(Strain, Condition, Cell, Timepoint, RemoveIQR)

temp = count_filtering %>%
  count(Condition, RemoveIQR, Timepoint, n) %>%
  mutate(RemoveIQR2 = case_when(RemoveIQR == "No" & n == 3 ~ "All_kept",
                                  RemoveIQR == "Yes" & n == 3 ~ "All_filtered",
                                  RemoveIQR == "Yes" & n == 1 ~ "One_filtered",
                                  RemoveIQR == "Yes" & n == 2 ~ "Two_filtered")) %>%
  filter(!is.na(RemoveIQR2)) 

#temp %>%
#  dplyr::select(Condition, Timepoint, RemoveIQR2, nn) %>%
#  pivot_wider(names_from = RemoveIQR2, values_from = nn, values_fill = 0)

temp %>%
  ggplot(aes(Timepoint, fill = RemoveIQR2, y = nn)) +
  geom_bar(stat = "identity") +
  facet_wrap(vars(Condition)) +
  scale_fill_manual(values = c("#DB2B39", "grey10", "grey30")) +
  theme_bw()



OD_resultsconv %>% ggplot(aes(Concentration, fill=Timepoint)) + geom_histogram()+facet_wrap(~Condition, scales="free") + labs(x=expression(paste("Concentration (*", 10^6, " spores/mL)")), y="Number of observations (all wells and all timepoints)", title="Frequency Distribution of Concentration Observations per Temperature", subtitle ="Before Machine Run IQR per time point filtering")+ scale_fill_viridis_d(direction = -1)+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))


toremoverepet1 = sd_od_result_filtered  %>% 
  dplyr::select(-RRepet, -Compare1and2, -Compare2and3, -Compare1and3, -Min_diff, -Max_diff) %>% 
  filter(RemoveIQR == "Yes") %>%
  mutate(Repet = as.integer(Repet)) 

OD_resultsconv1<-ungroup(OD_resultsconv) %>% full_join(ungroup(toremoverepet1)) %>% 
  mutate(RemoveIQR=replace_na(RemoveIQR, "No")) %>% 
  filter(RemoveIQR == "No") 


write_csv(OD_resultsconv1, "OD_resultsconv_IQRartifactremoval.csv")


OD_resultsconv1 %>% ggplot(aes(Concentration, fill=Timepoint)) + geom_histogram()+facet_wrap(~Condition, scales="free")+ labs(x=expression(paste("Concentration (*", 10^6, " spores/mL)")), y="Number of observations (all wells and all timepoints)", title="Frequency Distribution of Concentration Observations per Temperature", subtitle ="After Machine Run IQR per time point filtering")+ scale_fill_viridis_d(direction = -1)+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))




forpubOD_raw3<-left_join(forpubOD_raw1,(ungroup(OD_resultsconv) %>% full_join(ungroup(toremoverepet1))  %>% dplyr::select( -per_timepoint_threshold, -IQR_Max_diff, -Q3_IQR))) %>% rename(Slope_calcurve=slope, Adj_rsquared_calcurve=Adjusted_R_squared)%>% 
  mutate(RemoveIQR=replace_na(RemoveIQR, "No"))

forpubOD_raw3a<-forpubOD_raw3 %>% filter(RemoveIQR == "Yes") %>% mutate(Reason_removed="IQR_run_artifact")

forpubOD_raw3b<-forpubOD_raw3 %>% filter(RemoveIQR == "No") 

forpubOD_raw3<-full_join(forpubOD_raw3a, forpubOD_raw3b)


write.csv(forpubOD_raw3, "./01_Source_Tables/OD_results_rawtemperatures_placeandcoord.csv", row.names = FALSE)


OD_resultsconv<-OD_resultsconv1

```

## IQR Filtering to remove wells that might have a non-replicable starting concentration or that were contaminated per timepoint 

If there are some wells in the microtiter plate (there are 4 wells per treatment per strain) that might be have a very disperse values between them per time point, we performed an IQR filtering step to remove possible pipetting and contamination errors. Having a very different starting concentration might affect the replicability of the results and also, the wells could have a contamination which cannot be noticed until some time has passed. Both of these situations affect the growth curve results.    

```{r iqrmed, fig.cap="Density plots of maximum absolute difference and the logarithm base 10 + 0.001 of the maximum absolute difference between the three machine runs per time point.", message=FALSE, warning=FALSE, error=FALSE}


iqrmed_od_resultsconv<-ungroup(OD_resultsconv) %>% 
    dplyr::select(Strain, Condition, Cell, Timepoint, Concentration, Repet) %>% group_by(Strain, Condition, Timepoint, Cell) %>% summarize(medianConcentration=median(Concentration)) %>% ungroup() %>% group_by(Strain, Condition, Timepoint) %>% mutate(id = rep(1:4, length.out = n())) %>% dplyr::select(-Cell) %>% ungroup() %>%  
      pivot_wider(names_from = id, names_prefix="Celln_", values_from = medianConcentration) %>% drop_na(Celln_2)#This drop_na removes the measurements where I only have one well's median.

iqrmed_od_resultsconvnomed<-ungroup(OD_resultsconv) %>% 
    dplyr::select(Strain, Condition, Cell, Timepoint, Concentration, Repet) %>% group_by(Strain, Condition, Timepoint, Cell) %>% summarize(medianConcentration=median(Concentration)) %>% ungroup() %>% group_by(Strain, Condition, Timepoint) %>% mutate(Celln = rep(1:4, length.out = n())) %>% dplyr::select(-medianConcentration) %>% ungroup()


iqrmed_od_resultsconv1<-iqrmed_od_resultsconv %>% 
  rowwise() %>%  
  mutate(Compare1and2=abs(Celln_1 - Celln_2), 
         Compare2and3=abs(Celln_2 - Celln_3), 
         Compare1and3=abs(Celln_1 - Celln_3),
         Compare1and4=abs(Celln_1 - Celln_4), 
         Compare2and4=abs(Celln_2 - Celln_4), 
         Compare3and4=abs(Celln_3 - Celln_4)) %>% 
  rowwise() %>% 
  mutate(Max_med_diff=max(c(Compare1and2, Compare2and3, Compare1and3, Compare1and4, Compare2and4, Compare3and4), na.rm=T), 
         Min_med_diff=min(c(Compare1and2, Compare2and3, Compare1and3, Compare1and4, Compare2and4, Compare3and4), na.rm=T))


per_timepoint_sumstat = iqrmed_od_resultsconv1 %>%
  group_by(Timepoint) %>%
  summarize(IQRmed_Max_diff = IQR(Max_med_diff, na.rm = T),
            Q3_IQRmed_Max = quantile(Max_med_diff, na.rm = T, probs = 0.75),
            med_per_timepoint_threshold = IQRmed_Max_diff *3 + Q3_IQRmed_Max)


ggplot() +
  geom_density(data = iqrmed_od_resultsconv1 %>% filter(Max_med_diff>0), aes(Max_med_diff), fill = "#F3A712", alpha=0.8) +
  geom_vline(data = per_timepoint_sumstat, aes(xintercept = med_per_timepoint_threshold))+ 
  facet_grid(rows = vars(Timepoint), scales = "free_y")



ggplot() +
  geom_density(data = iqrmed_od_resultsconv1%>% filter(Max_med_diff>0), aes(Max_med_diff + 0.001), fill = "#F3A712", alpha=0.8) + 
  geom_vline(data = per_timepoint_sumstat, aes(xintercept = med_per_timepoint_threshold))+ 
  scale_x_log10() +
  facet_grid(rows = vars(Timepoint), scales = "free_y") +
  theme_minimal() +
  labs(x = "log10(Max_diff)") 


```


```{r iqrmedfilter, fig.cap="Number of measurements that were filtered or kept After Median Concentration per Well IQR filtering per condition.", message=FALSE, warning=FALSE, error=FALSE}

iqrmed_od_result_filtered <-iqrmed_od_resultsconv1  %>% 
  left_join(per_timepoint_sumstat) %>% 
  mutate(RemoveCelln_1=ifelse( (Compare1and2 > med_per_timepoint_threshold) & (Compare1and3 > med_per_timepoint_threshold) & (Compare1and4 > med_per_timepoint_threshold), "Yes", "No")) %>% 
  mutate(RemoveCelln_2=ifelse( (Compare1and2 > med_per_timepoint_threshold) & (Compare2and3 > med_per_timepoint_threshold) & (Compare2and4 > med_per_timepoint_threshold), "Yes", "No")) %>% 
  mutate(RemoveCelln_3=ifelse( (Compare1and3 > med_per_timepoint_threshold) & (Compare2and3 > med_per_timepoint_threshold) & (Compare3and4 > med_per_timepoint_threshold), "Yes", "No")) %>% 
  mutate(RemoveCelln_4=ifelse( (Compare1and4 > med_per_timepoint_threshold) & (Compare2and4 > med_per_timepoint_threshold) & (Compare3and4 > med_per_timepoint_threshold), "Yes", "No")) %>% 
  pivot_longer(cols=c("Celln_1", "Celln_2", "Celln_3", "Celln_4"), names_to =c("Celln"), 
               values_to =c("medianConcentration"), names_prefix = c("Celln_"  )) %>% 
  pivot_longer(cols=c("RemoveCelln_1", "RemoveCelln_2", "RemoveCelln_3", "RemoveCelln_4"), names_to =c("RCelln"), 
               values_to =c("RemoveIQRmed"), names_prefix = c("RemoveCelln_"  ))  %>%
  filter(Celln == RCelln)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
med_count_filtering =  iqrmed_od_result_filtered %>% 
  count(Strain, Condition, Timepoint, RemoveIQRmed)

temp = med_count_filtering %>%
  count(Condition, RemoveIQRmed, Timepoint, n) %>%
  mutate(RemoveIQRmed2 = case_when(RemoveIQRmed == "No" & n == 4 ~ "All_kept",
                                  RemoveIQRmed == "Yes" & n == 4 ~ "All_filtered",
                                  RemoveIQRmed == "Yes" & n == 1 ~ "One_filtered",
                                  RemoveIQRmed == "Yes" & n == 2 ~ "Two_filtered",
                                  RemoveIQRmed == "Yes" & n == 3 ~ "Three_filtered")
         ) %>%
  filter(!is.na(RemoveIQRmed2)) 

#temp %>%
#  dplyr::select(Condition, Timepoint, RemoveIQRmed2, nn) %>%
#  pivot_wider(names_from = RemoveIQRmed2, values_from = nn, values_fill = 0)

temp %>%
  ggplot(aes(Timepoint, fill = RemoveIQRmed2, y = nn)) +
  geom_bar(stat = "identity") +
  facet_wrap(vars(Condition)) +
  scale_fill_manual(values = c("grey10", "#368F8B", "#246A73")) +
  theme_bw()


toremoverepet1 = iqrmed_od_result_filtered  %>% 
  dplyr::select(-RCelln, -Compare1and2, -Compare2and3, -Compare1and3, -Compare1and4, -Compare2and4, -Compare3and4, -Min_med_diff, -Max_med_diff, -medianConcentration) %>% 
  filter(RemoveIQRmed == "Yes") %>%
  mutate(Celln = as.integer(Celln)) %>% full_join(iqrmed_od_resultsconvnomed) %>% drop_na(med_per_timepoint_threshold)



OD_resultsconv1<-ungroup(OD_resultsconv) %>% full_join(ungroup(toremoverepet1)) %>% 
  mutate(RemoveIQRmed=replace_na(RemoveIQRmed, "No")) %>% 
  filter(RemoveIQRmed == "No") %>% dplyr::select(-Celln, -med_per_timepoint_threshold, -per_timepoint_threshold, -IQR_Max_diff, -Q3_IQR, -IQRmed_Max_diff, -Q3_IQRmed_Max)


write_csv(OD_resultsconv1, "OD_resultsconv_IQRartifactremoval_IQRcontandpipetteerrorremoval.csv")




forpubOD_raw4<-left_join(forpubOD_raw3,(ungroup(OD_resultsconv) %>% full_join(ungroup(toremoverepet1)))) %>% mutate(RemoveIQRmed=replace_na(RemoveIQRmed, "No")) %>% dplyr::select(-Celln, -med_per_timepoint_threshold, -per_timepoint_threshold, -IQR_Max_diff, -Q3_IQR, -IQRmed_Max_diff, -Q3_IQRmed_Max, -slope, -Adjusted_R_squared) 

forpubOD_raw4a<-forpubOD_raw4 %>% filter(RemoveIQRmed == "Yes") %>% mutate(Reason_removed="IQR_median_well")

forpubOD_raw4b<-forpubOD_raw4 %>% filter(RemoveIQRmed == "No") 

forpubOD_raw4<-full_join(forpubOD_raw4a, forpubOD_raw4b)

write.csv(forpubOD_raw4, "./01_Source_Tables/OD_results_rawtemperatures_placeandcoord.csv", row.names = FALSE)



OD_resultsconv<-OD_resultsconv1




```



# Growth curves per strain per treatment results

## Number of strains per Genetic Cluster

The number of strains per genetic cluster is represented below (Figure \@ref(fig:numbofstrains)), however, the category "NA" is the group of strains that did not have an individual ancestry coefficient higher than 0.75, therefore, they were not assigned to a particular genetic cluster. In total, the phenotypes and genotypes for a total of 411 strains were used for this study, however, due to poor model performance, depending on which analysis some of the growth-associated variables calculated have a different total number of strains employed.

```{r numbofstrains, message=FALSE, fig.cap="Number of strains per genetic cluster.", warning=FALSE, error=FALSE}


OD_resultsconv %>% dplyr::select(Strain, Adm_Cluster) %>% group_by(Adm_Cluster)%>% summarize(Strain=unique(Strain))%>% count(Adm_Cluster) %>% ggplot(aes(x=Adm_Cluster, y=n, fill=Adm_Cluster, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = K_colors)+labs(x="Genetic Cluster", y="Number of strains", title="Number of strains per Genetic Cluster")+ geom_text(vjust=1.6, color="white", size=5)

```

## Growth curves per strain per treatment {.tabset}

There are 12 points in the growth curve graphs per strain per time point due to the fact that is 4 replicate wells and 3 replicate measurement runs in the machine, unless they were removed by the filters above.

### Temperature 25C

```{r summaryod,message=FALSE, warning=FALSE, error=FALSE}
OD_resultsconvsummary<-OD_resultsconv %>% group_by(Date, Plate, Condition, Strain, ID_genome_file,Location, Geographic_region, Year, Country) %>%
 dplyr::summarize(Concentration = median(Concentration)) 
```

```{r 25growthcurves, fig.cap="Growth curves at 25C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv25C<-OD_resultsconv %>% filter(Condition == "25C") 

OD_resultsconvsummary25C<-OD_resultsconvsummary  %>% filter(Condition == "25C")

growthcurves25C<- ggplot(OD_resultsconv25C, aes(x = Date, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1), data = OD_resultsconvsummary25C)  +
  geom_point(data = OD_resultsconvsummary25C, size = 2)+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 25C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves25C


```

### Temperature 22C

```{r 22growthcurves, fig.cap="Growth curves at 22C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv22C<-OD_resultsconv %>% filter(Condition == "22C") 

OD_resultsconvsummary22C<-OD_resultsconvsummary  %>% filter(Condition == "22C")

growthcurves22C<- ggplot(OD_resultsconv22C, aes(x = Date, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1), data = OD_resultsconvsummary22C)  +
  geom_point(data = OD_resultsconvsummary22C, size = 2)+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 22C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves22C


```

### Temperature 18C

```{r 18growthcurves, fig.cap="Growth curves at 18C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv18C<-OD_resultsconv %>% filter(Condition == "18C") 

OD_resultsconvsummary18C<-OD_resultsconvsummary  %>% filter(Condition == "18C")

growthcurves18C<- ggplot(OD_resultsconv18C, aes(x = Date, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1), data = OD_resultsconvsummary18C)  +
  geom_point(data = OD_resultsconvsummary18C, size = 2)+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 18C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves18C


```

### Temperature 15C

```{r 15growthcurves, fig.cap="Growth curves at 15C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv15C<-OD_resultsconv %>% filter(Condition == "15C") 

OD_resultsconvsummary15C<-OD_resultsconvsummary  %>% filter(Condition == "15C")

growthcurves15C<- ggplot(OD_resultsconv15C, aes(x = Date, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1), data = OD_resultsconvsummary15C)  +
  geom_point(data = OD_resultsconvsummary15C, size = 2)+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 15C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves15C


```

### Temperature 12C

```{r 12growthcurves, fig.cap="Growth curves at 12C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv12c<-OD_resultsconv %>% filter(Condition == "12C") 

OD_resultsconvsummary12c<-OD_resultsconvsummary  %>% filter(Condition == "12C")

growthcurves12c<- ggplot(OD_resultsconv12c, aes(x = Date, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1), data = OD_resultsconvsummary12c)  +
  geom_point(data = OD_resultsconvsummary12c, size = 2)+
  facet_wrap(~Strain, scales= "free_x", ncol=20) +
  labs(title = "Growth curves at 12C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves12c


```



## Normality check of the data {.tabset}

Density plots to observe the distribution were produced but also, the Shapiro-Wilk test was performed per strain for all the data points and checked if any strain curve per treatment was normally distributed. We encountered that one strain has normally distributed data for 15C (\@ref(tab:norm15)) and 27 strains have normally distributed data for 12C (\@ref(tab:norm12)).

### Temperature 25C

For this particular temperature, one strain has normally distributed data.

```{r norm25, fig.cap="Distribution concentration 25C.", eval=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.width=4, fig.height=4, fig.align='center'}

ggplot(OD_resultsconv25C, aes(x=Concentration)) +geom_density(fill="#A5333E", alpha=0.7)+
  theme(legend.position="none") + labs(x= expression(paste("Concentration (*", 10^6, " spores/mL)")))+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))# --> evident from the distribution it is not normally distributed. 

OD_resultsconv25C <-OD_resultsconv25C %>% filter(Strain != "Oregon_1990_a12_3B_19") %>% filter(Strain != "Oregon_1990_a12_3B_4")

My_list <- split(OD_resultsconv25C, f = list(OD_resultsconv25C$Strain))


loop_Shapiro <- list()

for (name in names(My_list)){
  My_sub <- My_list[[name]]
  loop_Shapiro[[name]] <- shapiro.test(My_sub$Concentration)
  My_sub <- My_sub %>% mutate(pvalshapw=loop_Shapiro[[name]]$p.value)
     
  if (name == names(My_list)[[1]]) { 
      OD_resultsconv25Cpvalshapw = My_sub
    }
    else { 
      OD_resultsconv25Cpvalshapw = bind_rows(OD_resultsconv25Cpvalshapw, My_sub)
    }
  
}

# Check for the strains with normal distribution

OD_resultsconv25Cpvalshapwnonnorm<-OD_resultsconv25Cpvalshapw %>% ungroup() %>% group_by(Strain, Condition, ID_genome_file) %>% summarize(Pval_ShapWilk= mean(pvalshapw)) %>% filter(Pval_ShapWilk >= 0.05)

knitr::kable(OD_resultsconv25Cpvalshapwnonnorm%>%arrange(desc(Strain)))%>% kable_styling()
#One strain has a normally distributed curve


```

### Temperature 22C

For this particular temperature, no strain has a normally distributed data.

```{r norm22, fig.cap="Distribution concentration 22C.", eval=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.width=4, fig.height=4, fig.align='center'}

ggplot(OD_resultsconv22C, aes(x=Concentration)) +geom_density(fill="#F4797D", alpha=0.7)+
  theme(legend.position="none") + labs(x= expression(paste("Concentration (*", 10^6, " spores/mL)")))+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))# --> evident from the distribution it is not normally distributed. 


My_list <- split(OD_resultsconv22C, f = list(OD_resultsconv22C$Strain))


loop_Shapiro <- list()

for (name in names(My_list)){
  My_sub <- My_list[[name]]
  loop_Shapiro[[name]] <- shapiro.test(My_sub$Concentration)
  My_sub <- My_sub %>% mutate(pvalshapw=loop_Shapiro[[name]]$p.value)
     
  if (name == names(My_list)[[1]]) { 
      OD_resultsconv22Cpvalshapw = My_sub
    }
    else { 
      OD_resultsconv22Cpvalshapw = bind_rows(OD_resultsconv22Cpvalshapw, My_sub)
    }
  
}

# Check for the strains with normal distribution

OD_resultsconv22Cpvalshapwnonnorm<-OD_resultsconv22Cpvalshapw %>% ungroup() %>% group_by(Strain, Condition, ID_genome_file) %>% summarize(Pval_ShapWilk= mean(pvalshapw)) %>% filter(Pval_ShapWilk >= 0.05)

#knitr::kable(OD_resultsconv22Cpvalshapwnonnorm%>%arrange(desc(Strain)))%>% kable_styling()
#No strain has a normally distributed curve


```

### Temperature 18C

For this particular temperature, no strain has a normally distributed data.

```{r norm18, fig.cap="Distribution concentration 18C.", eval=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.width=4, fig.height=4, fig.align='center'}

ggplot(OD_resultsconv18C, aes(x=Concentration)) +geom_density(fill="#955490", alpha=0.7)+
  theme(legend.position="none") + labs(x= expression(paste("Concentration (*", 10^6, " spores/mL)")))+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))# --> evident from the distribution it is not normally distributed. 


My_list <- split(OD_resultsconv18C, f = list(OD_resultsconv18C$Strain))


loop_Shapiro <- list()

for (name in names(My_list)){
  My_sub <- My_list[[name]]
  loop_Shapiro[[name]] <- shapiro.test(My_sub$Concentration)
  My_sub <- My_sub %>% mutate(pvalshapw=loop_Shapiro[[name]]$p.value)
     
  if (name == names(My_list)[[1]]) { 
      OD_resultsconv18Cpvalshapw = My_sub
    }
    else { 
      OD_resultsconv18Cpvalshapw = bind_rows(OD_resultsconv18Cpvalshapw, My_sub)
    }
  
}

# Check for the strains with normal distribution

OD_resultsconv18Cpvalshapwnonnorm<-OD_resultsconv18Cpvalshapw %>% ungroup() %>% group_by(Strain, Condition, ID_genome_file) %>% summarize(Pval_ShapWilk= mean(pvalshapw)) %>% filter(Pval_ShapWilk >= 0.05)

#knitr::kable(OD_resultsconv22Cpvalshapwnonnorm%>%arrange(desc(Strain)))%>% kable_styling()
#No strain has a normally distributed curve


```

### Temperature 15C

For this particular temperature, one strain has normally distributed data.

```{r norm15, fig.cap="Distribution concentration 15C.", eval=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.width=4, fig.height=4, fig.align='center'}

ggplot(OD_resultsconv15C, aes(x=Concentration)) +geom_density(fill="#3596B5", alpha=0.7)+
  theme(legend.position="none") + labs(x= expression(paste("Concentration (*", 10^6, " spores/mL)")))+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))# --> evident from the distribution it is not normally distributed. 


My_list <- split(OD_resultsconv15C, f = list(OD_resultsconv15C$Strain))


loop_Shapiro <- list()

for (name in names(My_list)){
  My_sub <- My_list[[name]]
  loop_Shapiro[[name]] <- shapiro.test(My_sub$Concentration)
  My_sub <- My_sub %>% mutate(pvalshapw=loop_Shapiro[[name]]$p.value)
     
  if (name == names(My_list)[[1]]) { 
      OD_resultsconv15Cpvalshapw = My_sub
    }
    else { 
      OD_resultsconv15Cpvalshapw = bind_rows(OD_resultsconv15Cpvalshapw, My_sub)
    }
  
}

# Check for the strains with normal distribution

OD_resultsconv15Cpvalshapwnonnorm<-OD_resultsconv15Cpvalshapw %>% ungroup() %>% group_by(Strain, Condition, ID_genome_file) %>% summarize(Pval_ShapWilk= mean(pvalshapw)) %>% filter(Pval_ShapWilk >= 0.05)

knitr::kable(OD_resultsconv15Cpvalshapwnonnorm%>%arrange(desc(Strain)), caption="Strains per with normal distribution for the temperature 15C.")%>% kable_styling()



```

### Temperature 12C

For this particular temperature, 16 strains have normally distributed data.

```{r norm12, fig.cap="Distribution concentration 12C.", eval=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.width=4, fig.height=4, fig.align='center'}

ggplot(OD_resultsconv12c, aes(x=Concentration)) +geom_density(fill="#74CCED", alpha=0.7)+
  theme(legend.position="none") + labs(x= expression(paste("Concentration (*", 10^6, " spores/mL)")))+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))# --> evident from the distribution it is not normally distributed. 


My_list <- split(OD_resultsconv12c, f = list(OD_resultsconv12c$Strain))


loop_Shapiro <- list()

for (name in names(My_list)){
  My_sub <- My_list[[name]]
  loop_Shapiro[[name]] <- shapiro.test(My_sub$Concentration)
  My_sub <- My_sub %>% mutate(pvalshapw=loop_Shapiro[[name]]$p.value)
     
  if (name == names(My_list)[[1]]) { 
      OD_resultsconv12Cpvalshapw = My_sub
    }
    else { 
      OD_resultsconv12Cpvalshapw = bind_rows(OD_resultsconv12Cpvalshapw, My_sub)
    }
  
}

# Check for the strains with normal distribution

OD_resultsconv12Cpvalshapwnonnorm<-OD_resultsconv12Cpvalshapw %>% ungroup() %>% group_by(Strain, Condition, ID_genome_file) %>% summarize(Pval_ShapWilk= mean(pvalshapw)) %>% filter(Pval_ShapWilk >= 0.05)

knitr::kable(OD_resultsconv12Cpvalshapwnonnorm%>%arrange(desc(Strain)), caption="Strains per with normal distribution for the temperature 12C.")%>% kable_styling()



```

# Growth variables, optimal temperature and sensitivities calculations

For all the variables after their calculation a LOF filter was applied with 20 neighbors to remove outliers. 


## OD results to eAUC calculation

First, we will calculate the right Riemann sum and then the left Riemann sum, from which we will get the average of both as an approximation of the empirical area under the curve ($eAUC_{T_{exp_n}}$). The calculations were done with the median concentration per time point. This code chunk was mostly created by Alice Feurtey (2021).

```{r categvars, error=FALSE, message=FALSE, warning=FALSE}
OD_resultsconv$Year<-as.character(OD_resultsconv$Year)

is.car <- sapply(OD_resultsconv, is.character)
characters.df <- OD_resultsconv[, is.car] %>% dplyr::select(-Cell, -Row, -for_display, -Reason_removed, -Plate, -Timepoint, -Temperature, -Condition, -Reduced_strainname) %>% group_by_all() %>% summarize()
listcategvars<-names(characters.df)

```

```{r rightriemann, error=FALSE, message=FALSE, warning=FALSE, eval=FALSE}

#Right Riemann sum: 

median_abs = OD_resultsconv %>%
  group_by(Date, Plate, Condition, Strain, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, ID_genome_file, climate, New_Location_year) %>%
 dplyr::summarize(Concentration_median = median(Concentration)) #get the medians of the concentration


data_isolate_number_bars = median_abs


area_approximations <- data.frame()


exp_length = data_isolate_number_bars %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>%
    group_by(Condition, Strain, Plate) %>%
    mutate(Min_time = min(Time), Max_time = max(Time),
           Exp_length = Max_time - Min_time) %>%
   dplyr::select(Condition, Strain, Plate, Exp_length) %>%
   distinct() #get the experiment length of each experiment
max_exp_length = max(exp_length$Exp_length) #get the max experiment length overall 

  
median_abs = full_join(data_isolate_number_bars, exp_length) %>%
    #dplyr::filter(Strain == strain_name) %>%
    #group_by(Date, Condition, Strain, Location, Geographic_region, Year, Country, Plate, Location_year, New_Location_year) %>%
    #dplyr::summarize(Normalized_absorbance = median(Concentration)) %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>% ungroup() %>%
    group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, New_Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, climate) %>%
    mutate(Start_time = min(Time),
           End_time = max(Time),
           New_time = ifelse(Time == End_time, Start_time + max_exp_length, Time)) ## Normalize the time of all the experiments


  ## Now sorting per time and calculating the rectangle areas
area_approximations = median_abs %>%
    dplyr::arrange(Strain, Condition, Date) %>%
    ungroup() %>%
    mutate(prev = lag(New_time), time_diff = New_time - prev) %>% filter(time_diff > 0) %>%
    mutate(rectangle = time_diff * Concentration_median) %>%
    group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, climate, New_Location_year) %>%
    dplyr::summarize(approx_area = sum(rectangle)) %>% mutate(Riemanside="Right")
  


#numberstrainseAUC<-area_approximations %>% ungroup() %>% group_by(Strain) %>% summarize(numb = n()) %>% dplyr::select(-numb) # 412
#missingstrainseAUC<- anti_join(numberstrainsODresults, numberstrainseAUC) # Check to control if everything is there
```

```{r leftriemann, error=FALSE, message=FALSE, warning=FALSE, eval=FALSE}

#Left Riemann sum:

median_abs = OD_resultsconv %>%
  group_by(Date, Plate, Condition, Strain, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, ID_genome_file, climate, New_Location_year) %>%
 dplyr::summarize(Concentration_median = median(Concentration)) %>% ungroup() %>%
  group_by(Plate, Condition, Strain, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, climate, New_Location_year) #%>% filter(n()<= 12) %>% ungroup() #get the medians of the concentration

data_isolate_number_bars = median_abs


area_approximations1 <- data.frame()


exp_length = data_isolate_number_bars %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>%
    group_by(Condition, Strain, Plate) %>%
    mutate(Min_time = min(Time), Max_time = max(Time),
           Exp_length = Max_time - Min_time) %>%
   dplyr::select(Condition, Strain, Plate, Exp_length) %>%
   distinct() #get the experiment length of each experiment
max_exp_length = max(exp_length$Exp_length) #get the max experiment length overall 


  
median_abs = full_join(data_isolate_number_bars, exp_length) %>%
    #dplyr::filter(Strain == strain_name) %>%
    #group_by(Date, Condition, Strain, Location, Geographic_region, Year, Country, Plate, Location_year) %>%
    #dplyr::summarize(Normalized_absorbance = median(Concentration)) %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>% ungroup() %>%
    group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, climate, New_Location_year) %>%
    mutate(Start_time = min(Time),
           End_time = max(Time),
           New_time = ifelse(Time == End_time, Start_time + max_exp_length, Time)) ## Normalize the time of all the experiments
  
  ## Now sorting per time and calculating the rectangle areas
area_approximations1 = median_abs %>%
    dplyr::arrange(Strain, Condition, Date) %>%
    ungroup() %>%
    mutate(nextfor = lead(New_time), time_diff = nextfor - New_time) %>% filter(time_diff > 0) %>%
    mutate(rectangle = time_diff * Concentration_median) %>%
    group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, climate, New_Location_year) %>% filter(New_time < End_time)%>%
    dplyr::summarize(approx_area = sum(rectangle)) %>% mutate(Riemanside="Left")
  


numberstrainseAUC1<-area_approximations1 %>% ungroup() %>% group_by(Strain) %>% summarize(numb = n()) %>% dplyr::select(-numb) # 420
#missingstrainseAUC<- anti_join(numberstrainsODresults, numberstrainseAUC) # Check to control if everything is there


area_approximations2<- full_join(area_approximations, area_approximations1)


area_approximations3<- area_approximations2 %>% ungroup() %>% group_by(Riemanside) %>% summarize (Mean_aprox= mean(approx_area))


#ggplot(area_approximations2, aes(x=approx_area, fill=Riemanside)) + geom_density(alpha=0.5)


###Average of right and left Riemann sums :

area_approximations<- area_approximations2 %>% ungroup() %>% group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, climate, New_Location_year) %>% summarize (approx_area= mean(approx_area))

write.csv(area_approximations, "area_approximationseAUC.csv", row.names = FALSE)


```


Two strains information for 25C temperature treatment are removed from all further analysis since the eAUC is equal to 0. There was no growth at all from these two strains for this temperature (Oregon_1990_a12_3B_19, Oregon_1990_a12_3B_4). 


```{r areaapprox, error=FALSE, message=FALSE, warning=FALSE, fig.cap="LOF filter with the density of eAUC calculation per temperature, and Number of observation kept or filtered out post LOF filtering eAUC calculations per temperature, and Number of growth curves with eAUC calculation per temperature."}

area_approximations<-read.csv("area_approximationseAUC.csv", header=TRUE)
area_approximations$Year<- as.character(area_approximations$Year)



area_approximations %>% filter(approx_area == 0)
area_approximations %>% filter(approx_area < 0) #NO NEGATIVE AREA VALUES
area_approximations<- area_approximations %>% filter(approx_area != 0)


conditions = c("12C", "15C", "18C", "22C", "25C")
kneighbors = c(3, 10, 20, 30, 40)

df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-area_approximations %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$approx_area)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(approx_area, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, approx_area)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofarea_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofarea_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="eAUC calculations kept or filtered per temperature")

lofarea_approx_filtered<-df20 %>% filter(lof_filter == "Non-outlier")%>% dplyr::select(-lof, -lof_filter, -kn, -scaled_data)

widelofarea_approx_filtered<- lofarea_approx_filtered %>% dplyr::select(ID_genome_file, Adm_Cluster, New_Location_year, Condition, approx_area) %>% ungroup() %>% pivot_wider(names_from = "Condition", names_prefix = "eAUC",values_from = "approx_area")

widelofarea_approx_filteredeu<- lofarea_approx_filtered %>% filter(Adm_Cluster=="Europe") %>% dplyr::select(ID_genome_file, Condition, approx_area) %>% ungroup() %>% pivot_wider(names_from = "Condition", names_prefix = "eAUC",values_from = "approx_area")

widelofarea_approx_filteredna<- lofarea_approx_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, approx_area) %>% ungroup() %>% pivot_wider(names_from = "Condition", names_prefix = "eAUC",values_from = "approx_area")

write_csv(widelofarea_approx_filtered, "Phenotype_eAUC_lofoutrem.csv")

write_csv(widelofarea_approx_filteredeu, "Phenotype_eAUC_lofoutremeu.csv")

write_csv(widelofarea_approx_filteredna, "Phenotype_eAUC_lofoutremna.csv")


lofarea_approx_filtered %>% drop_na(approx_area) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number eAUCs obtained per temperature")+ geom_text(vjust=1.6, color="white", size=6)


```




## Growth rate, Asymptote and Inflection calculations with logistic model

First, we will calculate the logistic growth using a nonlinear (weighted) least-squares estimates of the parameters of a nonlinear model with a self starting model that evaluates the logistic function and its gradient. It has an initial attribute that creates initial estimates of the parameters Asym (carrying capacity), xmid (inflection), and scal (scale). The calculations were done with the median concentration per time point. This code chunk was mostly created by Alice Feurtey (2022).  This is more accentuated in the 12C treatment (see Figure \@ref(fig:numofobloggrowth)) and it makes sense, since as seen in the growth curves below (see Figure \@ref(fig:12growthcurveslogis)), there is very little or no growth for the duration of the experiments.

```{r logistic, message=FALSE, fig.cap="Number of growth curves per temperature that fit the logistic model.", warning=FALSE, error=FALSE, eval=FALSE}


exp_length1 = OD_resultsconv%>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>%
    group_by(Condition, Strain, Plate) %>%
    mutate(Min_time = min(Time), Max_time = max(Time),
           Exp_length = Max_time - Min_time) %>%
   dplyr::select(Condition, Strain, Plate, Exp_length) %>%
   distinct() #get the experiment length of each experiment
max_exp_length1 = max(exp_length1$Exp_length)



Concconvtime = full_join(OD_resultsconv, exp_length1) %>%
    #dplyr::filter(Strain == strain_name) %>%
    group_by(Date, Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, climate, New_Location_year) %>%
    dplyr::summarize(Concentration = median(Concentration)) %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>% ungroup() %>%
    group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, climate, New_Location_year) %>%
    mutate(Start_time = min(Time),
           End_time = max(Time),
           New_time = ifelse(Time == End_time, Start_time + max_exp_length1, Time), NEW_TIMES=New_time-min(Time)) %>% ungroup() ## Normalize the time of all the experiments 

write_csv(Concconvtime, "Concconvtime.csv")


all_strains = unique(Concconvtime$Strain)
all_conditions = unique(Concconvtime$Condition)


# On all strains and all temperatures
# estimate the parameters of a logistic curve
# as well as the corresponding AUC
for (strain_name in all_strains )  {                           
for (tempe in all_conditions ) {
  
  #Fit the model
  temp6 = ungroup(Concconvtime) %>%
    filter(Strain == strain_name) %>%
    filter(Condition == tempe)


  
  lol = unique(temp6$Location)
    cont = unique(temp6$Geographic_region)
    coun = unique(temp6$Country)
    lol_year= unique(temp6$New_Location_year)
    year=unique(temp6$Year)
    
    try({
    #Fitting the non-linear regression to the data 
    #(with a self-starting function)
    m2 = nls( Concentration ~ SSlogis(NEW_TIMES, Asymptote, Inflection, Scale), data= temp6)

    #Get the point values as predicted by the model
    temp_tibble2 = tibble(NEW_TIMES = temp6$NEW_TIMES, y = predict(m2), 
                         Condition = tempe, Strain = strain_name, Location = lol, Geographic_region = cont, Country = coun, New_Location_year=lol_year, Year=year, Model="logistic")
    

    # Getting the area under the curve for the shortest experiment length
    # based on the estimated parameters
    start_time = min(temp6$NEW_TIMES)
    
    gamma = as.numeric(coef(m2)["Scale"])
    beta = exp(as.numeric(coef(m2)["Inflection"])/gamma)
    numerator = exp((start_time + max_exp_length1)/gamma) + beta
    denominator = exp(start_time/gamma) + beta
    
    model_area = as.numeric(coef(m2)["Asymptote"])*gamma*log(numerator/denominator)
    
    
    
    #Get all the model-based parameters in a tibble
    temp_sum_tibble2 = tibble(asymptote = coef(m2)["Asymptote"], 
                             inflection = coef(m2)["Inflection"],
                             yinflection= (coef(m2)["Asymptote"])/(1+exp(0/as.numeric(coef(m2)["Scale"]))),
                             growth_rate = 1/gamma,
                             Condition = tempe, 
                             Strain = strain_name,
                             Model_based_AUC = model_area,
                             Geographic_region = cont,
                             Location = lol,
                             New_Location_year=lol_year,
                             Year=year,
                             Country = coun,
                             Model="logistic")
   #print(temp_sum_tibble)
   
   
    #Concatenate the tables
    if (tempe == all_conditions[[1]] && strain_name == all_strains[[1]]) { 
      spline_curves2 = temp_tibble2 
      spline_summary2 = temp_sum_tibble2
    }
    else { 
      spline_curves2 = bind_rows(spline_curves2, temp_tibble2)
      spline_summary2 = bind_rows(spline_summary2, temp_sum_tibble2)
    }
  })
  
}
}



spline_summary2notwellfit<- spline_summary2 %>% filter((growth_rate < 0))
spline_summary2<- spline_summary2 %>% filter(!(growth_rate < 0)) %>% left_join(characters.df) %>% dplyr::select(-Model)

write.csv(spline_summary2, "spline_summary2.csv", row.names = FALSE)
write.csv(spline_curves2 %>% left_join(characters.df), "spline_curves2.csv", row.names = FALSE)


```



```{r numofobloggrowth, message=FALSE, fig.cap="LOF filter with the density of growth rate calculation per temperature, Number of observations kept or filtered out post LOF filtering growth rate calculations per temperature and Number of growth curves per temperature that fit the logistic growth model.", warning=FALSE, error=FALSE}

spline_summary2<-read.csv("spline_summary2.csv", header=TRUE)
spline_curves2<-read.csv("spline_curves2.csv", header=TRUE)
spline_summary2$Year<- as.character(spline_summary2$Year)
spline_curves2$Year<- as.character(spline_curves2$Year)


#spline_summary2 %>% filter(growth_rate == 0) #NO 0 GROWTH RATE VALUES
#spline_summary2  %>% filter(growth_rate < 0) #NO NEGATIVE GROWTH RATE VALUES


conditions = c("12C", "15C", "18C", "22C", "25C")
kneighbors = c(3, 10, 20, 30, 40)

df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_summary2 %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$growth_rate)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(growth_rate, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, growth_rate)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofgrowthrate_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofgrowthrate_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Growth rate calculations kept or filtered per temperature")

lofgrowth_rate_filtered<-df20 %>% filter(lof_filter == "Non-outlier")


df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_summary2 %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$asymptote)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(asymptote, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, asymptote)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofasymptote_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofasymptote_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Asymptote calculations kept or filtered per temperature")

lofasymptote_filtered<-df20 %>% filter(lof_filter == "Non-outlier")



df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_summary2 %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$inflection)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(inflection, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, inflection)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofinflection_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofinflection_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Inflection time calculations kept or filtered per temperature")

lofinflection_filtered<-df20 %>% filter(lof_filter == "Non-outlier")


loflogisticvars<-(ungroup(lofinflection_filtered)%>% dplyr::select(-asymptote, -yinflection, -growth_rate, -lof, -lof_filter, -kn, -scaled_data, -Model_based_AUC)) %>% full_join( (ungroup(lofasymptote_filtered)%>% dplyr::select(-inflection, -yinflection, -growth_rate, -lof, -lof_filter, -kn, -scaled_data, -Model_based_AUC))) %>% full_join((ungroup(lofgrowth_rate_filtered)%>% dplyr::select(-inflection, -yinflection, -asymptote, -lof, -lof_filter, -kn, -scaled_data, -Model_based_AUC)))

wideloflogisticvars_filtered<- (lofinflection_filtered %>% dplyr::select(ID_genome_file, Condition, inflection) %>% pivot_wider(names_from = "Condition", names_prefix = "inflection_",values_from = "inflection")) %>% full_join((lofasymptote_filtered %>% dplyr::select(ID_genome_file, Condition, asymptote) %>% pivot_wider(names_from = "Condition", names_prefix = "asymptote_",values_from = "asymptote"))) %>% full_join((lofgrowth_rate_filtered %>% dplyr::select(ID_genome_file, Condition, growth_rate) %>% pivot_wider(names_from = "Condition", names_prefix = "growthrate_",values_from = "growth_rate")))

wideloflogisticvars_filteredeu<- (lofinflection_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, inflection) %>% pivot_wider(names_from = "Condition", names_prefix = "inflection_",values_from = "inflection")) %>% full_join((lofasymptote_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, asymptote) %>% pivot_wider(names_from = "Condition", names_prefix = "asymptote_",values_from = "asymptote"))) %>% full_join((lofgrowth_rate_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, growth_rate) %>% pivot_wider(names_from = "Condition", names_prefix = "growthrate_",values_from = "growth_rate")))

wideloflogisticvars_filteredna<- (lofinflection_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, inflection) %>% pivot_wider(names_from = "Condition", names_prefix = "inflection_",values_from = "inflection")) %>% full_join((lofasymptote_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, asymptote) %>% pivot_wider(names_from = "Condition", names_prefix = "asymptote_",values_from = "asymptote"))) %>% full_join((lofgrowth_rate_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, growth_rate) %>% pivot_wider(names_from = "Condition", names_prefix = "growthrate_",values_from = "growth_rate")))


write_csv(wideloflogisticvars_filtered, "Phenotype_logisticregvars_lofoutrem.csv")

write_csv(wideloflogisticvars_filteredeu, "Phenotype_logisticregvars_lofoutremeu.csv")

write_csv(wideloflogisticvars_filteredna, "Phenotype_logisticregvars_lofoutremna.csv")


loflogisticvars %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of growth curves per temperature", subtitle= "that fit the logistic model")+ geom_text(vjust=1.6, color="white", size=6)




```

I compared the Riemann sum-based method versus the logistic model-based to see if they are similar or not (see Figure \@ref(fig:compareeaucmeasurements)). Actually, the logistic model-based approach has a higher density of higher eAUC values which is most likely due to the fact that the lower temperature growth curves failed to fit the model in a higher rate. However, still their distribution seems to be quite similar.

```{r compareeaucmeasurements, message=FALSE, warning=FALSE, error=FALSE, fig.cap="Comparison of eAUC calculations using Riemann sums and using the model-based integral."}

compeauc<-full_join(spline_summary2, area_approximations)

compeauc %>% ggplot() +
    geom_density(aes(x=Model_based_AUC), fill= "olivedrab4", alpha=0.5)+geom_density(aes(x=approx_area), fill= "purple1", alpha=0.5)+theme_minimal()+ theme(axis.line = element_line(colour = "grey50"))+labs(x="eAUC", y="Density", title="Comparison between logistic model-based eAUC and Riemann sums eAUC", subtitle = "Model-based eAUC in green and Riemann sums eAUC in purple")


#They seem quite similar

```



### Growth curves with logistic fit per strain per treatment {.tabset}


#### Temperature 25C



```{r 25growthcurveslogis, fig.cap="Growth curves at 25C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

Concconvtime<-read_csv("Concconvtime.csv")

OD_resultsconv25C<-Concconvtime %>% filter(Condition == "25C") 

OD_resultsconvsummary25C<-spline_curves2  %>% filter(Condition == "25C")

growthcurves25C<- ggplot(OD_resultsconv25C, aes(x = NEW_TIMES, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1, x=NEW_TIMES, y=y), data = OD_resultsconvsummary25C)  +
  geom_point(data = OD_resultsconvsummary25C, size = 2, color="#453823", aes( x=NEW_TIMES, y=y))+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 25C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves25C


```


#### Temperature 22C

```{r 22growthcurveslogis, fig.cap="Growth curves at 22C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv22C<-Concconvtime %>% filter(Condition == "22C") 

OD_resultsconvsummary22C<-spline_curves2 %>% filter(Condition == "22C")

growthcurves22C<- ggplot(OD_resultsconv22C, aes(x = NEW_TIMES, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1, y=y, x=NEW_TIMES), data = OD_resultsconvsummary22C)  +
  geom_point(data = OD_resultsconvsummary22C, size = 2, color="#453823", aes( y=y, x=NEW_TIMES))+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 22C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves22C


```



#### Temperature 18C

```{r 18growthcurveslogis, fig.cap="Growth curves at 18C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv18C<-Concconvtime %>% filter(Condition == "18C") 

OD_resultsconvsummary18C<-spline_curves2 %>% filter(Condition == "18C")

growthcurves18C<- ggplot(OD_resultsconv18C, aes(x = NEW_TIMES, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1, y=y, x=NEW_TIMES), data = OD_resultsconvsummary18C)  +
  geom_point(data = OD_resultsconvsummary18C, size = 2, color="#453823", aes(y=y, x=NEW_TIMES))+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 18C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves18C


```



#### Temperature 15C

```{r 15growthcurveslogis, fig.cap="Growth curves at 15C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv15C<-Concconvtime %>% filter(Condition == "15C") 

OD_resultsconvsummary15C<-spline_curves2 %>% filter(Condition == "15C")

growthcurves15C<- ggplot(OD_resultsconv15C, aes(x = NEW_TIMES, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1, x=NEW_TIMES, y=y), data = OD_resultsconvsummary15C)  +
  geom_point(data = OD_resultsconvsummary15C, size = 2, color="#453823", aes(x=NEW_TIMES, y=y))+
  facet_wrap(~Strain, scales= "free", ncol=20) +
  labs(title = "Growth curves at 15C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves15C


```


#### Temperature 12C

```{r 12growthcurveslogis, fig.cap="Growth curves at 12C.", eval=TRUE, fig.width=25, fig.height=25, message=FALSE, warning=FALSE, error=FALSE}

OD_resultsconv12c<-Concconvtime  %>% filter(Condition == "12C") 

OD_resultsconvsummary12c<-spline_curves2  %>% filter(Condition == "12C")

growthcurves12c<- ggplot(OD_resultsconv12c, aes(x = NEW_TIMES, y=Concentration, col=Strain)) +
  geom_point(alpha=0.3) + 
  geom_line(aes(group = 1, x=NEW_TIMES, y=y), data = OD_resultsconvsummary12c)  +
  geom_point(data = OD_resultsconvsummary12c, size = 2, color="#453823", aes(x=NEW_TIMES, y=y))+
  facet_wrap(~Strain, scales= "free_x", ncol=20) +
  labs(title = "Growth curves at 12C")+theme_minimal()+
  theme(legend.position="none", axis.line = element_line(colour = "grey50"))

growthcurves12c


```




## Optimal temperature quadratic regression of eAUC of the five temperature treatments {.tabset}


```{r optimaltempcalc, message=FALSE, warning=FALSE, error=FALSE, eval=FALSE}


area_approxforopt<- area_approximations %>% tidyr::separate_wider_position(Condition, c(Temperature=2, Celsius=1)) 


area_approxforopt$Temperature <- as.numeric(area_approxforopt$Temperature)
area_approxforopt$Temperature2 <- area_approxforopt$Temperature^2



all_strains = unique(area_approxforopt$Strain)


for (strain_name in all_strains )  {                           
  
  temp6 = ungroup(area_approxforopt) %>%
    filter(Strain == strain_name) 
  
  lol = unique(temp6$Location)
    cont = unique(temp6$Geographic_region)
    coun = unique(temp6$Country)
        lol_year= unique(temp6$New_Location_year)
    year=unique(temp6$Year)
  tempvalues=seq(12, 25, 0.1)
    try({
    #Fitting the quadratic regression
    mq = lm( approx_area ~ Temperature + Temperature2, data=temp6)

    #Get the point values as predicted by the model
    temp_tibbleq = tibble(Temperature = tempvalues, y = predict(mq, list(Temperature=tempvalues, Temperature2=tempvalues^2)), Strain = strain_name, Location = lol, Geographic_region = cont, Country = coun, New_Location_year=lol_year, Year=year, Model="quadratic", rsquaredadjmq = summary(mq)$adj.r.squared)

    #Concatenate the tables
    if (strain_name == all_strains[[1]]) { 
      spline_curvesq = temp_tibbleq
    }
    else { 
      spline_curvesq = bind_rows(spline_curvesq, temp_tibbleq)
    }
  })
  
}


spline_curvesq1<-spline_curvesq %>% filter(rsquaredadjmq > 0.0) %>% group_by(Strain) %>% filter(y == max(y))%>% left_join(characters.df) %>% dplyr::select(-Model) %>% filter (!(Temperature == 12 | Temperature == 25))

spline_curvesqbad<-spline_curvesq %>% filter(rsquaredadjmq < 0.6) %>% group_by(Strain) %>% filter(y == max(y))%>% left_join(characters.df) %>% dplyr::select(-Model) %>% filter (!(Temperature == 12 | Temperature == 25))


spline_curvesqall<-spline_curvesq %>% group_by(Strain) %>% filter(y == max(y))%>% left_join(characters.df) %>% dplyr::select(-Model)  %>% filter (!(Temperature == 12 | Temperature == 25))

spline_curvesqall %>% ggplot(aes(x=rsquaredadjmq))+geom_density()


write.csv(spline_curvesq1, "spline_curvesq1no12no25.csv", row.names = FALSE)

area_approxforopt<- lofarea_approx_filtered %>% tidyr::separate_wider_position(Condition, c(Temperature=2, Celsius=1)) 


area_approxforopt$Temperature <- as.numeric(area_approxforopt$Temperature)
area_approxforopt$Temperature2 <- area_approxforopt$Temperature^2



all_strains = unique(area_approxforopt$Strain)


for (strain_name in all_strains )  {                           
  
  temp6 = ungroup(area_approxforopt) %>%
    filter(Strain == strain_name) 
  
  lol = unique(temp6$Location)
    cont = unique(temp6$Geographic_region)
    coun = unique(temp6$Country)
        lol_year= unique(temp6$New_Location_year)
    year=unique(temp6$Year)
  tempvalues=seq(12, 25, 0.1)
    try({
    #Fitting the quadratic regression
    mq = lm( approx_area ~ Temperature + Temperature2, data=temp6)

    #Get the point values as predicted by the model
    temp_tibbleq = tibble(Temperature = tempvalues, y = predict(mq, list(Temperature=tempvalues, Temperature2=tempvalues^2)), Strain = strain_name, Location = lol, Geographic_region = cont, Country = coun, New_Location_year=lol_year, Year=year, Model="quadratic", rsquaredadjmq = summary(mq)$adj.r.squared)

    #Concatenate the tables
    if (strain_name == all_strains[[1]]) { 
      spline_curvesq = temp_tibbleq
    }
    else { 
      spline_curvesq = bind_rows(spline_curvesq, temp_tibbleq)
    }
  })
  
}


spline_curvesq1eauclof<-spline_curvesq %>% filter(rsquaredadjmq > 0.0) %>% group_by(Strain) %>% filter(y == max(y))%>% left_join(characters.df) %>% dplyr::select(-Model) %>% filter (!(Temperature == 12 | Temperature == 25))


write.csv(spline_curvesq1eauclof, "spline_curvesq1no12no25eauclof.csv", row.names = FALSE)


```







```{r opttempres, message=FALSE, warning=FALSE, error=FALSE, fig.cap="LOF filter with the density of number of neighbors Optimal temperature calculation, and Number of observations kept or filtered out post LOF filtering of the Optimal temperature calculation per genetic cluster."}

spline_curvesq1<-read_csv("spline_curvesq1no12no25.csv")
spline_curvesq1$Year<-as.character(spline_curvesq1$Year)

spline_curvesq1eauclof<-read_csv("spline_curvesq1no12no25eauclof.csv")
spline_curvesq1eauclof$Year<-as.character(spline_curvesq1eauclof$Year)

#anti_join(spline_curvesq1, spline_curvesq1eauclof)

df = tibble()
  for (kn in kneighbors) {
    temp<-spline_curvesq1eauclof
    temp$scaled_data <- scale(temp$Temperature)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}


df %>%
  ggplot(aes(Temperature, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(cols = vars(kn))

df$kn<-as.character(df$kn)

df %>%
  ggplot(aes(kn,Temperature)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4)  +
  labs(subitle = "The axis is the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofopttemperature_count_filtering =  df20 %>% 
  count(Strain, Adm_Cluster, lof_filter)

temp = lofopttemperature_count_filtering  %>%
  count(Adm_Cluster, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Adm_Cluster, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Optimal temperature calculations kept or filtered per genetic cluster")

lofopttemperature_filtered<-df20 %>% filter(lof_filter == "Non-outlier")%>% dplyr::select(-lof, -lof_filter, -kn, -scaled_data,  -Isolate_full_name_phenotyping, -rsquaredadjmq)

widelofopttempx_filtered<- lofopttemperature_filtered %>% dplyr::select(ID_genome_file, Temperature) %>% rename(OptTemperature=Temperature) 

widelofopttempx_filteredeu<- lofopttemperature_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Temperature) %>% rename(OptTemperature=Temperature) 

widelofopttempx_filteredna<- lofopttemperature_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Temperature) %>% rename(OptTemperature=Temperature) 

write_csv(widelofopttempx_filtered, "Phenotype_Opttemp_lofoutrem.csv")

write_csv(widelofopttempx_filteredeu, "Phenotype_Opttemp_lofoutremeu.csv")

write_csv(widelofopttempx_filteredna, "Phenotype_Opttemp_lofoutremna.csv")

```

For obtaining the optimal temperature ($T_{opt}$), I conducted a quadratic regression of the eAUC values of the temperature range tested (12-25C, not including the extremes) for strains that had observations for the whole range. The estimated optimal temperature was calculated for `r length(unique(lofopttemperature_filtered$Strain))` eAUC curves in total (`r length(unique(lofopttemperature_filtered$Strain))` strains). 


### Number of eAUC curves from the five temperature range that fit a Quadratic Model per Genetic Cluster

There is one curve per each strain.

```{r opttempnumb, message=FALSE, fig.cap="Number of eAUC curves (one per strain) that fit the quadratic growth model.", warning=FALSE, error=FALSE}

spline_curvesq1<-read.csv("spline_curvesq1.csv", header=TRUE)
spline_curvesq1$Year<-as.character(spline_curvesq1$Year)
spline_curvesq1 %>% count(Adm_Cluster) %>% ggplot(aes(x=Adm_Cluster, y=n, fill=Adm_Cluster, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = K_colors)+labs(x="Genetic Cluster", y="Number of observations", title="Number of eAUC curves per Genetic Cluster", subtitle= "that fit the quadratic model")+ geom_text(vjust=1.6, color="white", size=6)

```

### Estimated Optimal Temperature calculated per Strain grouped by Geographic Region


```{r opttempgeoreg, message=FALSE, fig.cap="Estimated Optimal Temperature for Strains from different Genetic Clusters.", warning=FALSE, error=FALSE, fig.height=11, fig.width=10}

spline_curvesq1 %>% ggplot(aes(x=Temperature, y=Strain, colour=Temperature))+geom_point(size=4, alpha=0.5)+theme_minimal()+theme(legend.position="none", axis.line = element_line(colour = "grey50"), axis.text.y=element_blank(), panel.background = element_rect(fill = NA),
  panel.grid.major = element_line(colour = NA),axis.ticks = element_line(linewidth = 1))+ scale_colour_gradientn(colours = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Strain", title="Estimated Optimal Temperature for Strains from different Genetic Clusters") + facet_grid(rows=vars(Adm_Cluster)) +
scale_x_continuous(breaks = scales::pretty_breaks(n = 15))

```

## Temperature Sensitivity using the Estimated Optimal Temperature 

It is the "y" column from the quadratic regression of the eAUC for all the temperatures evaluated. It represents the theoretical eAUC value at which the isolate would reach at their optimal temperature condition. Then, the $n$ temperature sensitivity is the fraction between the experimental (obtained through experiment) eAUC at the $n$ temperature ($eAUC_{T_{exp_n}}$) and the estimated eAUC at the optimal temperature ($eAUC_{T_{opt}}$). Please see Equation \@ref(eq:tempsens).

\begin{equation} 
  T_{sens} = \frac{eAUC_{T_{exp_n}}}{eAUC_{T_{opt}}}
  (\#eq:tempsens)
\end{equation}  

```{r tempsenscal, message=FALSE, warning=FALSE, error=FALSE, eval=FALSE}

exp_length1 = OD_resultsconv%>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>%
    group_by(Condition, Strain, Plate) %>%
    mutate(Min_time = min(Time), Max_time = max(Time),
           Exp_length = Max_time - Min_time) %>%
   dplyr::select(Condition, Strain, Plate, Exp_length) %>%
   distinct() #get the experiment length of each experiment
max_exp_length1 = max(exp_length1$Exp_length)



Concconvtime = full_join(OD_resultsconv, exp_length1) %>%
    #dplyr::filter(Strain == strain_name) %>%
    group_by(Date, Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety, Plate, New_Location_year) %>%
    dplyr::summarize(Concentration = median(Concentration)) %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>% ungroup() %>%
    group_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, Location_year, Cluster, Continent, Adm_Cluster, Variety,New_Location_year) %>%
    mutate(Start_time = min(Time),
           End_time = max(Time),
           New_time = ifelse(Time == End_time, Start_time + max_exp_length1, Time), NEW_TIMES=New_time-min(Time)) %>% ungroup() ## Normalize the time of all the experiments 


all_strains = unique(Concconvtime$Strain)
all_conditions = unique(Concconvtime$Condition)



for (strain_name in all_strains )  {                           
for (tempe in all_conditions) {
  
  #Fit the model
  temptemp = ungroup(lofarea_approx_filtered) %>%
    filter(Strain == strain_name) %>%
    filter(Condition == tempe)
  
  tempopt=ungroup(lofopttemperature_filtered) %>%
    filter(Strain == strain_name)
  
  lol = unique(temptemp$Location)
    cont = unique(temptemp$Continent)
    coun = unique(temptemp$Country)
    lol_year= unique(temptemp$New_Location_year)
    year=unique(temptemp$Year)

  temp1=full_join(temptemp, tempopt)
  
  temp_tibble1 = tibble(Tempoptsens=temp1$approx_area/temp1$y, Strain = strain_name, Location = lol, Continent = cont, Country = coun, New_Location_year=lol_year, Year=year, Condition=tempe)
  
  if (strain_name == all_strains[[1]]) { 
      spline_curvessensopttemp = temp_tibble1
    }
    else { 
      spline_curvessensopttemp = bind_rows(spline_curvessensopttemp, temp_tibble1)
    }

}
}

spline_curvessensopttemp<-spline_curvessensopttemp%>% drop_na(Tempoptsens) %>% left_join(characters.df) 

write.csv(spline_curvessensopttemp, "spline_curvessensopttempeaucloffil.csv", row.names = FALSE)


```






```{r tempsens, message=FALSE, fig.cap="LOF filter with the density of temperature sensitivity calculation per temperature, Number of observations kept or filtered out post LOF filtering temperature sensitivity calculations per temperature and Number of calculations of temperature sensitivity per temperature.", warning=FALSE, error=FALSE}

spline_curvessensopttemp<-read.csv("spline_curvessensopttempeaucloffil.csv", header=TRUE)

spline_curvessensopttemp$Year<- as.character(spline_curvessensopttemp$Year)


#spline_curvessensopttemp %>% filter(Tempoptsens < 0) 

conditions = c("12C", "15C", "18C", "22C", "25C")
kneighbors = c(3, 10, 20, 30, 40)

df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvessensopttemp %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$Tempoptsens)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(Tempoptsens, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(cols = vars(kn))

df$kn<-as.character(df$kn)

df %>%
  ggplot(aes(kn,Tempoptsens)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4)  +
  labs(subitle = "The axis is the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofopttempsens_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofopttempsens_count_filtering %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Temperature sensitivity calculations kept or filtered per temperature")

lofopttempsens_filtered<-df20 %>% filter(lof_filter == "Non-outlier")%>% dplyr::select(-lof, -lof_filter, -kn, -scaled_data)

widelofopttempsens_filtered<- lofopttempsens_filtered %>% dplyr::select(ID_genome_file, Condition, Tempoptsens) %>% ungroup() %>% pivot_wider(names_from = "Condition", names_prefix = "Tempsens",values_from = "Tempoptsens")

widelofopttempsens_filteredeu<- lofopttempsens_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, Tempoptsens) %>% ungroup() %>% pivot_wider(names_from = "Condition", names_prefix = "Tempsens",values_from = "Tempoptsens")

widelofopttempsens_filteredna<- lofopttempsens_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, Tempoptsens) %>% ungroup() %>% pivot_wider(names_from = "Condition", names_prefix = "Tempsens",values_from = "Tempoptsens")

write_csv(widelofopttempsens_filtered, "Phenotype_tempsens_lofoutrem.csv")

write_csv(widelofopttempsens_filteredeu, "Phenotype_tempsens_lofoutremeu.csv")

write_csv(widelofopttempsens_filteredna, "Phenotype_tempsens_lofoutremna.csv")

lofopttempsens_filtered %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of eAUC-based Temperature Sensitivities", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)


```



## Time for each growth curve to arrive to the 1st, 2nd and 3rd quantiles of the concentration {.tabset}

To explore other variables involved that describe our curves, we used each individual curve to obtain the time it takes to arrive to the 1st quantile (25%), time to arrive to the 2nd quantile (50%) and time to arrive to the 3rd quantile (75%). Each curve was analyzed by using the **loess** function which generates a locally weighted polynomial regression. However, there are discrepancies with the number of observations between the eAUC calculations (Figure \@ref(fig:areaapprox)), which is non model dependent, and the time to quantiles calculations (Figures \@ref(fig:q25num), \@ref(fig:q50num), \@ref(fig:q75num)).  

```{r quantcal, message=FALSE, warning=FALSE, error=FALSE, eval=FALSE}


#Fit a loess to then get the the curve and then with that we can find the exact concentration each time and check the time that they reach that concentration value


exp_length1 = OD_resultsconv%>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>%
    group_by(Condition, Strain, Plate) %>%
    mutate(Min_time = min(Time), Max_time = max(Time),
           Exp_length = Max_time - Min_time) %>%
   dplyr::select(Condition, Strain, Plate, Exp_length) %>%
   distinct() #get the experiment length of each experiment
max_exp_length1 = max(exp_length1$Exp_length)



Concconvtime = full_join(OD_resultsconv, exp_length1) %>%
    #dplyr::filter(Strain == strain_name) %>%
    group_by(Date, Condition, Strain, Location, Continent, Year, Country, Plate, Location_year, New_Location_year) %>%
    dplyr::summarize(Concentration = median(Concentration)) %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>% ungroup() %>%
    group_by(Condition, Strain, Location, Continent, Year, Country, Location_year, New_Location_year) %>%
    mutate(Start_time = min(Time),
           End_time = max(Time),
           New_time = ifelse(Time == End_time, Start_time + max_exp_length1, Time), NEW_TIMES=New_time-min(Time)) %>% ungroup() ## Normalize the time of all the experiments 


all_strains = unique(Concconvtime$Strain)
all_conditions = unique(Concconvtime$Condition)



for (strain_name in all_strains )  {                           
for (tempe in all_conditions ) {
  
  temp6 = ungroup(Concconvtime) %>%
    filter(Strain == strain_name) %>%
    filter(Condition == tempe)


  
  lol = unique(temp6$Location)
    cont = unique(temp6$Continent)
    coun = unique(temp6$Country)
    lol_year= unique(temp6$New_Location_year)
    year=unique(temp6$Year)
    try({

    mspline = loess(Concentration~NEW_TIMES, temp6)
    
    
    temp_tibblespli = tibble(q25total = approx(x=mspline$fitted, y=mspline$x, xout =((quantile(temp6$Concentration)[2])))$y, q50total =approx(x=mspline$fitted, y=mspline$x, xout =((quantile(temp6$Concentration)[3])))$y , q75total = approx(x=mspline$fitted, y=mspline$x, xout =((quantile(temp6$Concentration)[4])))$y, q100total= approx(x=mspline$fitted, y=mspline$x, xout =((quantile(temp6$Concentration)[5])))$y,
                         Condition = tempe, Strain = strain_name, Location = lol, New_Location_year=lol_year, Year=year, Continent = cont, Country = coun, Model="loess")

    if (tempe == all_conditions[[1]] && strain_name == all_strains[[1]]) { 
      spline_curvesspli = temp_tibblespli 
    }
    else { 
      spline_curvesspli = bind_rows(spline_curvesspli, temp_tibblespli)

    }
    })
}
}    
    

spline_curvesspli<-spline_curvesspli %>% dplyr::select(-q100total, -Model)  %>% left_join(characters.df) 

write.csv(spline_curvesspli, "spline_curvesspli.csv", row.names = FALSE)


```



```{r quantiles, message=FALSE, warning=FALSE, error=FALSE, fig.cap="LOF filter with the density of time to 1st, 2nd and 3rd quantile of the concentration per curve calculation per temperature, and Number of observations kept or filtered out post LOF filtering of the time to 1st, 2nd and 3rd quantile of the concentration per curve calculations per temperature."}

spline_curvesspli<-read.csv("spline_curvesspli.csv", header=TRUE)

spline_curvesspli$Year<- as.character(spline_curvesspli$Year)


conditions = c("12C", "15C", "18C", "22C", "25C")
kneighbors = c(3, 10, 20, 30, 40)

df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvesspli %>% drop_na(q25total) %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$q25total)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(q25total, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, q25total)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofq25total_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofq25total_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Time to 1st quantile of the concentration calculations kept or filtered per temperature")

lofq25total_filtered<-df20 %>% filter(lof_filter == "Non-outlier")


df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvesspli %>% drop_na(q50total) %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$q50total)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(q50total, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, q50total)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofq50total_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofq50total_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Time to second quantile concentration calculations kept or filtered per temperature")

lofq50total_filtered<-df20 %>% filter(lof_filter == "Non-outlier")



df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvesspli %>% drop_na(q75total) %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$q75total)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(q75total, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, q75total)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofq75total_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofq75total_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Time to third quantile concentration calculations kept or filtered per temperature")

lofq75total_filtered<-df20 %>% filter(lof_filter == "Non-outlier")


lofquantvars<-(ungroup(lofq25total_filtered)%>% dplyr::select(-q50total, -q75total, -lof, -lof_filter, -kn, -scaled_data)) %>% full_join( (ungroup(lofq50total_filtered)%>% dplyr::select(-q25total, -q75total, -lof, -lof_filter, -kn, -scaled_data))) %>% full_join((ungroup(lofq75total_filtered)%>% dplyr::select(-q25total, -q50total, -lof, -lof_filter, -kn, -scaled_data)))


widelofquantvars_filtered<- (lofq25total_filtered %>% dplyr::select(ID_genome_file, Condition, q25total) %>% pivot_wider(names_from = "Condition", names_prefix = "q25total_",values_from = "q25total")) %>% full_join((lofq50total_filtered %>% dplyr::select(ID_genome_file, Condition, q50total) %>% pivot_wider(names_from = "Condition", names_prefix = "q50total_",values_from = "q50total"))) %>% full_join((lofq75total_filtered %>% dplyr::select(ID_genome_file, Condition, q75total) %>% pivot_wider(names_from = "Condition", names_prefix = "q75total_",values_from = "q75total")))

widelofquantvars_filteredeu<- (lofq25total_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, q25total) %>% pivot_wider(names_from = "Condition", names_prefix = "q25total_",values_from = "q25total")) %>% full_join((lofq50total_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, q50total) %>% pivot_wider(names_from = "Condition", names_prefix = "q50total_",values_from = "q50total"))) %>% full_join((lofq75total_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, q75total) %>% pivot_wider(names_from = "Condition", names_prefix = "q75total_",values_from = "q75total")))

widelofquantvars_filteredna<- (lofq25total_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, q25total) %>% pivot_wider(names_from = "Condition", names_prefix = "q25total_",values_from = "q25total")) %>% full_join((lofq50total_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, q50total) %>% pivot_wider(names_from = "Condition", names_prefix = "q50total_",values_from = "q50total"))) %>% full_join((lofq75total_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, q75total) %>% pivot_wider(names_from = "Condition", names_prefix = "q75total_",values_from = "q75total")))

write_csv(widelofquantvars_filtered, "Phenotype_quantilevars_lofoutrem.csv")

write_csv(widelofquantvars_filteredeu, "Phenotype_quantilevars_lofoutremeu.csv")

write_csv(widelofquantvars_filteredna, "Phenotype_quantilevars_lofoutremna.csv")


```


### Number of observations of time to 1st quantile per temperature

```{r q25num, message=FALSE, fig.cap="Number of times to 1st quantile per temperature treatment.", warning=FALSE, error=FALSE}

lofquantvars %>% dplyr::select(-q50total, -q75total) %>% drop_na(q25total) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of time calculations to the 1st quantile of the concentration", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)


```

### Number of observations of time to 2nd quantile per temperature

```{r q50num, message=FALSE, fig.cap="Number of times to 2nd quantile per temperature treatment.", warning=FALSE, error=FALSE}

lofquantvars %>% dplyr::select(-q25total, -q75total) %>% drop_na(q50total) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of time calculations to the 2nd quantile of the concentration", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)


```

### Number of observations of time to 3rd quantile per temperature

```{r q75num, message=FALSE, fig.cap="Number of times to 3rd quantile per temperature treatment.", warning=FALSE, error=FALSE}

lofquantvars %>% dplyr::select(-q25total, -q50total) %>% drop_na(q75total) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of time calculations to the 3rd quantile of the concentration", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)


```



## Concentration every 48 hours during the experiment {.tabset}

We used again each individual curve to obtain now the concentration of spores at day 2 (48 hours after first measurement), day 4 (96 hours after first measurement) and day 6 (last day of experiment, but not last measurement, just 144 hours after first measurement). Each curve was analyzed by using the **loess** function which generates a locally weighted polynomial regression.


```{r daycal, message=FALSE, warning=FALSE, error=FALSE, eval=FALSE}

exp_length1 = OD_resultsconv%>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>%
    group_by(Condition, Strain, Plate) %>%
    mutate(Min_time = min(Time), Max_time = max(Time),
           Exp_length = Max_time - Min_time) %>%
   dplyr::select(Condition, Strain, Plate, Exp_length) %>%
   distinct() #get the experiment length of each experiment
max_exp_length1 = max(exp_length1$Exp_length)



Concconvtime = full_join(OD_resultsconv, exp_length1) %>%
    #dplyr::filter(Strain == strain_name) %>%
    group_by(Date, Condition, Strain, Location, Continent, Year, Country, Plate, Location_year, New_Location_year) %>%
    dplyr::summarize(Concentration = median(Concentration)) %>%
    mutate(Time = (as.numeric(Date) / 60 / 60)) %>% ungroup() %>%
    group_by(Condition, Strain, Location, Continent, Year, Country, Location_year, New_Location_year) %>%
    mutate(Start_time = min(Time),
           End_time = max(Time),
           New_time = ifelse(Time == End_time, Start_time + max_exp_length1, Time), NEW_TIMES=New_time-min(Time)) %>% ungroup() ## Normalize the time of all the experiments 


all_strains = unique(Concconvtime$Strain)
all_conditions = unique(Concconvtime$Condition)



for (strain_name in all_strains )  {                           
for (tempe in all_conditions ) {
  
  temp6 = ungroup(Concconvtime) %>%
    filter(Strain == strain_name) %>%
    filter(Condition == tempe)

  
  lol = unique(temp6$Location)
    cont = unique(temp6$Continent)
    coun = unique(temp6$Country)
    lol_year= unique(temp6$New_Location_year)
    year=unique(temp6$Year)
  
    try({

    msplineday = loess(Concentration~NEW_TIMES, temp6)
    
    
    temp_tibblespliday = tibble(day1 = approx(x=msplineday$x, y=msplineday$fitted, xout = 24)$y, day2 =approx(x=msplineday$x, y=msplineday$fitted, xout = 48)$y , day3 = approx(x=msplineday$x, y=msplineday$fitted, xout = 72)$y, day4 = approx(x=msplineday$x, y=msplineday$fitted, xout =96)$y, day5 = approx(x=msplineday$x, y=msplineday$fitted, xout = 120)$y, day6 = approx(x=msplineday$x, y=msplineday$fitted, xout =144)$y,
                         Condition = tempe, Strain = strain_name, Location = lol, Continent = cont, Country = coun, New_Location_year=lol_year, Year=year, Model="loess")

    if (tempe == all_conditions[[1]] && strain_name == all_strains[[1]]) { 
      spline_curvesspliday = temp_tibblespliday
    }
    else { 
      spline_curvesspliday = bind_rows(spline_curvesspliday, temp_tibblespliday)

    }
    })
}
}    

spline_curvesspliday<-spline_curvesspliday %>% dplyr::select(-day1, -day3, -day5, -Model)  %>% left_join(characters.df) 

write.csv(spline_curvesspliday, "spline_curvesspliday.csv", row.names = FALSE)


```


```{r days, message=FALSE, warning=FALSE, error=FALSE, fig.cap="LOF filter with the density of concentration after 48h, 96h and 144h per curve calculation per temperature, and Number of observations kept or filtered out post LOF filtering of the concentration after 48h, 96h and 144h per curve calculation per temperature."}

spline_curvesspliday<-read.csv("spline_curvesspliday.csv", header=TRUE)

spline_curvesspliday$Year<- as.character(spline_curvesspliday$Year)


conditions = c("12C", "15C", "18C", "22C", "25C")
kneighbors = c(3, 10, 20, 30, 40)

df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvesspliday %>% drop_na(day2) %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$day2)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(day2, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, day2)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofday2_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofday2_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Concentration after 48h calculations kept or filtered per temperature")

lofday2_filtered<-df20 %>% filter(lof_filter == "Non-outlier")


df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvesspliday %>% drop_na(day4) %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$day4)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(day4, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, day4)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofday4_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofday4_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Concentration after 96 hours calculations kept or filtered per temperature")

lofday4_filtered<-df20 %>% filter(lof_filter == "Non-outlier")



df = tibble()
for ( cond in conditions) {
  for (kn in kneighbors) {
    temp<-spline_curvesspliday %>% drop_na(day6) %>% filter(Condition == cond)
    temp$scaled_data <- scale(temp$day6)
    temp$kn <- kn
    temp$lof <- dbscan::lof(temp$scaled_data, minPts = kn)
    temp$lof_filter = ifelse(temp$lof > 2 , "Outlier", "Non-outlier")
    df = bind_rows(df, temp)
}
}

df %>%
  ggplot(aes(day6, lof, color = lof_filter)) + 
  geom_point()+ 
  facet_grid(rows = vars(Condition), cols = vars(kn))


df %>%
  ggplot(aes(Condition, day6)) + 
  geom_jitter(aes(color = lof_filter), width = .2, alpha = .4)+ 
  geom_violin(alpha = .4) + 
  facet_grid(rows = vars(kn)) +
  labs(subitle = "The facets are the number of neighhors used.") +
  coord_flip() +
  scale_color_manual(values = c("grey", "goldenrod"))


df20<-df %>% filter(kn == 20)

#Create some stats to estimate how much is filtered and kept per timepoint and condition
lofday6_count_filtering =  df20 %>% 
  count(Strain, Condition, lof_filter)

temp = lofday6_count_filtering  %>%
  count(Condition, lof_filter, n) %>%
  mutate(lof_filter = case_when(lof_filter == "Non-outlier" ~ "Kept",
                                  lof_filter == "Outlier" ~ "Filtered-out")
         ) %>%
  filter(!is.na(lof_filter)) 


temp %>%
  ggplot(aes(Condition, fill = lof_filter, y = nn)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#368F8B", "grey10")) +
  theme_bw()+labs(y="Number of observations", title="Concentration after 144 hours calculations kept or filtered per temperature")

lofday6_filtered<-df20 %>% filter(lof_filter == "Non-outlier")


lofdaysvars<-(ungroup(lofday2_filtered)%>% dplyr::select(-day4, -day6, -lof, -lof_filter, -kn, -scaled_data)) %>% full_join( (ungroup(lofday4_filtered)%>% dplyr::select(-day2, -day6, -lof, -lof_filter, -kn, -scaled_data))) %>% full_join((ungroup(lofday6_filtered)%>% dplyr::select(-day2, -day4, -lof, -lof_filter, -kn, -scaled_data)))

widelofdaysvars_filtered<- (lofday2_filtered %>% dplyr::select(ID_genome_file, Condition, day2) %>% pivot_wider(names_from = "Condition", names_prefix = "day2_",values_from = "day2")) %>% full_join((lofday4_filtered %>% dplyr::select(ID_genome_file, Condition, day4) %>% pivot_wider(names_from = "Condition", names_prefix = "day4_",values_from = "day4"))) %>% full_join((lofday6_filtered %>% dplyr::select(ID_genome_file, Condition, day6) %>% pivot_wider(names_from = "Condition", names_prefix = "day6_",values_from = "day6")))

widelofdaysvars_filteredeu<- (lofday2_filtered %>% filter(Adm_Cluster == "Europe") %>% dplyr::select(ID_genome_file, Condition, day2) %>% pivot_wider(names_from = "Condition", names_prefix = "day2_",values_from = "day2")) %>% full_join((lofday4_filtered %>% filter(Adm_Cluster == "Europe")%>% dplyr::select(ID_genome_file, Condition, day4) %>% pivot_wider(names_from = "Condition", names_prefix = "day4_",values_from = "day4"))) %>% full_join((lofday6_filtered %>% filter(Adm_Cluster == "Europe")%>% dplyr::select(ID_genome_file, Condition, day6) %>% pivot_wider(names_from = "Condition", names_prefix = "day6_",values_from = "day6")))

widelofdaysvars_filteredna<- (lofday2_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, day2) %>% pivot_wider(names_from = "Condition", names_prefix = "day2_",values_from = "day2")) %>% full_join((lofday4_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, day4) %>% pivot_wider(names_from = "Condition", names_prefix = "day4_",values_from = "day4"))) %>% full_join((lofday6_filtered %>% filter(Adm_Cluster %in% c("USA")) %>% dplyr::select(ID_genome_file, Condition, day6) %>% pivot_wider(names_from = "Condition", names_prefix = "day6_",values_from = "day6")))

write_csv(widelofdaysvars_filtered, "Phenotype_daysvars_lofoutrem.csv")

write_csv(widelofdaysvars_filteredeu, "Phenotype_daysvars_lofoutremeu.csv")

write_csv(widelofdaysvars_filteredna, "Phenotype_daysvars_lofoutremna.csv")

```


### Number of spore concentrations after 48 hours of the start of the experiment per temperature treatment

```{r day2num, message=FALSE, fig.cap="Number of concentrations after 48 hours after the start of the experiment per temperature treatment.", warning=FALSE, error=FALSE}

lofdaysvars %>% dplyr::select(-day4, -day6) %>% drop_na(day2) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of spore concentration calculations after 48h of experiment", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)


```

### Number of spore concentrations after 96 hours of the start of the experiment per temperature treatment

```{r day4num, message=FALSE, fig.cap="Number of concentrations after 96 hours after the start of the experiment per temperature treatment.", warning=FALSE, error=FALSE}

lofdaysvars %>% dplyr::select(-day2, -day6) %>% drop_na(day4) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of spore concentration calculations after 96h of experiment", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)


```

### Number of spore concentrations after 144 hours of the start of the experiment per temperature treatment

```{r day6num, message=FALSE, fig.cap="Number of concentrations after 144 hours after the start of the experiment per temperature treatment.", warning=FALSE, error=FALSE}

lofdaysvars %>% dplyr::select(-day2, -day4) %>% drop_na(day6) %>% count(Condition) %>% ggplot(aes(x=Condition, y=n, fill=Condition, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+labs(x="Temperature", y="Number of observations", title="Number of spore concentration calculations after 144h of experiment", subtitle= "per temperature treatment")+ geom_text(vjust=1.6, color="white", size=6)



```





## Correlations between the variables for the different temperatures

It is quite evident from the calculations and how we obtained them that most of the variables analyzed are correlated with each other (see Figure \@ref(fig:correl)), since they all come from the growth curves themselves. The asymptote of the logistic regression surprisingly is the variable that mostly is not correlated with many of the other variables, except for the inflection time. Additionally, the $T_{opt}$ or optimal temperature (just Temperature in the graph) has inappreciable Pearson's correlation coefficients (<0.15) with the rest of the variables. 


```{r sumallvars, message=FALSE, warning=FALSE, error=FALSE}



summaryallvars<- full_join(lofarea_approx_filtered %>% dplyr::select(-Plate), loflogisticvars%>% dplyr::select(-Isolate_full_name_phenotyping, -RemoveIQR, -RemoveIQRmed) %>% ungroup()) %>% full_join((lofopttemperature_filtered %>% dplyr::select(-RemoveIQR, -RemoveIQRmed)%>% ungroup())%>% rename(OptTemperature=Temperature)) %>% full_join((lofopttempsens_filtered %>% ungroup() %>% dplyr::select(-Isolate_full_name_phenotyping, -RemoveIQR, -RemoveIQRmed))) %>% full_join((lofquantvars %>% ungroup() %>%dplyr::select(-Isolate_full_name_phenotyping, -RemoveIQR, -RemoveIQRmed))) %>% full_join((lofdaysvars %>% ungroup() %>% dplyr::select(-Isolate_full_name_phenotyping, -RemoveIQR, -RemoveIQRmed))) 

opttemponly<-summaryallvars%>% ungroup()  %>% dplyr::select(ID_genome_file, OptTemperature) %>% drop_na(OptTemperature) %>% group_by(ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature))

#summaryallvarswide<- full_join((area_approximations %>% ungroup() %>% dplyr::select(-Plate)%>% pivot_wider(names_from = "Condition", names_prefix = "eAUC",                            values_from = "approx_area")), (spline_summary2 %>% ungroup() %>% dplyr::select(-yinflection, -Model_based_AUC, -Isolate_full_name_phenotyping) %>% pivot_wider(names_from = Condition,
 #   names_sep = "_",
  #  values_from = c(asymptote, inflection, growth_rate)))) %>% full_join(spline_curvesq1%>% ungroup() %>% dplyr::select(-y, -Isolate_full_name_phenotyping, -rsquaredadjmq)%>% rename(Temperatureopt=Temperature))%>% full_join(spline_curvessensopttemp%>% ungroup() %>% dplyr::select(-Isolate_full_name_phenotyping) %>% pivot_wider(names_from =Condition, names_prefix = "Tempsens", names_sep = "_", values_from=c(Tempoptsens))) %>% full_join(spline_curvesspli%>% dplyr::select(-Isolate_full_name_phenotyping)%>% ungroup()%>% pivot_wider(names_from = Condition, names_sep = "_",                            values_from = c(q25total, q50total, q75total)) ) %>% full_join(spline_curvesspliday%>% dplyr::select(-Isolate_full_name_phenotyping)%>% ungroup() %>% pivot_wider(names_from = Condition, names_prefix = "Conc", names_sep = "_", values_from = c(day2, day4, day6))) 


summaryallvarswide1<- summaryallvars %>% ungroup() %>% group_by(ID_genome_file) %>% pivot_wider(names_from = "Condition", values_from = c("approx_area", "asymptote", "inflection", "growth_rate", "Tempoptsens", "q25total", "q50total", "q75total", "day2", "day4", "day6"))%>% dplyr::select(-y)


```


```{r correl, message=FALSE, fig.cap="pearson's correlations and their significance (*) of all the growth variables, optimal temperature and temperature sensitivities.", warning=FALSE, error=FALSE, fig.height=15, fig.width=15}

summaryallvars<-ungroup(summaryallvars) 

ggpairs(summaryallvars, columns =c(15:27), title = "",  
  axisLabels = "show", columnLabels = colnames(summaryallvars[, c(15:27)]), ggplot2::aes(colour = Condition, alpha=0.3), upper=list(continuous = wrap("cor", method = "pearson")))



```

# Phenotypic analysis of the temperature-associated growth, optimal temperature, sensitivity and climatic variables 

Therefore, from the `r length(row.names(area_approximations %>% ungroup() %>% dplyr::select(Location_year, Strain) %>% group_by(Location_year, Strain) %>% summarize()%>% ungroup() %>% group_by(Location_year) %>% count()))` initial populations' locations we obtained the the Kppen climate class. The Kppen climate categories are described in the Figure \@ref(fig:koppenexp) retrieved from Wikipedia (11/11/2023).

```{r koppenexp, fig.cap="Kppen climate classification notation description.", out.width = '50%', message=FALSE, warning=FALSE, error=FALSE} 

knitr::include_graphics("koppenclimateexplanation.png") 

```

```{r allvarsnumvarmoren5, message=FALSE, warning=FALSE, error=FALSE}

summaryallvars %>% ungroup() %>% dplyr::select(New_Location_year, Strain) %>% group_by(New_Location_year, Strain) %>% summarize()%>% ungroup() %>% group_by(New_Location_year) %>% count()%>% kable(caption="Number of strains per population (single location).")%>% kable_styling()


summaryallvars %>% ungroup() %>% dplyr::select(Condition, approx_area, asymptote, inflection, growth_rate, Tempoptsens, q25total, q50total, q75total, day2, day4, day6) %>% group_by(Condition) %>%
  summarise_at(vars(approx_area:day6), function(x) sum(x > 0, na.rm = TRUE))%>% kable(caption="Number of observations per each growth and sensitivity varible per condition for all populations.")%>% kable_styling()

```


```{r setupworldclim, message=FALSE, warning=FALSE, error=FALSE}


summaryallvars<-summaryallvars %>% left_join((OD_resultsconv %>% ungroup() %>% dplyr::select(Latitude, Longitude, New_Location_year, ID_genome_file) %>% group_by(Latitude, Longitude, New_Location_year, ID_genome_file) %>% summarize())) %>% drop_na(Latitude)

temp =  summaryallvars %>% mutate(X = as.numeric(unlist(Longitude)), Y = as.numeric(unlist(Latitude))) %>% dplyr::select(X, Y) %>% distinct()

sp = SpatialPoints(temp[, c("X", "Y")])

#Only the ones associated to temperature
Bioclim_var = c("Annual Mean Temperature", "Mean Diurnal Range ", "Isothermality (BIO2/BIO7) (100)", "Temperature Seasonality (standard deviation 100)","Max Temperature of Warmest Month", "Min Temperature of Coldest Month", "Temperature Annual Range (BIO5-BIO6)", "Mean Temperature of Wettest Quarter","Mean Temperature of Driest Quarter", "Mean Temperature of Warmest Quarter", "Mean Temperature of Coldest Quarter")

bio_list = list()
for (i in c(1:19)) {
  file_name=paste0("wc2.1__bio_tifffiles/wc2.1_10m_bio_", i, ".tif")
  rast_temp = raster(file_name)    
  bio_list[[Bioclim_var[i]]] = raster::extract(rast_temp, sp)
}

climate_per_coordinates = cbind(temp, do.call(cbind, bio_list)) %>% pivot_longer(cols=Bioclim_var,
                    names_to='Bioclim_var',
                    values_to='Estimate') %>% dplyr::select(-'NA') 

dat = summaryallvars %>% ungroup()%>% full_join((climate_per_coordinates %>% ungroup() %>% rename(Longitude=X, Latitude=Y))) 

datwidegea = summaryallvars %>% ungroup()%>% full_join((climate_per_coordinates %>% ungroup() %>% rename(Longitude=X, Latitude=Y))) %>%ungroup()%>% pivot_wider(names_from = Bioclim_var, values_from = Estimate) 


datwide<- dat %>%  pivot_wider(names_from = Bioclim_var, values_from = Estimate) 


datwide$Year<-as.integer(datwide$Year)

forpubOD_raw1.1<-forpubOD_raw1 %>% left_join(datwide, join_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, New_Location_year, Cluster, Continent, Adm_Cluster, Variety,  climate))

write.csv(forpubOD_raw1.1, "OD_resultsallvarsextra.csv", row.names = FALSE)


extrametadatasra<- read_csv("extrametadatasra.csv") %>% dplyr::select(-4) %>% rename(Previously_published=Not_previously_published)
extrametadatasra[is.na(extrametadatasra)] <- c("Yes")

forpubOD_raw4.1<-forpubOD_raw4 %>% left_join(datwide, join_by(Condition, Strain, ID_genome_file, Location, Geographic_region, Year, Country, New_Location_year, Cluster, Continent, Adm_Cluster, Variety,  climate)) %>% left_join(extrametadatasra)




write.csv(forpubOD_raw4.1, "OD_resultsallvarsextra_ALL_ALL.csv", row.names = FALSE)

summaryallvars$Year<-as.numeric(summaryallvars$Year)

metadataforpub<-datwide %>% group_by(ID_genome_file, Strain, Location, Geographic_region, Year, Country, New_Location_year, Adm_Cluster, Variety, climate) %>% count() %>% dplyr::select(-n) %>% ungroup() %>% left_join(extrametadatasra) %>% left_join(summaryallvars %>% dplyr::select(ID_genome_file, Longitude, Latitude)%>% group_by_all() %>% count() %>% dplyr::select(-n) %>% ungroup()) 

write.csv(metadataforpub, "metadataforpub.csv", row.names = FALSE)

dat %>% dplyr::select(Bioclim_var) %>% ungroup() %>% group_by(Bioclim_var) %>% summarize() %>% kable(caption="Temperature-associated bioclimatic variables from the database WorldClim2.")%>% kable_styling()

```


The climatic variables per populations' location that we used are the `r length(Bioclim_var)` bioclimatic temperature-associated variables from the database WorldClim2 (https://www.worldclim.org/) which was retrieved on October 6th 2023. The variables can be found in Table \@ref(tab:setupworldclim). The estimates for all these bioclimatic variables for the different populations are in Figure \@ref(fig:setupworldclim).


## Bioclimatic variables associated to temperature from populations' locations {.tabset}

```{r normalitycheckallvars,  message=FALSE, warning=FALSE, error=FALSE, eval=FALSE}



all_conditions<-unique(datwide$Condition)[c(1:5)]
allvars<-names(datwide%>% dplyr::select(-Longitude, -Latitude))[c(15:40)]

                         
for (tempe in all_conditions ) {
    for (i in 1:(length(names(datwide%>% dplyr::select(-Longitude, -Latitude))[c(15:40)]) )) {

      temp<-datwide %>% filter(Condition == tempe)


    lshap <- lapply(temp[,c(15:40)], shapiro.test)
    varname <- paste("pvalshapw", (names(datwide%>% dplyr::select(-Longitude, -Latitude))[c(17:41)][[i]]) , sep=".")
    My_sub <- temp %>% mutate(!!varname := lshap[[i]]$p.value)
    
  
     if (tempe == all_conditions[[1]]  && i == 1) { 
      pvalshapw2 = My_sub
    }
    else { 
      pvalshapw2 = full_join(pvalshapw2, My_sub)
    }
  } 
  
}

pvalshapw3<-pvalshapw2 %>% pivot_longer( 
    cols = starts_with("pvalshapw."), values_to = "normalpval")

pvalshapw4<- pvalshapw3 %>% filter(normalpval>0.05) %>% ungroup() %>% group_by(name, Condition) %>% count() %>% dplyr::select(-n)%>% kable(caption="Phenotypic variables per condition that are normally distributed (includes bioclimatic variables).")%>% kable_styling()
  
```

### Number of strains per bioclimatic variable estimates per population 

```{r numbperpopworldclim, message=FALSE, warning=FALSE, error=FALSE, fig.height=10, fig.width=16, fig.cap="Number of stains per estimates of bioclimatic variables associated to temperature per population."}



preadybioclim2<-ggplot(dat, aes(Estimate, fill =New_Location_year, color =New_Location_year)) +
   geom_histogram(position = "stack") +facet_wrap(.~Bioclim_var, scales = "free") +theme_minimal()+ theme( axis.line = element_line(colour = "grey50")) + scale_color_manual(values=palette_infinite(37))+ scale_fill_manual(values=palette_infinite(37))+labs(title="Number of strains per estimates", subtitle = "of temperature-associated bioclimatic variables for populations", y="Number of strains")

preadybioclim2


```


### Principal component analysis (PCA) of the temperature-associated bioclimatic variables per variables contribution to the first two components

Observing the bioclimatic variable contribution to the PCA with the scaled estimates we see that there are three variables that contribute the least: Mean Diurnal Range, Isothermality and the Temperature of the Wettest Quarter. 


```{r pcabioclimvars, message=FALSE, warning=FALSE, error=FALSE}

dataforpca = dat %>% ungroup() %>% group_by(Adm_Cluster, Continent, New_Location_year, Bioclim_var) %>% summarize(Estimate=mean(Estimate)) %>%
  pivot_wider(names_from = Bioclim_var, values_from = Estimate) 

dataforpca.pca <- prcomp(dataforpca[, 4:14],  
                   scale = TRUE)

#eigs <- dataforpca.pca$sdev^2

#rbind(SD = sqrt(eigs),Proportion = eigs/sum(eigs),Cumulative = cumsum(eigs)/sum(eigs))

#summary(dataforpca.pca)

PC1<-dataforpca.pca$x[,1]
PC2<-dataforpca.pca$x[,2]



fviz_pca_var(dataforpca.pca,
             col.var = "contrib", # Color by contributions to the PC
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"),
             repel = TRUE     # Avoid text overlapping
             )


stats <- summary(dataforpca.pca)
# extract variable importance and list items explaining up to 85% variance
importance <- stats$importance[3,]
#importance[importance <= 0.95]
resultNames <- names(importance[importance <= 0.95])
# return factor scores 
x_result <- as.data.frame(dataforpca.pca$x[,resultNames])
#head(x_result)

mergedData <- cbind(dataforpca,x_result) %>% ungroup() %>% dplyr::select(-Adm_Cluster, -Continent) %>% group_by_all() %>% summarize() %>% column_to_rownames('New_Location_year')



```

### First two components of the Principal component analysis (PCA) of the temperature-associated bioclimatic variables per population's genetic cluster


```{r pcabioclimadmclus, message=FALSE, warning=FALSE, error=FALSE}

mergedData1<-dataforpca %>% left_join(dat %>% dplyr::select(New_Location_year, climate, Adm_Cluster) %>% ungroup%>% group_by_all() %>% summarise()%>% ungroup()%>%  drop_na (Adm_Cluster, climate))


ggplot(mergedData1, 
       aes(x = PC1, 
           y = PC2, 
           color = Adm_Cluster, label = New_Location_year)) +
  geom_point() +
  scale_color_manual(values=K_colors) +
  geom_text(hjust = 0, nudge_x = 0.05) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+ labs(x="PC1 (52%)", y="PC2 (30%)")

```

### First two components of the Principal component analysis (PCA) of the temperature-associated bioclimatic variables per population's K-G climate class



```{r pcabioclimclimate, message=FALSE, warning=FALSE, error=FALSE}

mergedData2<-dataforpca %>% left_join(dat %>% dplyr::select(New_Location_year, climate, Adm_Cluster) %>% ungroup%>% group_by_all() %>% summarise()%>% drop_na(climate))

ggplot(mergedData2, aes(x=PC1, y=PC2, color = climate, label = New_Location_year)) +geom_point( size=4) +
    xlab("PC1") +
    ylab("PC2")+
  geom_text(hjust = 0, nudge_x = 0.05, color="black")+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+ labs(x="PC1 (52%)", y="PC2 (30%)")+
  scale_color_manual(values=c(clim_colors, "grey50")) +
  xlim(-8, 8)

```







## Correlations between the bioclimatic variables and the growth, optimal temperature and sensitivity variables for the different temperatures treatments {.tabset}

### Overall correlations of all the variables


```{r correlallvarswithbioclim, message=FALSE, fig.cap="Pearson's correlations and their significance (*) of all the growth variables, optimal temperature and temperature sensitivities and the temperature-associated bioclimatic variables.", warning=FALSE, error=FALSE, fig.height=20, fig.width=20}

datwide3<-datwide %>% dplyr::select(-Longitude, -Latitude)


#ggpairs(datwide3, columns =15:38, title = "",  
#  axisLabels = "show", columnLabels = colnames(datwide3[, 15:38]), ggplot2::aes(colour = #Condition, alpha=0.3), upper = list(continuous = wrap("cor", method = "pearson")))


datwide3eu<-datwide3 %>% ungroup() %>% filter(Adm_Cluster == "Europe")

#ggpairs(datwide3eu, columns =15:38, title = "",  
#  axisLabels = "show", columnLabels = colnames(datwide3eu[, 15:38]), ggplot2::aes(colour = #Condition, alpha=0.3), upper = list(continuous = wrap("cor", method = "pearson")))


datwide3na<-datwide3 %>% ungroup() %>% filter(Adm_Cluster == "USA")

#ggpairs(datwide3na, columns =15:38, title = "",  
#  axisLabels = "show", columnLabels = colnames(datwide3na[, 15:38]), ggplot2::aes(colour = #Condition, alpha=0.3), upper = list(continuous = wrap("cor", method = "pearson")))

opttemp<- datwide3 %>% dplyr::select(OptTemperature, ID_genome_file, climate, `Annual Mean Temperature`, `Mean Temperature of Warmest Quarter`, `Mean Temperature of Coldest Quarter`, `Mean Temperature of Wettest Quarter`, `Mean Temperature of Driest Quarter`)  %>% ungroup() %>% group_by(ID_genome_file, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T), Annual_Mean_Temperature=mean(`Annual Mean Temperature`, na.rm=T), Mean_Temperature_Warmest_Quarter=mean(`Mean Temperature of Warmest Quarter`, na.rm=T), Mean_Temperature_Coldest_Quarter=mean(`Mean Temperature of Coldest Quarter`, na.rm=T), Mean_Temperature_Wettest_Quarter=mean(`Mean Temperature of Wettest Quarter`, na.rm=T), Mean_Temperature_Driest_Quarter=mean(`Mean Temperature of Driest Quarter`, na.rm=T)) %>% mutate_all(~ifelse(is.nan(.), NA, .)) 

datwide4<-datwide3 %>% ungroup()%>% dplyr::select(ID_genome_file, Condition, growth_rate, inflection, asymptote, day2, day4, day6, q25total, q50total, q75total, Tempoptsens, approx_area) %>% group_by(ID_genome_file)%>% pivot_wider(names_from = "Condition", values_from = c("growth_rate", "inflection", "asymptote", "day2", "day4", "day6", "q25total", "q50total", "q75total", "Tempoptsens", "approx_area" )) %>% full_join(opttemp) %>% dplyr::select(-Annual_Mean_Temperature, -Mean_Temperature_Warmest_Quarter, -Mean_Temperature_Coldest_Quarter, -Mean_Temperature_Wettest_Quarter, -Mean_Temperature_Driest_Quarter, -climate)%>%ungroup()


#matdatwide4<-as.matrix(datwide4 %>% dplyr::select(-ID_genome_file))


#correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")
#testRes = cor.mtest(matdatwide4, conf.level = 0.95)
#please<-as.data.frame(correlplot2)%>% rownames_to_column(var = "Var1")
#pleasewhy<-left_join(please2, please3)
#write.csv(please, "Correlationsgrowthvariables.csv")

```

### Overall correlations of all the variables pertinent


```{r correlallvarswithbioclimsmall, message=FALSE, fig.cap="Pearson's correlations and their significance (*) of all the growth variables, optimal temperature and temperature sensitivities and the temperature-associated bioclimatic variables.", warning=FALSE, error=FALSE, fig.height=5, fig.width=5, eval=FALSE}

datwide3<-datwide %>% dplyr::select(-Longitude, -Latitude) 

opttemp<- datwide3 %>% dplyr::select(OptTemperature, ID_genome_file, climate, `Annual Mean Temperature`, `Mean Temperature of Warmest Quarter`, `Mean Temperature of Coldest Quarter`, `Mean Temperature of Wettest Quarter`, `Mean Temperature of Driest Quarter`)  %>% ungroup() %>% group_by(ID_genome_file, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T), Annual_Mean_Temperature=mean(`Annual Mean Temperature`, na.rm=T), Mean_Temperature_Warmest_Quarter=mean(`Mean Temperature of Warmest Quarter`, na.rm=T), Mean_Temperature_Coldest_Quarter=mean(`Mean Temperature of Coldest Quarter`, na.rm=T), Mean_Temperature_Wettest_Quarter=mean(`Mean Temperature of Wettest Quarter`, na.rm=T), Mean_Temperature_Driest_Quarter=mean(`Mean Temperature of Driest Quarter`, na.rm=T)) %>% mutate_all(~ifelse(is.nan(.), NA, .)) 

datwide4<-datwide3 %>% dplyr::select(ID_genome_file, climate, Condition, Adm_Cluster, growth_rate, inflection) %>% pivot_wider(names_from = "Condition", values_from = c("growth_rate", "inflection")) %>% full_join(opttemp) 

ggpairs(datwide4, columns =4:18, title = "",  
  axisLabels = "show", columnLabels = colnames(datwide4[, 4:18]), ggplot2::aes( alpha=0.3), upper = list(continuous = wrap("cor", method = "pearson")))



matdatwide4<-as.matrix(datwide4 %>% dplyr::select(4:18))


correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")

corrplot(correlplot2, type="lower", insig = "blank", tl.col="black")


matdatwide4<-as.matrix(datwide4 %>% filter(Adm_Cluster=="USA") %>% dplyr::select(4:18))

correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")


corrplot(correlplot2, type="lower", insig = "blank", tl.col="black")


matdatwide4<-as.matrix(datwide4 %>% filter(Adm_Cluster=="Europe") %>% dplyr::select(4:18))

correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")



corrplot(correlplot2, type="lower", insig = "blank", tl.col="black")


datwide4<-datwide3 %>% dplyr::select(ID_genome_file, climate, Condition, Adm_Cluster, growth_rate, day4) %>% pivot_wider(names_from = "Condition", values_from = c("growth_rate", "day4")) %>% full_join(opttemp) 

ggpairs(datwide4, columns =4:19, title = "",  
  axisLabels = "show", columnLabels = colnames(datwide4[, 4:19]), ggplot2::aes( alpha=0.3), upper = list(continuous = wrap("cor", method = "pearson")))



colnames(datwide4)<- c("ID_genome_file","climate","Adm_Cluster","growth rate 12C", "growth rate 15C", "growth rate 18C", "growth rate 22C", "growth rate 25C", "concentration after 96 h 12C", "concentration after 96 h 15C", "concentration after 96 h 18C", "concentration after 96 h 22C", "concentration after 96 h 25C",  "Optimal Temperature", "Annual Mean Temperature", "Mean Temperature of the Warmest Quarter", "Mean Temperature of the Coldest Quarter", "Mean Temperature of the Wettest Quarter", "Mean Temperature of the Driest Quarter")

matdatwide4<-as.matrix(datwide4 %>% dplyr::select(4:19))

#cor.test(datwide4$OptTemperature, datwide4$Mean_Temperature_Coldest_Quarter, method = "pearson")



correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")


#colnames(correlplot2)<- c("growth rate 12C", "growth rate 15C", "growth rate 18C", "growth rate 22C", "growth rate 25C", "concentration after 96 h 12C", "concentration after 96 h 15C", "concentration after 96 h 18C", "concentration after 96 h 22C", "concentration after 96 h 25C",  "Optimal Temperature", "Annual Mean Temperature", "Mean Temperature of the Warmest Quarter", "Mean Temperature of the Coldest Quarter", "Mean Temperature of the Wettest Quarter", "Mean Temperature of the Driest Quarter")

#rownames(correlplot2)<- c("growth rate 12C", "growth rate 15C", "growth rate 18C", "growth rate 22C", "growth rate 25C", "concentration after 96 h 12C", "concentration after 96 h 15C", "concentration after 96 h 18C", "concentration after 96 h 22C", "concentration after 96 h 25C",  "Optimal temperature", "Annual Mean Temperature", "Mean Temperature of the Warmest Quarter", "Mean Temperature of the Coldest Quarter", "Mean Temperature of the Wettest Quarter", "Mean Temperature of the Driest Quarter")

#colnames(matdatwide4)<- c("growth rate 12C", "growth rate 15C", "growth rate 18C", "growth rate 22C", "growth rate 25C", "concentration after 96 h 12C", "concentration after 96 h 15C", "concentration after 96 h 18C", "concentration after 96 h 22C", "concentration after 96 h 25C",  "Optimal Temperature", "Annual Mean Temperature", "Mean Temperature of the Warmest Quarter", "Mean Temperature of the Coldest Quarter", "Mean Temperature of the Wettest Quarter", "Mean Temperature of the Driest Quarter")

testRes = cor.mtest(matdatwide4, conf.level = 0.95)

corrplot(correlplot2, type="lower", p.mat = testRes$p, sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9, insig = 'label_sig', pch.col = 'grey20', tl.col="black", col=colorRampPalette(c("#5B8C5A","white","#8F3985"))(200), mar=c(1,1,1,1), tl.pos = "ld", tl.cex = 0.5)
#, col=colorRampPalette(c("#2f6952","white","#692f46"))(200)

matdatwide4<-as.matrix(datwide4 %>% filter(Adm_Cluster=="USA") %>% dplyr::select(4:18))

correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")


corrplot(correlplot2, type="lower", insig = "blank", tl.col="black")


matdatwide4<-as.matrix(datwide4 %>% filter(Adm_Cluster=="Europe") %>% dplyr::select(4:18))

correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")



corrplot(correlplot2, type="lower", insig = "blank", tl.col="black")


datwide4<-datwide3 %>% dplyr::select(ID_genome_file, climate, Condition, Adm_Cluster, growth_rate, inflection, day4, day6, q50total, approx_area, Tempoptsens) %>% pivot_wider(names_from = "Condition", values_from = c("growth_rate", "inflection", "day4", "day6","q50total", "approx_area",  "Tempoptsens")) %>% full_join(opttemp) %>% dplyr::select( -growth_rate_NA, -inflection_NA, -day4_NA, -day6_NA, -q50total_NA, -approx_area_NA, -Tempoptsens_NA)

matdatwide4<-as.matrix(datwide4 %>% dplyr::select(4:44))

correlplot2<-cor(matdatwide4, use = "pairwise.complete.obs", method="pearson")

#colnames(correlplot2)<- c("growth rate 12C", "growth rate 15C", "growth rate 18C", "growth rate 22C", "growth rate 25C", "concentration after 96 h 12C", "concentration after 96 h 15C", "concentration after 96 h 18C", "concentration after 96 h 22C", "concentration after 96 h 25C",  "Optimal Temperature", "Annual Mean Temperature", "Mean Temperature of the Warmest Quarter", "Mean Temperature of the Coldest Quarter", "Mean Temperature of the Wettest Quarter", "Mean Temperature of the Driest Quarter")

#rownames(correlplot2)<- c("growth rate 12C", "growth rate 15C", "growth rate 18C", "growth rate 22C", "growth rate 25C", "concentration after 96 h 12C", "concentration after 96 h 15C", "concentration after 96 h 18C", "concentration after 96 h 22C", "concentration after 96 h 25C",  "Optimal temperature", "Annual Mean Temperature", "Mean Temperature of the Warmest Quarter", "Mean Temperature of the Coldest Quarter", "Mean Temperature of the Wettest Quarter", "Mean Temperature of the Driest Quarter")


corrplot(correlplot2, type="lower", insig = "blank", tl.col="black")

```




### Annual Mean Temperature vs. Optimal Temperature (($T_{opt}$))

The first figure shows the graphical representation of the relationship between the Annual Mean Temperature and the estimated Optimal Temperature (($T_{opt}$)) displaying the pearson coefficient for all the populations with more than 5 strains with defined ancestry. The second figure excludes the Canadian population. 

```{r optandannualtemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Annual Mean Temperature."}

datwide3<-datwide3 %>%rename(Annual_Mean_Temperature='Annual Mean Temperature')
datwide3opt<-datwide3 %>% ungroup() %>% group_by(climate,Location, OptTemperature, Annual_Mean_Temperature, Adm_Cluster, Strain) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) 
datwide3sum<- datwide3opt  %>% left_join(metadataforpub %>% dplyr::select(ID_genome_file, Strain, New_Location_year)) %>% dplyr::select(climate,Location, OptTemperature, Annual_Mean_Temperature, Adm_Cluster, New_Location_year) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate, Adm_Cluster, New_Location_year) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), OptTemperature=mean(OptTemperature, na.rm=TRUE))

names(clim_colors)<-c('BSh','BWk', 'Cfa', 'Cfb', 'Csa', 'Csb','Cwb','Dfb')

ggscatter(datwide3opt%>% drop_na(OptTemperature), x='Annual_Mean_Temperature', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=OptTemperature, color=climate), size=5, alpha=0.5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)





ggscatter(datwide3opt %>% drop_na(OptTemperature), x='Annual_Mean_Temperature', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "spearman", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


ggscatter(datwide3opt %>% filter(!(Location == "Canada")), x='Annual_Mean_Temperature', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location == "Canada")), aes(x=Annual_Mean_Temperature, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location == "Canada")), aes(x=Annual_Mean_Temperature, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


ggscatter(datwide3opt %>% drop_na(OptTemperature) %>% filter(Adm_Cluster=="USA"), x='Annual_Mean_Temperature', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(Adm_Cluster=="USA"), aes(x=Annual_Mean_Temperature, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(Adm_Cluster=="USA"), aes(x=Annual_Mean_Temperature, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3opt %>% drop_na(OptTemperature) %>% filter(Adm_Cluster=="Europe"), x='Annual_Mean_Temperature', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(Adm_Cluster=="Europe"), aes(x=Annual_Mean_Temperature, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(Adm_Cluster=="Europe"), aes(x=Annual_Mean_Temperature, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)






#

datwide3sum<- datwide3opt %>% left_join(metadataforpub %>% dplyr::select(ID_genome_file, Strain, New_Location_year)) %>% dplyr::select(climate,Location, OptTemperature, Annual_Mean_Temperature, Adm_Cluster, New_Location_year) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate, New_Location_year) %>% summarise(Annual_Mean_Temperature=median(Annual_Mean_Temperature, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3opt%>% drop_na(OptTemperature), x='Annual_Mean_Temperature', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey50", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=OptTemperature, color=climate), size=5, alpha=0.6, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)+labs(y="Optimal temperature (C)", x="Annual Mean Temperature (C)")


```

### Mean Temperature of the Wettest Quarter vs. Optimal Temperature (($T_{opt}$))


```{r optandwettest, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Mean Temperature of the Wettest Quarter."}

datwide3<-datwide3 %>%rename(Mean_Temperature_of_the_Wettest_Quarter="Mean Temperature of Wettest Quarter")
datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Mean_Temperature_of_the_Wettest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_the_Wettest_Quarter=mean(Mean_Temperature_of_the_Wettest_Quarter, na.rm=TRUE), OptTemperature=mean(OptTemperature, na.rm=TRUE))

ggscatter(datwide3, x='Mean_Temperature_of_the_Wettest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=OptTemperature, color=climate), size=5, alpha=0.5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=OptTemperature, label=Location), size=3)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

#

datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Mean_Temperature_of_the_Wettest_Quarter, Adm_Cluster) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_the_Wettest_Quarter=median(Mean_Temperature_of_the_Wettest_Quarter, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3%>% drop_na(OptTemperature), x='Mean_Temperature_of_the_Wettest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey50", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=OptTemperature, color=climate), size=5, alpha=0.6, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```

### Mean Diurnal Range vs. Optimal Temperature (($T_{opt}$))


```{r optandmeandiur, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Mean Diurnal Range."}

datwide3<-datwide3 %>%rename(Mean_Diurnal_Range="Mean Diurnal Range ")
datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Mean_Diurnal_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Diurnal_Range=mean(Mean_Diurnal_Range, na.rm=TRUE), OptTemperature=mean(OptTemperature, na.rm=TRUE))

ggscatter(datwide3, x='Mean_Diurnal_Range', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Diurnal_Range, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Diurnal_Range, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Min Temperature of the Coldest Month vs. Optimal Temperature (($T_{opt}$))


```{r optandmincoldtemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Min Temperature of the Coldest Month."}

datwide3<-datwide3 %>%rename(Min_Temperature_of_the_Coldest_Month="Min Temperature of Coldest Month" )
datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Min_Temperature_of_the_Coldest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=mean(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), OptTemperature=mean(OptTemperature, na.rm=TRUE))

ggscatter(datwide3, x='Min_Temperature_of_the_Coldest_Month', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="gray80", size=1) +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=OptTemperature, color=climate), alpha=0.5, size=4) + geom_text_repel(data=datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=OptTemperature, label=Location), size=3)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)+ labs(x="Minimum Temperature of the Coldest Month (C)", y= bquote(Temperature[opt] (C)))

ggscatter((datwide3 %>% filter(!(Location=="Canada"))), x='Min_Temperature_of_the_Coldest_Month', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location=="Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location=="Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)+ labs(x="Minimum Temperature of the Coldest Month (C)", y= bquote(Temperature[opt] (C)))

#

datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Min_Temperature_of_the_Coldest_Month, Adm_Cluster) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=median(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3%>% drop_na(OptTemperature), x='Min_Temperature_of_the_Coldest_Month', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey30", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=OptTemperature, color=climate), size=5, alpha=0.7, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Mean Temperature of the Coldest Quarter vs. Optimal Temperature (($T_{opt}$))


```{r optandmeancoldtemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Mean Temperature of the Coldest Quarter."}

datwide3<-datwide3 %>%rename(Mean_Temperature_of_Coldest_Quarter="Mean Temperature of Coldest Quarter" )



datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Mean_Temperature_of_Coldest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=mean(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), OptTemperature=mean(OptTemperature, na.rm=TRUE))

ggscatter(datwide3, x='Mean_Temperature_of_Coldest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter((datwide3 %>% filter(!(Location=="Canada"))), x='Mean_Temperature_of_Coldest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location=="Canada")), aes(x=Mean_Temperature_of_Coldest_Quarter, y=OptTemperature, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location=="Canada")), aes(x=Mean_Temperature_of_Coldest_Quarter, y=OptTemperature, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

#

datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Mean_Temperature_of_Coldest_Quarter, Adm_Cluster) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=median(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3%>% drop_na(OptTemperature), x='Mean_Temperature_of_Coldest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey30", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=OptTemperature, color=climate), size=5, alpha=0.7, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Mean Temperature of the Driest Quarter vs. Optimal Temperature (($T_{opt}$))


```{r optandmeandriesttemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Mean Temperature of the Driest Quarter."}


datwide3<-datwide3 %>%rename(Mean_Temperature_of_Driest_Quarter="Mean Temperature of Driest Quarter")



#


datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Mean_Temperature_of_Driest_Quarter, Adm_Cluster) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Driest_Quarter=median(Mean_Temperature_of_Driest_Quarter, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3%>% drop_na(OptTemperature), x='Mean_Temperature_of_Driest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey50", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Driest_Quarter, y=OptTemperature, color=climate), size=5, alpha=0.6, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```

### Mean Temperature of the Warmest Quarter vs. Optimal Temperature (($T_{opt}$))


```{r optandmeanwarmtemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Mean Temperature of the Warmest Quarter."}


datwide3<-datwide3 %>%rename(Mean_Temperature_of_Warmest_Quarter="Mean Temperature of Warmest Quarter")



#
datwide3opt<-datwide3 %>% ungroup() %>% group_by(climate,New_Location_year, OptTemperature, Mean_Temperature_of_Warmest_Quarter, Adm_Cluster, Strain) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) 
datwide3sum<- datwide3opt%>% left_join(metadataforpub %>% dplyr::select(ID_genome_file, Strain, New_Location_year)) %>% dplyr::select(climate, OptTemperature, Mean_Temperature_of_Warmest_Quarter, Adm_Cluster, New_Location_year) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by( climate, New_Location_year) %>% summarise(Mean_Temperature_of_Warmest_Quarter=median(Mean_Temperature_of_Warmest_Quarter, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3opt%>% drop_na(OptTemperature), x='Mean_Temperature_of_Warmest_Quarter', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey50", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=OptTemperature, color=climate), size=5, alpha=0.6, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)+ labs(y="Optimal temperature (C)", x="Mean Temperature of Warmest Quarter (C)")


```


### Max Temperature of the Warmest Quarter vs. Optimal Temperature (($T_{opt}$))


```{r optandmaxwarmtemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the optimal temperature and the Max Temperature of the Warmest Quarter."}


datwide3<-datwide3 %>%rename(Max_Temperature_of_Warmest_Month="Max Temperature of Warmest Month")




datwide3sum<- datwide3 %>% dplyr::select(climate,Location, OptTemperature, Max_Temperature_of_Warmest_Month, Adm_Cluster) %>% drop_na(OptTemperature)%>% ungroup() %>% group_by(Location, climate) %>% summarise(Max_Temperature_of_Warmest_Month=median(Max_Temperature_of_Warmest_Month, na.rm=TRUE), OptTemperature=median(OptTemperature, na.rm=TRUE)) %>% filter(climate != "BWk") %>% filter(climate != "CWb")

ggscatter(datwide3%>% drop_na(OptTemperature), x='Max_Temperature_of_Warmest_Month', y="OptTemperature",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey30", alpha=0.5) +  geom_point(data= datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=OptTemperature, color=climate), size=5, alpha=0.7, shape=17) + theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```




### Min Temperature of the Coldest Month vs. inflection time at 12C


```{r mincoldtempandinfle12, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 12C and the Min Temperature of the Coldest Month."}

datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location, inflection, Min_Temperature_of_the_Coldest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=mean(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="12C"), x='Min_Temperature_of_the_Coldest_Month', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3 %>% filter(Condition=="12C") %>% filter(!(Location =="Canada")), x='Min_Temperature_of_the_Coldest_Month', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location =="Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location =="Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```

### Annual Mean Temperature vs. inflection time at 22C


```{r annualmeantempandinfle22, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 22C and the Annual Mean Temperature."}



datwide3sum<- datwide3 %>% filter(Condition=="22C") %>% dplyr::select(climate,Location, inflection, Annual_Mean_Temperature) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Annual_Mean_Temperature', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



ggscatter(datwide3 %>% filter(Condition=="22C")%>% filter(!(Location == "Canada")), x='Annual_Mean_Temperature', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location == "Canada")), aes(x=Annual_Mean_Temperature, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location == "Canada")), aes(x=Annual_Mean_Temperature, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```


### Isothermality vs. inflection time at 12C


```{r isotherandinfle12, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 12C and the Isothermality."}

datwide3<-datwide3 %>%rename(Isothermality="Isothermality (BIO2/BIO7) (100)")

datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location, inflection, Isothermality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Isothermality=mean(Isothermality, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Isothermality', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Isothermality, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Isothermality, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



ggscatter(datwide3 %>% filter(Condition=="12C")%>% filter(!(Location == "Canada")), x='Isothermality', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location == "Canada")), aes(x=Isothermality, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location == "Canada")), aes(x=Isothermality, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```

### Seasonality vs. inflection time at 12C

```{r seasonandinfle12, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 12C and the Seasonality."}

datwide3<-datwide3 %>%rename(Seasonality="Temperature Seasonality (standard deviation 100)")

datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location, inflection, Seasonality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Seasonality=mean(Seasonality, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Seasonality', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Seasonality, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Seasonality, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)




```

### Seasonality vs. inflection time at 18C

```{r seasonandinfle18, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 18C and the Seasonality."}


datwide3sum<- datwide3 %>% filter(Condition=="18C") %>% dplyr::select(climate,Location, inflection, Seasonality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Seasonality=mean(Seasonality, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="18C"), x='Seasonality', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Seasonality, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Seasonality, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)




```

### Max Temperature of Warmest Month vs. inflection time at 22C


```{r maxtempandinfle22, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 22C and the Max Temperature of the Warmest Month."}

#datwide3<-datwide3 %>%rename(Max_Temperature_of_Warmest_Month="Max Temperature of Warmest Month")

datwide3sum<- datwide3 %>% filter(Condition=="22C") %>% dplyr::select(climate,Location, inflection, Max_Temperature_of_Warmest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Max_Temperature_of_Warmest_Month=mean(Max_Temperature_of_Warmest_Month, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Max_Temperature_of_Warmest_Month', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Mean Temperature of Warmest Quarter vs. inflection time at 22C


```{r meanwarmtempandinfle22, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 22C and the Mean Temperature of the Warmest Quarter."}

#datwide3<-datwide3 %>%rename(Mean_Temperature_of_Warmest_Quarter="Mean Temperature of Warmest Quarter")

datwide3sum<- datwide3 %>% filter(Condition=="22C") %>% dplyr::select(climate,Location, inflection, Mean_Temperature_of_Warmest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Warmest_Quarter=mean(Mean_Temperature_of_Warmest_Quarter, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Mean_Temperature_of_Warmest_Quarter', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Mean Temperature of Warmest Quarter vs. inflection time at 18C


```{r maxtempandinfle18, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the inflection time at 18C and the Mean Temperature of the Warmest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="18C") %>% dplyr::select(climate,Location, inflection, Mean_Temperature_of_Warmest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Warmest_Quarter=mean(Mean_Temperature_of_Warmest_Quarter, na.rm=TRUE), inflection=mean(inflection, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="18C"), x='Mean_Temperature_of_Warmest_Quarter', y="inflection",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=inflection, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=inflection, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```



### Seasonality vs. logistic growth rate at 18C

```{r seasonandgrow18, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 18C and the Seasonality."}


datwide3sum<- datwide3 %>% filter(Condition=="18C") %>% dplyr::select(climate,Location, growth_rate, Seasonality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Seasonality=mean(Seasonality, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="18C"), x='Seasonality', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Seasonality, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Seasonality, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)




```




### Temperature Annual Range vs. logistic growth rate at 15C

```{r tempannualrangeandgrow15, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 15C and the Temperature Annual Range."}

datwide3<-datwide3 %>%rename(Temperature_Annual_Range='Temperature Annual Range (BIO5-BIO6)')
datwide3sum<- datwide3 %>% filter(Condition=="15C") %>% dplyr::select(climate,Location, growth_rate, Temperature_Annual_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Temperature_Annual_Range=mean(Temperature_Annual_Range, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="15C"), x='Temperature_Annual_Range', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Temperature_Annual_Range, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Temperature_Annual_Range, y=growth_rate, label=Location), size=4)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3%>% filter(Condition=="15C")%>% filter(!(Location=="Canada")), x='Temperature_Annual_Range', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location=="Canada")), aes(x=Temperature_Annual_Range, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location=="Canada")), aes(x=Temperature_Annual_Range, y=growth_rate, label=Location), size=4)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Annual Mean Temperature vs. logistic growth rate at 15C

```{r annualmeantempandgrow15, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 15C and the Annual Mean Temperature."}


datwide3sum<- datwide3 %>% filter(Condition=="15C") %>% dplyr::select(climate,Location, growth_rate, Annual_Mean_Temperature) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="15C"), x='Annual_Mean_Temperature', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=growth_rate, label=Location), size=4)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3%>% filter(Condition=="15C")%>% filter(!(Location=="Canada")), x='Annual_Mean_Temperature', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location=="Canada")), aes(x=Annual_Mean_Temperature, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location=="Canada")), aes(x=Annual_Mean_Temperature, y=growth_rate, label=Location), size=4)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Temperature Annual Range vs. logistic growth rate at 18C

```{r tempannualrangeandgrow18, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 18C and the Temperature Annual Range."}


datwide3sum<- datwide3 %>% filter(Condition=="18C") %>% dplyr::select(climate,Location, growth_rate, Temperature_Annual_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Temperature_Annual_Range=mean(Temperature_Annual_Range, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="18C"), x='Temperature_Annual_Range', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Temperature_Annual_Range, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Temperature_Annual_Range, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3%>% filter(Condition=="18C")%>% filter(!(Location=="Canada")), x='Temperature_Annual_Range', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location=="Canada")), aes(x=Temperature_Annual_Range, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location=="Canada")), aes(x=Temperature_Annual_Range, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Mean Temperature of the Wettest Quarter vs. logistic growth rate 15C


```{r grow15andwettest, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate 15C and the Mean Temperature of the Wettest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="15C")%>% dplyr::select(climate,Location, growth_rate, Mean_Temperature_of_the_Wettest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_the_Wettest_Quarter=mean(Mean_Temperature_of_the_Wettest_Quarter, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="15C"), x='Mean_Temperature_of_the_Wettest_Quarter', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Min Temperature of the Coldest Month vs. logistic growth rate at 15C


```{r mincoldtempandgrowth15, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 15C and the Min Temperature of the Coldest Month."}

datwide3sum<- datwide3 %>% filter(Condition=="15C") %>% dplyr::select(climate,Location, growth_rate, Min_Temperature_of_the_Coldest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=mean(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="15C"), x='Min_Temperature_of_the_Coldest_Month', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3 %>% filter(Condition=="15C") %>% filter(!(Location =="Canada")), x='Min_Temperature_of_the_Coldest_Month', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location =="Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location =="Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Mean Temperature of the Driest Quarter vs. logistic growth rate at 22C

```{r driestandgrowth22, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 22C and the Mean Temperature of the Driest Quarter."}

#datwide3<-datwide3 %>%rename(Mean_Temperature_of_Driest_Quarter="Mean Temperature of Driest Quarter")
datwide3sum<- datwide3 %>% filter(Condition=="22C") %>% dplyr::select(climate,Location, growth_rate, Mean_Temperature_of_Driest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Driest_Quarter=mean(Mean_Temperature_of_Driest_Quarter, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Mean_Temperature_of_Driest_Quarter', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Driest_Quarter, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Driest_Quarter, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```

### Mean Temperature of the Warmest Quarter vs. logistic growth rate at 15C

```{r warmestandgrowth15, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 15C and the Mean Temperature of the Warmest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="15C") %>% dplyr::select(climate,Location, growth_rate, Mean_Temperature_of_Warmest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Warmest_Quarter=mean(Mean_Temperature_of_Warmest_Quarter, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="15C"), x='Mean_Temperature_of_Warmest_Quarter', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```


### Mean Temperature of the Coldest Quarter vs. logistic growth rate at 15C

```{r coldestandgrowth15, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 15C and the Mean Temperature of the Coldest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="15C") %>% dplyr::select(climate,Location, growth_rate, Mean_Temperature_of_Coldest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=mean(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="15C"), x='Mean_Temperature_of_Coldest_Quarter', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```

### Mean Temperature of the Coldest Quarter vs. logistic growth rate at 12C

```{r coldestandgrowth12, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the logistic growth rate at 12C and the Mean Temperature of the Coldest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location, growth_rate, Mean_Temperature_of_Coldest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=mean(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), growth_rate=mean(growth_rate, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Mean_Temperature_of_Coldest_Quarter', y="growth_rate",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=growth_rate, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=growth_rate, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```



### Mean Temperature of the Coldest Quarter vs. asymptote 12C


```{r asym12andmeancold, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the asymptote 12C and the Mean Temperature of the Coldest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, asymptote, Mean_Temperature_of_Coldest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=mean(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), asymptote=mean(asymptote, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Mean_Temperature_of_Coldest_Quarter', y="asymptote",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=asymptote, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=asymptote, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Min Temperature of the Coldest Month vs. asymptote 12C


```{r asym12andmincold, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the asymptote 12C and the Min Temperature of the Coldest Month."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, asymptote, Min_Temperature_of_the_Coldest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=mean(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), asymptote=mean(asymptote, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Min_Temperature_of_the_Coldest_Month', y="asymptote",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=asymptote, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=asymptote, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Isothermality vs. asymptote 12C


```{r asym12andisotherm, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the asymptote 12C and the Isothermality."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, asymptote, Isothermality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Isothermality=mean(Isothermality, na.rm=TRUE), asymptote=mean(asymptote, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Isothermality', y="asymptote",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Isothermality, y=asymptote, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Isothermality, y=asymptote, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```



### Mean Diurnal Range vs. eAUC at 25C

```{r meandiuandeAUC25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the eAUC at 25C and the Mean Diurnal Range."}


datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location, approx_area, Mean_Diurnal_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Diurnal_Range=mean(Mean_Diurnal_Range, na.rm=TRUE), approx_area=mean(approx_area, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Mean_Diurnal_Range', y="approx_area",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Diurnal_Range, y=approx_area, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Diurnal_Range, y=approx_area, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```


### Temperature Annual Range vs. the eAUC at 25C

```{r tempannualeauc25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the eAUC at 25C and the Temperature Annual Range."}


datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location, approx_area, Temperature_Annual_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Temperature_Annual_Range=mean(Temperature_Annual_Range, na.rm=TRUE), approx_area=mean(approx_area, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Temperature_Annual_Range', y="approx_area",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Temperature_Annual_Range, y=approx_area, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Temperature_Annual_Range, y=approx_area, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```

### Seasonality vs. the eAUC at 22C

```{r seasonalityeauc25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the eAUC at 22C and the Temperature Seasonality."}


datwide3sum<- datwide3 %>% filter(Condition=="22C") %>% dplyr::select(climate,Location, approx_area, Seasonality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Seasonality=mean(Seasonality, na.rm=TRUE), approx_area=mean(approx_area, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Seasonality', y="approx_area",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Seasonality, y=approx_area, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Seasonality, y=approx_area, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```



### Mean Temperature of the Driest Quarter vs. the time to 3rd quantile at 12C 

```{r driestandq75, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the time to 3rd quantile at 12C and the Mean Temperature of the Driest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location, q75total, Mean_Temperature_of_Driest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Driest_Quarter=mean(Mean_Temperature_of_Driest_Quarter, na.rm=TRUE), q75total=mean(q75total, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Mean_Temperature_of_Driest_Quarter', y="q75total",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Driest_Quarter, y=q75total, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Driest_Quarter, y=q75total, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)


```


### Mean Temperature of the Warmest Quarter vs. the time to 3rd quantile at 12C


```{r q75andmeanwarm, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the time to 3rd quantile at 12C and the Mean Temperature of the Warmest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, q75total, Mean_Temperature_of_Warmest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Warmest_Quarter=mean(Mean_Temperature_of_Warmest_Quarter, na.rm=TRUE), q75total=mean(q75total, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Mean_Temperature_of_Warmest_Quarter', y="q75total",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=q75total, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=q75total, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Mean Temperature of the Coldest Quarter vs. the time to 3rd quantile at 12C


```{r q75andmeancold, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the time to 3rd quantile at 12C and the Mean Temperature of the Coldest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, q75total, Mean_Temperature_of_Coldest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=mean(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), q75total=mean(q75total, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Mean_Temperature_of_Coldest_Quarter', y="q75total",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=q75total, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=q75total, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Min Temperature of the Coldest Month vs. the time to 3rd quantile at 12C


```{r q75andmincold, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the time to 3rd quantile at 12C and the Min Temperature of the Coldest Month."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, q75total, Min_Temperature_of_the_Coldest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=mean(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), q75total=mean(q75total, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Min_Temperature_of_the_Coldest_Month', y="q75total",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=q75total, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=q75total, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Annual Mean Temperature vs. the time to 3rd quantile at 12C


```{r q75andannualmeantemp, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the time to 3rd quantile at 12C and the Annual Mean Temperature."}


datwide3sum<- datwide3 %>% filter(Condition=="12C")%>% dplyr::select(climate,Location, q75total, Annual_Mean_Temperature) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), q75total=mean(q75total, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Annual_Mean_Temperature', y="q75total",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=q75total, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=q75total, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Annual Mean Temperature vs. concentration after 48 hours at 12C

```{r annualmeantempandday2, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 48 hours at 12C and the Annual Mean Temperature."}


datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location,day2, Annual_Mean_Temperature) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), day2=mean(day2, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Annual_Mean_Temperature', y="day2",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=day2, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=day2, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```



### Max Temperature of the Warmest Month vs. concentration after 48 hours at 12C

```{r maxtempwarmandday2, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 48 hours at 12C and the Max Temperature of the Warmest Month."}


datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location,day2, Max_Temperature_of_Warmest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Max_Temperature_of_Warmest_Month=mean(Max_Temperature_of_Warmest_Month, na.rm=TRUE), day2=mean(day2, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Max_Temperature_of_Warmest_Month', y="day2",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=day2, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=day2, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Mean Temperature of the Warmest Quarter vs. concentration after 48 hours at 12C

```{r meantempwarmandday2, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 48 hours at 12C and the Mean Temperature of the Warmest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="12C") %>% dplyr::select(climate,Location,day2, Mean_Temperature_of_Warmest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Warmest_Quarter=mean(Mean_Temperature_of_Warmest_Quarter, na.rm=TRUE), day2=mean(day2, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="12C"), x='Mean_Temperature_of_Warmest_Quarter', y="day2",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=day2, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Warmest_Quarter, y=day2, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Seasonality vs. concentration after 96 hours at 22C

```{r seasonandday4, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 96 hours at 22C and the Seasonality."}


datwide3sum<- datwide3 %>% filter(Condition=="22C") %>% dplyr::select(climate,Location,day4, Seasonality) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Seasonality=mean(Seasonality, na.rm=TRUE), day4=mean(day4, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Seasonality', y="day4",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Seasonality, y=day4, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Seasonality, y=day4, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Mean Diurnal Range vs. concentration after 96 hours at 25C

```{r meandiurrangeandday4, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 96 hours at 25C and the Mean Diurnal Range."}


datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location,day4, Mean_Diurnal_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Diurnal_Range=mean(Mean_Diurnal_Range, na.rm=TRUE), day4=mean(day4, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Mean_Diurnal_Range', y="day4",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Diurnal_Range, y=day4, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Diurnal_Range, y=day4, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Temperature Annual Range vs. concentration after 96 hours at 25C

```{r tempannualrangeandday4, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 96 hours at 25C and the Temperature Annual Range."}


datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location,day4, Temperature_Annual_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Temperature_Annual_Range=mean(Temperature_Annual_Range, na.rm=TRUE), day4=mean(day4, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Temperature_Annual_Range', y="day4",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Temperature_Annual_Range, y=day4, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Temperature_Annual_Range, y=day4, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```



### Annual Mean Temperature vs. concentration after 144 hours at 18C


```{r annualmeantempandday618, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 144 hours at 18C and the Annual Mean Temperature."}



datwide3sum<- datwide3 %>% filter(Condition=="18C") %>% dplyr::select(climate,Location,day6, Annual_Mean_Temperature) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), day6=mean(day6, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="18C"), x='Annual_Mean_Temperature', y="day6",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=day6, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=day6, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)




```


### Mean Temperature of the Wettest Quarter vs. concentration after 144 hours at 25C


```{r day625andwettest, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the concentration after 144 hours at 25C and the Mean Temperature of the Wettest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="25C")%>% dplyr::select(climate,Location, day6, Mean_Temperature_of_the_Wettest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_the_Wettest_Quarter=mean(Mean_Temperature_of_the_Wettest_Quarter, na.rm=TRUE), day6=mean(day6, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Mean_Temperature_of_the_Wettest_Quarter', y="day6",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=day6, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=day6, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Annual Mean Temperature vs. temperature sensitivity at 25C


```{r annualmeantempandtempsens25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 25C and the Annual Mean Temperature."}

datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location, Tempoptsens, Annual_Mean_Temperature) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Annual_Mean_Temperature=mean(Annual_Mean_Temperature, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="25C"), x='Annual_Mean_Temperature', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Annual_Mean_Temperature, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Annual_Mean_Temperature, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3 %>% filter(Condition=="25C")%>% filter(!(Location == "Canada")), x='Annual_Mean_Temperature', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location == "Canada")), aes(x=Annual_Mean_Temperature, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location == "Canada")), aes(x=Annual_Mean_Temperature, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```



### Mean Diurnal Range vs. temperature sensitivity at 18C


```{r meandiurrangeandtempsens18, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 18C and the Mean Diurnal Range."}

datwide3sum<- datwide3 %>% filter(Condition=="18C") %>% dplyr::select(climate,Location, Tempoptsens, Mean_Diurnal_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Diurnal_Range=mean(Mean_Diurnal_Range, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="18C"), x='Mean_Diurnal_Range', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Diurnal_Range, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Diurnal_Range, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Mean Diurnal Range vs. temperature sensitivity at 25C


```{r meandiurrangeandtempsens25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 25C and the Mean Diurnal Range."}

datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location, Tempoptsens, Mean_Diurnal_Range) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Diurnal_Range=mean(Mean_Diurnal_Range, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="25C"), x='Mean_Diurnal_Range', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Diurnal_Range, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Diurnal_Range, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Min Temperature of the Coldest Month vs. temperature sensitivity at 25C


```{r mincoldtempandtempsens25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 25C and the Min Temperature of the Coldest Month."}

datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location, Tempoptsens, Min_Temperature_of_the_Coldest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Min_Temperature_of_the_Coldest_Month=mean(Min_Temperature_of_the_Coldest_Month, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="25C"), x='Min_Temperature_of_the_Coldest_Month', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Min_Temperature_of_the_Coldest_Month, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3 %>% filter(Condition=="25C")%>% filter(!(Location == "Canada")), x='Min_Temperature_of_the_Coldest_Month', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location == "Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location == "Canada")), aes(x=Min_Temperature_of_the_Coldest_Month, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```


### Max Temperature of the Warmest Month vs. temperature sensitivity at 25C


```{r maxwarmtempandtempsens25, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 25C and the Max Temperature of the Warmest Month."}

datwide3sum<- datwide3 %>% filter(Condition=="25C") %>% dplyr::select(climate,Location, Tempoptsens, Max_Temperature_of_Warmest_Month) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Max_Temperature_of_Warmest_Month=mean(Max_Temperature_of_Warmest_Month, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3 %>% filter(Condition=="25C"), x='Max_Temperature_of_Warmest_Month', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Max_Temperature_of_Warmest_Month, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

ggscatter(datwide3 %>% filter(Condition=="25C")%>% filter(!(Location == "Canada")), x='Max_Temperature_of_Warmest_Month', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum%>% filter(!(Location == "Canada")), aes(x=Max_Temperature_of_Warmest_Month, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum%>% filter(!(Location == "Canada")), aes(x=Max_Temperature_of_Warmest_Month, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)

```



### Mean Temperature of the Wettest Quarter vs. temperature sensitivity at 22C


```{r tempsens22andwettest, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 22C and the Mean Temperature of the Wettest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="22C")%>% dplyr::select(climate,Location, Tempoptsens, Mean_Temperature_of_the_Wettest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_the_Wettest_Quarter=mean(Mean_Temperature_of_the_Wettest_Quarter, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="22C"), x='Mean_Temperature_of_the_Wettest_Quarter', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=Tempoptsens, label=Location), size=2)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```

### Mean Temperature of the Wettest Quarter vs. temperature sensitivity at 25C


```{r tempsens25andwettest, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 25C and the Mean Temperature of the Wettest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="25C")%>% dplyr::select(climate,Location, Tempoptsens, Mean_Temperature_of_the_Wettest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_the_Wettest_Quarter=mean(Mean_Temperature_of_the_Wettest_Quarter, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Mean_Temperature_of_the_Wettest_Quarter', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_the_Wettest_Quarter, y=Tempoptsens, label=Location), size=3)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```


### Mean Temperature of the Coldest Quarter vs. temperature sensitivity at 25C


```{r tempsens25andmeancold, warning=FALSE, error=FALSE, message=FALSE, fig.cap="Pearson's correlations and their significance (p<0.05) of the temperature sensitivity at 25C and the Mean Temperature of the Coldest Quarter."}


datwide3sum<- datwide3 %>% filter(Condition=="25C")%>% dplyr::select(climate,Location, Tempoptsens, Mean_Temperature_of_Coldest_Quarter) %>% ungroup() %>% group_by(Location, climate) %>% summarise(Mean_Temperature_of_Coldest_Quarter=mean(Mean_Temperature_of_Coldest_Quarter, na.rm=TRUE), Tempoptsens=mean(Tempoptsens, na.rm=TRUE))

ggscatter(datwide3%>% filter(Condition=="25C"), x='Mean_Temperature_of_Coldest_Quarter', y="Tempoptsens",add = "reg.line", conf.int = TRUE, cor.coef = TRUE, cor.method = "pearson", shape=1, color="grey") +  geom_point(data= datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=Tempoptsens, color=climate, alpha=0.5), size=5) + geom_text(data=datwide3sum, aes(x=Mean_Temperature_of_Coldest_Quarter, y=Tempoptsens, label=Location), size=3)+ theme_minimal() + theme(axis.line = element_line(colour = "grey50"))+scale_color_manual(values=clim_colors)



```



## Bioclimatic variables per climate plots {.tabset}


### Annual Mean Temperature 


```{r annualmeantempperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Annual_Mean_Temperature)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=palette_temperature)+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Annual Mean Temperature (C)")+ labs(fill = "", y="")

```


```{r annualmeantempperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Annual_Mean_Temperature , y = Location, fill = Annual_Mean_Temperature)) +
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=palette_temperature)+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Annual Mean Temperature (C)", y="")+
  theme(strip.text = element_text(colour = 'black', size=12))+
  theme(strip.background =element_rect(fill="white"))



p1<-datwide3  %>% ggplot(aes(x = Annual_Mean_Temperature , y = climate, fill = Annual_Mean_Temperature)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=palette_temperature)+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Annual Mean Temperature (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



```


### Mean Diurnal Range

"The mean of the monthly temperature ranges (monthly maximum minus monthly minimum). Since the climate data inputs are monthly or averaged months across multiple years, this calculation uses recorded temperature fluctuation within a month to capture diurnal temperature range. Using monthly averages in this manner is mathematically equivalent to calculating the temperature range for each day in a month, and averaging these values for the month."


```{r meandiurnalrangeperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Mean_Diurnal_Range)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Mean Diurnal Range (C)")+ labs(fill = "", y="")

```


```{r meandiurnalrangeperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Mean_Diurnal_Range , y = Location, fill = Mean_Diurnal_Range))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Mean Diurnal Range (C)", y="")

datwide3  %>% ggplot(aes(x = Mean_Diurnal_Range , y = climate, fill = Mean_Diurnal_Range)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Mean Diurnal Range (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



datwide3  %>% ggplot(aes(x = Mean_Diurnal_Range , y = climate, fill = Mean_Diurnal_Range))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Mean Diurnal Range (C)", y="")



```

### Isothermality

"Isothermality quantifies how large the day-tonight temperatures oscillate relative to the summer to-winter (annual) oscillations. An isothermal value of 100 indicates the diurnal temperature range is equivalent to the annual temperature range, while anything less than 100 indicates a smaller level of temperature variability within an average month relative to the year."


```{r isothermalityperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Isothermality)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Isothermality (%)")+ labs(fill = "", y="")

```


```{r isothermalityperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Isothermality , y = Location, fill = Isothermality))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Isothermality (%)", y="")

datwide3  %>% ggplot(aes(x = Isothermality , y = climate, fill = Isothermality)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Isothermality (%)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



datwide3  %>% ggplot(aes(x = Isothermality , y = climate, fill = Isothermality))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Isothermality (%)", y="")




```

### Seasonality

"The amount of temperature variation over a given period based on the ratio of the standard deviation of the monthly mean temperatures to the mean monthly temperature (also known as the coefficient of variation (CV))."


```{r seasonalityperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Seasonality)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Seasonality (SD x 100)")+ labs(fill = "", y="")

```


```{r seasonalityperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Seasonality , y = Location, fill = Seasonality))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Seasonality (SD x 100)", y="")

datwide3  %>% ggplot(aes(x = Seasonality , y = climate, fill = Seasonality)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Seasonality (SD x 100)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



datwide3  %>% ggplot(aes(x = Seasonality , y = climate, fill = Seasonality))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#ded2c8","#b2977e","#8d7054","#68533e","#4f3f2f"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Seasonality (SD x 100)", y="")

```


### Max Temperature of Warmest Month

"The maximum monthly temperature occurrence over a given year (time-series) or averaged span of years (normal)."


```{r maxtempwarmmonthperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Max_Temperature_of_Warmest_Month)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Max Temperature of Warmest Month (C)")+ labs(fill = "", y="")

```


```{r maxtempwarmmonthperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Max_Temperature_of_Warmest_Month , y = Location, fill = Max_Temperature_of_Warmest_Month))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Max Temperature of Warmest Month (C)", y="")

datwide3  %>% ggplot(aes(x = Max_Temperature_of_Warmest_Month , y = climate, fill = Max_Temperature_of_Warmest_Month)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Max Temperature of Warmest Month (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



datwide3  %>% ggplot(aes(x = Max_Temperature_of_Warmest_Month , y = climate, fill = Max_Temperature_of_Warmest_Month))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Max Temperature of Warmest Month (C)", y="")



```

### Min Temperature of Coldest Month

"The minimum monthly temperature occurrence over a given year (time-series) or averaged span of years (normal)."


```{r mintempcoldmonthperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Min_Temperature_of_the_Coldest_Month)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Min Temperature of Coldest Month (C)")+ labs(fill = "", y="")

```


```{r mintempcoldmonthperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Min_Temperature_of_the_Coldest_Month , y = Location, fill = Min_Temperature_of_the_Coldest_Month))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Min Temperature of Coldest Month (C)", y="")


datwide3  %>% ggplot(aes(x = Min_Temperature_of_the_Coldest_Month , y = climate, fill = Min_Temperature_of_the_Coldest_Month)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Min Temperature of Coldest Month (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))




datwide3  %>% ggplot(aes(x = Min_Temperature_of_the_Coldest_Month , y = climate, fill = Min_Temperature_of_the_Coldest_Month))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Min Temperature of Coldest Month (C)", y="")


```

### Temperature Annual Range

"A measure of temperature variation over a given period."


```{r tempannualrangeperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Temperature_Annual_Range)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Temperature Annual Range (C)")+ labs(fill = "", y="")

```


```{r tempannualrangeperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Temperature_Annual_Range , y = Location, fill = Temperature_Annual_Range))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Temperature Annual Range (C)", y="")

datwide3  %>% ggplot(aes(x = Temperature_Annual_Range , y = climate, fill = Temperature_Annual_Range))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Temperature Annual Range (C)", y="")

datwide3  %>% ggplot(aes(x = Temperature_Annual_Range , y = climate, fill = Temperature_Annual_Range)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Temperature Annual Range (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))

```

### Mean Temperature of the Wettest Quarter

"This quarterly index approximates mean temperatures that prevail during the wettest season."


```{r wettempperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Mean_Temperature_of_the_Wettest_Quarter)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Mean Temperature of the Wettest Quarter (C)")+ labs(fill = "", y="")

```


```{r wettempperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Mean_Temperature_of_the_Wettest_Quarter , y = Location, fill = Mean_Temperature_of_the_Wettest_Quarter)) +
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Mean Temperature of the Wettest Quarter (C)", y="")

p4<-datwide3  %>% ggplot(aes(x = Mean_Temperature_of_the_Wettest_Quarter , y = climate, fill = Mean_Temperature_of_the_Wettest_Quarter)) +
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12))+ theme(legend.position = "none")+ labs(x="Mean Temperature of the Wettest Quarter (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))




```


### Mean Temperature of the Driest Quarter

"This quarterly index approximates mean temperatures that prevail during the driest quarter."


```{r drytempperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Mean_Temperature_of_Driest_Quarter)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Mean Temperature of the Driest Quarter (C)")+ labs(fill = "", y="")

```


```{r drytempperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Mean_Temperature_of_Driest_Quarter , y = Location, fill = Mean_Temperature_of_Driest_Quarter)) +
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Mean Temperature of Driest Quarter (C)", y="")


p5<-datwide3  %>% ggplot(aes(x = Mean_Temperature_of_Driest_Quarter , y = climate, fill = Mean_Temperature_of_Driest_Quarter))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1) +scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12))+ theme(legend.position = "none")+ labs(x="Mean Temperature of Driest Quarter (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



```


### Mean Temperature of the Warmest Quarter

"This quarterly index approximates mean temperatures that prevail during the warmest quarter."


```{r warmquarttempperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Mean_Temperature_of_Warmest_Quarter)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Mean Temperature of the Warmest Quarter (C)")+ labs(fill = "", y="")

```


```{r warmquarttempperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Mean_Temperature_of_Warmest_Quarter , y = Location, fill = Mean_Temperature_of_Warmest_Quarter))+
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Mean Temperature of Warmest Quarter (C)", y="")

p2<-datwide3  %>% ggplot(aes(x = Mean_Temperature_of_Warmest_Quarter , y = climate, fill = Mean_Temperature_of_Warmest_Quarter))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1)+scale_fill_gradientn(colours=c("#fbf6f6","#dcada5","#ba5b4c","#a93320","#541910"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12))+ theme(legend.position = "none")+ labs(x="Mean Temperature of Warmest Quarter (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))




```


### Mean Temperature of the Coldest Quarter

"This quarterly index approximates mean temperatures that prevail during the coldest quarter."



```{r coldquarttempperclim, message=FALSE, warning=FALSE, error=FALSE,fig.width=5, fig.height=9}

ggplot(datwide3, aes(x = 1, y = Location, fill = Mean_Temperature_of_Coldest_Quarter)) +
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +facet_grid(climate ~ ., drop = TRUE,space="free",scales="free_y")+scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())+ggtitle("Mean Temperature of the Coldest Quarter (C)")+ labs(fill = "", y="")

```


```{r coldquarttempperclimloli, message=FALSE, warning=FALSE, error=FALSE,fig.width=6, fig.height=9}


datwide3  %>% ggplot(aes(x = Mean_Temperature_of_Coldest_Quarter , y = Location, fill = Mean_Temperature_of_Coldest_Quarter)) +
  geom_point( size=4, alpha=0.6, shape=21, stroke=1) +facet_grid(climate ~ ., space="free",scales="free", drop = TRUE)+scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"))+ theme(legend.position = "none")+ labs(x="Mean Temperature of Coldest Quarter (C)", y="")

p3<-datwide3  %>% ggplot(aes(x = Mean_Temperature_of_Coldest_Quarter , y = climate, fill = Mean_Temperature_of_Coldest_Quarter))+
  geom_point( size=4, alpha=0.1, shape=21, stroke=1)+scale_fill_gradientn(colours=c("#001450","#00238b","#002db2","#295fff","#c6d4ff"))+ theme_light() + theme(axis.line = element_line(colour = "grey50"), axis.text=element_text(size=12), axis.title=element_text(size=12) )+ theme(legend.position = "none")+ labs(x="Mean Temperature of Coldest Quarter (C)", y="")+
  theme(plot.margin = margin(1.2,1.2,1.2,1.2, "cm"))



plot_grid(
  p1, p2, p3, p4, p5, nrow = 2
)

```




## Interactions (statistical tests of differences between groups for all variables)

```{r interactions, message=FALSE, warning=FALSE, error=FALSE}

datwide3opttemp<-datwide3 %>% ungroup() %>% dplyr::select(Strain, New_Location_year, climate, Adm_Cluster, OptTemperature) %>% group_by(Strain, New_Location_year, climate, Adm_Cluster) %>% summarize(OptTemperature =mean(OptTemperature, na.rm=T))


datwide3opttemp[sapply(datwide3opttemp, is.nan)] <- NA

datwide3opttemp<-datwide3opttemp %>% ungroup() %>% drop_na(OptTemperature)

pop<-datwide3opttemp %>% group_by (Strain, New_Location_year) %>% summarize() %>% ungroup() %>% group_by(New_Location_year) %>% count() %>% filter(n < 5)

lessthan5pop<-pop$New_Location_year

clim<-datwide3opttemp %>% group_by (Strain, climate) %>% summarize() %>% ungroup() %>% group_by(climate) %>% count() %>% filter(n < 5)

lessthan5clim<-clim$climate




#Optimal Temperature

kruskal.test(OptTemperature ~ Adm_Cluster, data = datwide3opttemp)
datwide3opttemp %>% kruskal_effsize(OptTemperature ~ Adm_Cluster) #moderate
kruskal.test(OptTemperature ~ climate, data = datwide3opttemp  %>% filter(!(climate %in% lessthan5clim)))
datwide3opttemp %>% filter(!(climate %in% lessthan5clim)) %>% kruskal_effsize(OptTemperature ~ climate  ) #moderate
kruskal.test(OptTemperature ~ New_Location_year, data = datwide3opttemp %>% filter(!(New_Location_year %in% lessthan5pop)))
datwide3opttemp %>% filter(!(New_Location_year %in% lessthan5pop)) %>% kruskal_effsize(OptTemperature ~ New_Location_year) #large
leveneTest(OptTemperature ~ New_Location_year, data = datwide3opttemp %>% filter(!(New_Location_year %in% lessthan5pop)))
leveneTest(OptTemperature ~ climate, data = datwide3opttemp %>% filter(!(climate %in% lessthan5clim)))

#Sample size corrected 
lessthan5pop2<-c(lessthan5pop, "Iran_Dezful", "Switzerland_Eschikon_July_2016")
pop1<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop2)) %>% group_by (Strain, New_Location_year) %>% summarize() %>% ungroup() %>% group_by(New_Location_year) %>% count() %>% ungroup() %>% slice_min(n)

lessthan5pop1<-c(5)

datwide3subsamploc1<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc2<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc3<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc4<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc5<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc6<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc7<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc8<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc9<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

datwide3subsamploc10<-datwide3opttemp %>% ungroup()%>% filter(!(New_Location_year %in% lessthan5pop)) %>% group_by(New_Location_year) %>% sample_n(lessthan5pop1) %>% ungroup()

clim1<-datwide3opttemp%>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by (Strain, climate) %>% summarize() %>% ungroup() %>% group_by(climate) %>% count() %>% ungroup() %>% slice_min(n)

lessthan5clim1<-clim1$n

datwide3subsampclim1<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim2<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim3<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim4<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim5<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim6<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim7<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim8<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim9<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

datwide3subsampclim10<-datwide3opttemp %>% ungroup()%>% filter(!(climate %in% lessthan5clim)) %>% group_by(climate) %>% sample_n(lessthan5clim1)%>% ungroup()

k1<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim1 )
k2<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim2 )
k3<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim3 )
k4<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim4 )
k5<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim5 )
k6<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim6 )
k7<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim7 )
k8<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim8 )
k9<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim9 )
k10<-kruskal.test(OptTemperature ~ climate, data = datwide3subsampclim10 )
kruskalavg<-mean(k1$p.value, k2$p.value, k3$p.value, k4$p.value, k5$p.value, k6$p.value, k7$p.value, k8$p.value, k9$p.value, k10$p.value) #0.0473

k1<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc1 )
k2<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc2 )
k3<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc3 )
k4<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc4 )
k5<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc5 )
k6<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc6 )
k7<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc7 )
k8<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc8 )
k9<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc9 )
k10<-kruskal.test(OptTemperature ~ New_Location_year, data = datwide3subsamploc10 )
kruskalavg<-mean(k1$p.value, k2$p.value, k3$p.value, k4$p.value, k5$p.value, k6$p.value, k7$p.value, k8$p.value, k9$p.value, k10$p.value) # 0.0162

l1<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc1 )
l2<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc2 )
l3<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc3 )
l4<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc4 )
l5<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc5 )
l6<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc6 )
l7<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc7 )
l8<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc8 )
l9<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc9 )
l10<-leveneTest(OptTemperature ~ New_Location_year, data = datwide3subsamploc10 )
leveneavg<-mean(l1$`Pr(>F)`[1], l2$`Pr(>F)`[1], l3$`Pr(>F)`[1], l4$`Pr(>F)`[1], l5$`Pr(>F)`[1], l6$`Pr(>F)`[1], l7$`Pr(>F)`[1], l8$`Pr(>F)`[1], l9$`Pr(>F)`[1], l10$`Pr(>F)`[1]) # 0.9556

l1<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim1 )
l2<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim2 )
l3<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim3 )
l4<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim4 )
l5<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim5 )
l6<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim6 )
l7<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim7 )
l8<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim8 )
l9<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim9 )
l10<-leveneTest(OptTemperature ~ climate, data = datwide3subsampclim10 )
leveneavg<-mean(l1$`Pr(>F)`[1], l2$`Pr(>F)`[1], l3$`Pr(>F)`[1], l4$`Pr(>F)`[1], l5$`Pr(>F)`[1], l6$`Pr(>F)`[1], l7$`Pr(>F)`[1], l8$`Pr(>F)`[1], l9$`Pr(>F)`[1], l10$`Pr(>F)`[1]) # 0.9772




wilcoxclimopttemp<-datwide3opttemp %>% ungroup()%>% drop_na(OptTemperature) %>% wilcox_test( OptTemperature ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimopttemp <- datwide3opttemp %>% ungroup()%>% drop_na(OptTemperature) %>% dunn_test(OptTemperature ~ climate, p.adjust.method = "bonferroni")



#eAUC

kruskal.test(approx_area ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(approx_area ~ Condition)#large
kruskal.test(approx_area ~ climate, data = datwide3)
kruskal.test(approx_area ~ Adm_Cluster, data = datwide3)

kruskal.test(approx_area ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(approx_area ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(approx_area ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(approx_area ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(approx_area ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(approx_area ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="18C"))#significant
kruskal.test(approx_area ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(approx_area ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="22C"))#significant
kruskal.test(approx_area ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))#significant
kruskal.test(approx_area ~ climate, data = datwide3%>% filter(Condition=="25C"))#significant
datwide3 %>% filter(Condition=="25C") %>% kruskal_effsize(approx_area ~ climate)#small
kruskal.test(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="25C"))# significant

leveneTest(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="25C"))
leveneTest(approx_area ~ climate, data = datwide3%>% filter(Condition=="25C"))
leveneTest(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="22C"))
leveneTest(approx_area ~ climate, data = datwide3%>% filter(Condition=="22C"))
leveneTest(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="18C"))
leveneTest(approx_area ~ climate, data = datwide3%>% filter(Condition=="18C"))
leveneTest(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="15C"))
leveneTest(approx_area ~ climate, data = datwide3%>% filter(Condition=="15C"))
leveneTest(approx_area ~ New_Location_year, data = datwide3%>% filter(Condition=="12C"))
leveneTest(approx_area ~ climate, data = datwide3%>% filter(Condition=="12C"))

wilcoxclimapprox_area25c<-datwide3 %>% filter(Condition=="25C") %>% ungroup()%>% drop_na(approx_area) %>% wilcox_test( approx_area ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimapprox_area25c <- datwide3 %>% ungroup()%>% drop_na(approx_area) %>% filter(Condition=="25C") %>% dunn_test(approx_area ~ climate, p.adjust.method = "bonferroni")



#growthrate

kruskal.test(growth_rate ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(growth_rate ~ Condition)#moderate
kruskal.test(growth_rate ~ climate, data = datwide3)
kruskal.test(growth_rate ~ Adm_Cluster, data = datwide3)


kruskal.test(growth_rate ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(growth_rate ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(growth_rate ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))#significant
kruskal.test(growth_rate ~ climate, data = datwide3%>% filter(Condition=="15C"))#significant
datwide3 %>% filter(Condition=="15C") %>% kruskal_effsize(growth_rate ~ climate)#small
kruskal.test(growth_rate ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(growth_rate ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(growth_rate ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(growth_rate ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(growth_rate ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))#significant
kruskal.test(growth_rate ~ climate, data = datwide3%>% filter(Condition=="25C"))

wilcoxclimgrowthrate15c<-datwide3 %>% filter(Condition=="15C") %>% ungroup()%>% drop_na(growth_rate) %>% wilcox_test( growth_rate ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimgrowthrate15c <- datwide3 %>% ungroup()%>% drop_na(growth_rate) %>% filter(Condition=="15C") %>% dunn_test(growth_rate ~ climate, p.adjust.method = "bonferroni")



#inflection

kruskal.test(inflection ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(inflection ~ Condition)#large
kruskal.test(inflection ~ climate, data = datwide3)
kruskal.test(inflection ~ Adm_Cluster, data = datwide3)


kruskal.test(inflection ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))#significant
kruskal.test(inflection ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(inflection ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))#significant
kruskal.test(inflection ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(inflection ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C")) #significant
kruskal.test(inflection ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(inflection ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C")) # significant
kruskal.test(inflection ~ climate, data = datwide3%>% filter(Condition=="22C")) # significant
datwide3 %>% filter(Condition=="22C") %>% kruskal_effsize(inflection ~ climate)#small
kruskal.test(inflection ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))
kruskal.test(inflection ~ climate, data = datwide3%>% filter(Condition=="25C"))

wilcoxcliminflection22c<-datwide3 %>% filter(Condition=="22C") %>% ungroup()%>% drop_na(inflection) %>% wilcox_test( inflection ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

dunecliminflection22c <- datwide3 %>% ungroup()%>% drop_na(inflection) %>% filter(Condition=="22C") %>% dunn_test(inflection ~ climate, p.adjust.method = "bonferroni")



#asymptote

kruskal.test(asymptote ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(asymptote ~ Condition)#large
kruskal.test(asymptote ~ climate, data = datwide3)
kruskal.test(asymptote ~ Adm_Cluster, data = datwide3)


kruskal.test(asymptote ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(asymptote ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(asymptote ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(asymptote ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(asymptote ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(asymptote ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(asymptote ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C")) # significant
kruskal.test(asymptote ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(asymptote ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))
kruskal.test(asymptote ~ climate, data = datwide3%>% filter(Condition=="25C"))



#Tempoptsens

kruskal.test(Tempoptsens ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(Tempoptsens ~ Condition)#large
kruskal.test(Tempoptsens ~ climate, data = datwide3)
kruskal.test(Tempoptsens ~ Adm_Cluster, data = datwide3)


kruskal.test(Tempoptsens ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(Tempoptsens ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(Tempoptsens ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))#significant
kruskal.test(Tempoptsens ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(Tempoptsens ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C")) #significant
kruskal.test(Tempoptsens ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(Tempoptsens ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C")) # significant
kruskal.test(Tempoptsens ~ climate, data = datwide3%>% filter(Condition=="22C")) # significant
datwide3 %>% filter(Condition=="22C") %>% kruskal_effsize(Tempoptsens ~ climate)#small
kruskal.test(Tempoptsens ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C")) # significant
kruskal.test(Tempoptsens ~ climate, data = datwide3%>% filter(Condition=="25C")) #significant
datwide3 %>% filter(Condition=="25C") %>% kruskal_effsize(Tempoptsens ~ climate)#small

wilcoxclimTempoptsens22c<-datwide3 %>% filter(Condition=="22C") %>% ungroup()%>% drop_na(Tempoptsens) %>% wilcox_test( Tempoptsens ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimTempoptsens22c <- datwide3 %>% ungroup()%>% drop_na(Tempoptsens) %>% filter(Condition=="22C") %>% dunn_test(Tempoptsens ~ climate, p.adjust.method = "bonferroni")


wilcoxclimTempoptsens25c<-datwide3 %>% filter(Condition=="25C") %>% ungroup()%>% drop_na(Tempoptsens) %>% wilcox_test( Tempoptsens ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimTempoptsens25c <- datwide3 %>% ungroup()%>% drop_na(Tempoptsens) %>% filter(Condition=="25C") %>% dunn_test(Tempoptsens ~ climate, p.adjust.method = "bonferroni")

#q25total

kruskal.test(q25total ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(q25total ~ Condition)#small
kruskal.test(q25total ~ climate, data = datwide3)
kruskal.test(q25total ~ Adm_Cluster, data = datwide3)



kruskal.test(q25total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))#significant
kruskal.test(q25total ~ climate, data = datwide3%>% filter(Condition=="12C"))#significant
datwide3 %>% filter(Condition=="12C") %>% kruskal_effsize(q25total ~ climate)#small
kruskal.test(q25total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(q25total ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(q25total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(q25total ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(q25total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(q25total ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(q25total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))#significant
kruskal.test(q25total ~ climate, data = datwide3%>% filter(Condition=="25C"))

wilcoxclimq25total12c<-datwide3 %>% filter(Condition=="12C") %>% ungroup()%>% drop_na(q25total) %>% wilcox_test( q25total ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimq25total12c <- datwide3 %>% ungroup()%>% drop_na(q25total) %>% filter(Condition=="12C") %>% dunn_test(q25total ~ climate, p.adjust.method = "bonferroni")


#q50total

kruskal.test(q50total ~ Condition, data = datwide3)
kruskal.test(q50total ~ climate, data = datwide3)
kruskal.test(q50total ~ Adm_Cluster, data = datwide3)



kruskal.test(q50total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(q50total ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(q50total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(q50total ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(q50total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(q50total ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(q50total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(q50total ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(q50total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))
kruskal.test(q50total ~ climate, data = datwide3%>% filter(Condition=="25C"))


#q75total

kruskal.test(q75total ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(q75total ~ Condition)#small
kruskal.test(q75total ~ climate, data = datwide3)
kruskal.test(q75total ~ Adm_Cluster, data = datwide3)



kruskal.test(q75total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))#significant
kruskal.test(q75total ~ climate, data = datwide3%>% filter(Condition=="12C"))#significant
datwide3 %>% filter(Condition=="12C") %>% kruskal_effsize(q75total ~ climate)#small
kruskal.test(q75total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(q75total ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(q75total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(q75total ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(q75total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(q75total ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(q75total ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))
kruskal.test(q75total ~ climate, data = datwide3%>% filter(Condition=="25C"))

wilcoxclimq75total12c<-datwide3 %>% filter(Condition=="12C") %>% ungroup()%>% drop_na(q75total) %>% wilcox_test( q75total ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimq75total12c <- datwide3 %>% ungroup()%>% drop_na(q75total) %>% filter(Condition=="12C") %>% dunn_test(q75total ~ climate, p.adjust.method = "bonferroni")


#day2

kruskal.test(day2 ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(day2 ~ Condition)#large
kruskal.test(day2 ~ climate, data = datwide3)
kruskal.test(day2 ~ Adm_Cluster, data = datwide3)



kruskal.test(day2 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(day2 ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(day2 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(day2 ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(day2 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(day2 ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(day2 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(day2 ~ climate, data = datwide3%>% filter(Condition=="22C"))#significant
datwide3 %>% filter(Condition=="22C") %>% kruskal_effsize(day2 ~ climate)#small
kruskal.test(day2 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))
kruskal.test(day2 ~ climate, data = datwide3%>% filter(Condition=="25C"))#significant
datwide3 %>% filter(Condition=="25C") %>% kruskal_effsize(day2 ~ climate)#small

wilcoxclimday222c<-datwide3 %>% filter(Condition=="22C") %>% ungroup()%>% drop_na(day2) %>% wilcox_test( day2 ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimday222c <- datwide3 %>% ungroup()%>% drop_na(day2) %>% filter(Condition=="22C") %>% dunn_test(day2 ~ climate, p.adjust.method = "bonferroni")

wilcoxclimday225c<-datwide3 %>% filter(Condition=="25C") %>% ungroup()%>% drop_na(day2) %>% wilcox_test( day2 ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimday225c <- datwide3 %>% ungroup()%>% drop_na(day2) %>% filter(Condition=="25C") %>% dunn_test(day2 ~ climate, p.adjust.method = "bonferroni")


#day4

kruskal.test(day4 ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(day4 ~ Condition)#large
kruskal.test(day4 ~ climate, data = datwide3)
kruskal.test(day4 ~ Adm_Cluster, data = datwide3)



kruskal.test(day4 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))
kruskal.test(day4 ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(day4 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(day4 ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(day4 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(day4 ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(day4 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(day4 ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(day4 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))#significant
kruskal.test(day4 ~ climate, data = datwide3%>% filter(Condition=="25C"))#significant
datwide3 %>% filter(Condition=="25C") %>% kruskal_effsize(day4 ~ climate)#small

wilcoxclimday425c<-datwide3 %>% filter(Condition=="25C") %>% ungroup()%>% drop_na(day4) %>% wilcox_test( day4 ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimday425c <- datwide3 %>% ungroup()%>% drop_na(day4) %>% filter(Condition=="25C") %>% dunn_test(day4 ~ climate, p.adjust.method = "bonferroni")


#day6

kruskal.test(day6 ~ Condition, data = datwide3)
datwide3 %>% kruskal_effsize(day6 ~ Condition)#large
kruskal.test(day6 ~ climate, data = datwide3)
kruskal.test(day6 ~ Adm_Cluster, data = datwide3)



kruskal.test(day6 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="12C"))#significant
kruskal.test(day6 ~ climate, data = datwide3%>% filter(Condition=="12C"))
kruskal.test(day6 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="15C"))
kruskal.test(day6 ~ climate, data = datwide3%>% filter(Condition=="15C"))
kruskal.test(day6 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="18C"))
kruskal.test(day6 ~ climate, data = datwide3%>% filter(Condition=="18C"))
kruskal.test(day6 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="22C"))
kruskal.test(day6 ~ climate, data = datwide3%>% filter(Condition=="22C"))
kruskal.test(day6 ~ Adm_Cluster, data = datwide3 %>% filter(Condition=="25C"))#significant
kruskal.test(day6 ~ climate, data = datwide3%>% filter(Condition=="25C"))#significant
datwide3 %>% filter(Condition=="25C") %>% kruskal_effsize(day6 ~ climate)#small

wilcoxclimday625c<-datwide3 %>% filter(Condition=="25C") %>% ungroup()%>% drop_na(day6) %>% wilcox_test(day6 ~ climate)%>% adjust_pvalue(method = "bonferroni") #%>% add_significance("p.adj")

duneclimday625c <- datwide3 %>% ungroup()%>% drop_na(day6) %>% filter(Condition=="25C") %>% dunn_test(day6 ~ climate, p.adjust.method = "bonferroni")





```

##Plots of the interactions

```{r interactionplots, message=FALSE, warning=FALSE, error=FALSE}

d<-datwide3  %>% ungroup() %>% filter(Condition == "25C") %>% drop_na(approx_area)

d_summary <- datwide3 %>% ungroup() %>% filter(Condition == "25C") %>%
  group_by(climate) %>%
  summarise(
    approx_area = mean(approx_area, na.rm=TRUE))

wilcoxtempperconteauc25c<-d %>% ungroup() %>% filter(Condition == "25C") %>% pairwise_wilcox_test(approx_area ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p17<- d %>% ggplot(aes(x = climate, y = approx_area )) + geom_jitter(alpha=0.3, color = "#A5333E" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#A5333E", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = expression(paste("eAUC ((" ,"10^6", " spores /mL)","^2)")), title="eAUC at 25C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0.05, 0.05)), limits = c(0, 1000))+ stat_pvalue_manual(wilcoxtempperconteauc25c, tip.length=0, hide.ns = TRUE,  y.position = 825, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p17


d<-datwide3  %>% ungroup() %>% drop_na(growth_rate) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(growth_rate) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    growth_rate = mean(growth_rate, na.rm=TRUE))

wilcoxtemppercontgrowthrate<-d %>% ungroup() %>% drop_na(growth_rate) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(growth_rate ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p16<- d %>% ungroup() %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = growth_rate )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Logistic growth rate ", title="Growth rate at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.0025)), limits = c(0, 0.12))+ stat_pvalue_manual(wilcoxtemppercontgrowthrate, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p16


d<-datwide3  %>% ungroup() %>% filter(Condition == "15C") %>% drop_na(growth_rate)

d_summary <- datwide3 %>% ungroup() %>% filter(Condition == "15C") %>%
  group_by(climate) %>%
  summarise(
    growth_rate = mean(growth_rate, na.rm=TRUE))

wilcoxtemppercontgrowthrate15c<-d %>% ungroup() %>% filter(Condition == "15C") %>% pairwise_wilcox_test(growth_rate ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p18<- d %>% ungroup() %>% drop_na(growth_rate) %>% ggplot(aes(x = climate, y = growth_rate )) + geom_jitter(alpha=0.3, color = "#3596B5" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#3596B5", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Logistic growth rate", title="Growth rate at 15C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(0, 0.1))+ stat_pvalue_manual(wilcoxtemppercontgrowthrate15c, tip.length=0, hide.ns = TRUE,  y.position = 0.09, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p18



d<-datwide3  %>% ungroup() %>% drop_na(inflection) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(inflection) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    inflection = mean(inflection, na.rm=TRUE))

wilcoxtemppercontinflection<-d %>% ungroup() %>% drop_na(inflection) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(inflection ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20<- d %>% ungroup()%>% drop_na(inflection) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = inflection )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Inflection time (h)", title="Inflection time at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontinflection, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20



d<-datwide3  %>% ungroup() %>% drop_na(asymptote) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(asymptote) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    asymptote = mean(asymptote, na.rm=TRUE))

wilcoxtemppercontasymptote<-d %>% ungroup() %>% drop_na(asymptote) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(asymptote ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20a<- d %>% ungroup()%>% drop_na(asymptote) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = asymptote )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Asymptote (10^6 spores/mL)", title="Asymptote at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontasymptote, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20a



d<-datwide3  %>% ungroup() %>% drop_na(approx_area) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(approx_area) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    approx_area = mean(approx_area, na.rm=TRUE))

wilcoxtemppercontapprox_area<-d %>% ungroup() %>% drop_na(approx_area) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(approx_area ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20b<- d %>% ungroup()%>% drop_na(approx_area) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = approx_area )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "eAUC (10^6 spores/mL)^2", title="eAUC at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontapprox_area, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20b




d<-datwide3  %>% ungroup() %>% drop_na(day2) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(day2) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    day2 = mean(day2, na.rm=TRUE))

wilcoxtemppercontday2<-d %>% ungroup() %>% drop_na(day2) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(day2 ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20c<- d %>% ungroup()%>% drop_na(day2) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = day2 )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Concentration after 48h (10^6 spores/mL)", title="Concentration after 48h at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontday2, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20c


d<-datwide3  %>% ungroup() %>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    day4 = mean(day4, na.rm=TRUE))

wilcoxtemppercontday4<-d %>% ungroup() %>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(day4 ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20d<- d %>% ungroup()%>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = day4 )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Concentration after 96h (10^6 spores/mL)", title="Concentration after 96h at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontday4, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20d


d<-datwide3  %>% ungroup() %>% drop_na(day6) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(day6) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    day6 = mean(day6, na.rm=TRUE))

wilcoxtemppercontday6<-d %>% ungroup() %>% drop_na(day6) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(day6 ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20e<- d %>% ungroup()%>% drop_na(day6) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = day6 )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Concentration after 144h (10^6 spores/mL)", title="Concentration after 144h at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontday6, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20e




d<-datwide3  %>% ungroup() %>% drop_na(q25total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(q25total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    q25total = mean(q25total, na.rm=TRUE))

wilcoxtemppercontq25total<-d %>% ungroup() %>% drop_na(q25total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(q25total ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20f<- d %>% ungroup()%>% drop_na(q25total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = q25total )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Time to 1st quantile of concentration (h)", title="Time to 1st quantile of concentration at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontq25total, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20f



d<-datwide3  %>% ungroup() %>% drop_na(q50total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(q50total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    q50total = mean(q50total, na.rm=TRUE))

wilcoxtemppercontq50total<-d %>% ungroup() %>% drop_na(q50total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(q50total ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20g<- d %>% ungroup()%>% drop_na(q50total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = q50total )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Time to 2nd quantile of concentration (h)", title="Time to 2nd quantile of concentration at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontq50total, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20g


d<-datwide3  %>% ungroup() %>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    q75total = mean(q75total, na.rm=TRUE))

wilcoxtemppercontq75total<-d %>% ungroup() %>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(q75total ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20h<- d %>% ungroup()%>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = q75total )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Time to 3rd quantile of concentration (h)", title="Time to 3rd quantile of concentration at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontq75total, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p20h



d<-datwide3  %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    Tempoptsens = mean(Tempoptsens, na.rm=TRUE))

wilcoxtemppercontTempoptsens<-d %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(Tempoptsens ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p20i<- d %>% ungroup()%>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = Tempoptsens )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Temperature Sensitivity", title="Temperature Sensitivity at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontTempoptsens, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_point(aes(group = 1), data = d_summary, color="red")+
  geom_hline(yintercept=1, linetype='dotted', color="grey30")


p20i







d<-datwide3  %>% ungroup() %>% filter(Condition == "22C") %>% drop_na(inflection)

d_summary <- datwide3 %>% ungroup() %>% drop_na(inflection) %>% filter(Condition == "22C") %>%
  group_by(climate) %>%
  summarise(
    inflection = mean(inflection, na.rm=TRUE))

wilcoxtemppercontinflection22c<-d %>% ungroup() %>% drop_na(inflection) %>% filter(Condition == "22C") %>% pairwise_wilcox_test(inflection ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p21<- d %>% ungroup() %>% drop_na(inflection) %>% ggplot(aes(x = climate, y = inflection )) + geom_jitter(alpha=0.3, color = "#F4797D" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#F4797D", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Inflection time (h)", title="Inflection at 22C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(0, 250))+ stat_pvalue_manual(wilcoxtemppercontinflection22c, tip.length=0, hide.ns = TRUE,  y.position = 200, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p21



d<-datwide3  %>% ungroup() %>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    q75total = mean(q75total, na.rm=TRUE))

wilcoxtemppercontq75total<-d %>% ungroup() %>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(q75total ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p22<- d %>% ungroup()%>% drop_na(q75total) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = q75total )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Time (h)", title="Time to concentration's 3rd quantile at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontq75total, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p22


d<-datwide3  %>% ungroup() %>% filter(Condition == "12C") %>% drop_na(q75total)

d_summary <- datwide3 %>% ungroup() %>% drop_na(q75total) %>% filter(Condition == "12C") %>%
  group_by(climate) %>%
  summarise(
    q75total = mean(q75total, na.rm=TRUE))

wilcoxtemppercontq75total12c<-d %>% ungroup() %>% drop_na(q75total) %>% filter(Condition == "12C") %>% pairwise_wilcox_test(q75total ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p23<- d %>% ungroup() %>% drop_na(q75total) %>% ggplot(aes(x = climate, y = q75total )) + geom_jitter(alpha=0.3, color = "#74CCED" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#74CCED", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Time (h)", title="Time to concentration's 3rd quantile at 12C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(0, 170))+ stat_pvalue_manual(wilcoxtemppercontq75total12c, tip.length=0, hide.ns = TRUE,  y.position = 150, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p23



d<-datwide3  %>% ungroup() %>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    day4 = mean(day4, na.rm=TRUE))

wilcoxtemppercontday4<-d %>% ungroup() %>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(day4 ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p24<- d %>% ungroup()%>% drop_na(day4) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = day4 )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Concentration (10^6 spores/mL)", title="Concentration after 96h at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtemppercontday4, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p24


d<-datwide3  %>% ungroup() %>% filter(Condition == "25C") %>% drop_na(day4)

d_summary <- datwide3 %>% ungroup() %>% drop_na(day4) %>% filter(Condition == "25C") %>%
  group_by(climate) %>%
  summarise(
    day4 = mean(day4, na.rm=TRUE))

wilcoxtemppercontday425c<-d %>% ungroup() %>% drop_na(day4) %>% filter(Condition == "25C") %>% pairwise_wilcox_test(day4 ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p25<- d %>% ungroup() %>% drop_na(day4) %>% ggplot(aes(x = climate, y = day4 )) + geom_jitter(alpha=0.3, color = "#A5333E" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#A5333E", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Concentration (10^6 spores/mL)", title="Concentration after 96h at 25C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(-0.99, 10))+ stat_pvalue_manual(wilcoxtemppercontday425c, tip.length=0, hide.ns = TRUE,  y.position = 9, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p25



d<-datwide3  %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

d_summary <- datwide3 %>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>%
  group_by(climate, Condition) %>%
  summarise(
    Tempoptsens = mean(Tempoptsens, na.rm=TRUE))

wilcoxtempperconttempoptsens<-d %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% group_by(climate) %>% pairwise_wilcox_test(Tempoptsens ~ Condition)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( x = "Condition", step.increase = 0.05) 



p26<- d %>% ungroup()%>% drop_na(Tempoptsens) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ggplot(aes(x = Condition, y = Tempoptsens )) + scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ geom_jitter(alpha=0.2, aes(color = Condition), show.legend = F, size=1.5) + 
    geom_boxplot(aes(fill=Condition), show.legend = F, outlier.shape = NA, alpha=0.6) + scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Temperature Sensitivity", title="Temperature sensitivity at 5 temperatures", subtitle="per climates with more than 10 strains") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)))+ stat_pvalue_manual(wilcoxtempperconttempoptsens, tip.length=0, hide.ns = TRUE, step.increase = 0)+facet_wrap(~climate, scales="fixed", ncol=8)+ geom_line(aes(group = 1), data = d_summary, color="red")+ geom_point(aes(group = 1), data = d_summary, color="red")


p26


d<-datwide3  %>% ungroup() %>% filter(Condition == "25C") %>% drop_na(Tempoptsens)

d_summary <- datwide3 %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(Condition == "25C") %>%
  group_by(climate) %>%
  summarise(
    Tempoptsens = mean(Tempoptsens, na.rm=TRUE))

wilcoxtempperconttempoptsens25c<-d %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(Condition == "25C") %>% pairwise_wilcox_test(Tempoptsens ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p27<- d %>% ungroup() %>% drop_na(Tempoptsens) %>% ggplot(aes(x = climate, y = Tempoptsens )) + geom_jitter(alpha=0.3, color = "#A5333E" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#A5333E", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Temperature sensitivity", title="Temperature sensitivity at 25C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(0, 1.3))+ stat_pvalue_manual(wilcoxtempperconttempoptsens25c, tip.length=0, hide.ns = TRUE,  y.position = 1.1, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p27



d<-datwide3  %>% ungroup() %>% filter(Condition == "22C") %>% drop_na(Tempoptsens)

d_summary <- datwide3 %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(Condition == "22C") %>%
  group_by(climate) %>%
  summarise(
    Tempoptsens = mean(Tempoptsens, na.rm=TRUE))

wilcoxtempperconttempoptsens22c<-d %>% ungroup() %>% drop_na(Tempoptsens) %>% filter(Condition == "22C") %>% pairwise_wilcox_test(Tempoptsens ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p28<- d %>% ungroup() %>% drop_na(Tempoptsens) %>% ggplot(aes(x = climate, y = Tempoptsens )) + geom_jitter(alpha=0.3, color = "#F4797D" , show.legend = F, size=1.5) + 
    geom_boxplot(fill="#F4797D", show.legend = F, outlier.shape = NA, alpha=0.5) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Temperature sensitivity", title="Temperature sensitivity at 22C treatment", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(0, 1.5))+ stat_pvalue_manual(wilcoxtempperconttempoptsens22c, tip.length=0, hide.ns = TRUE,  y.position = 1.3, step.increase = 0.03)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)


p28


d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location, Country, Adm_Cluster) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))

d_summary <- datwide3 %>% ungroup() %>% drop_na(OptTemperature) %>%
  group_by(climate) %>%
  summarise(
    OptTemperature = mean(OptTemperature, na.rm=TRUE))%>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))

wilcoxtemppercontopttemperature<-d %>% ungroup()%>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk")) %>% ungroup() %>% pairwise_wilcox_test(OptTemperature ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p29<- d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% filter(!(climate == "Cwb")) %>% filter(!(climate == "BWk"))%>% ggplot(aes(x = climate, y = OptTemperature )) + geom_jitter(alpha=0.4, aes(colour=OptTemperature) , show.legend = F, size=2, width = 0.15) + 
    geom_violin(fill="grey", show.legend = F, outlier.shape = NA, alpha=0.3) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y ="Optimal temperature (C)") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(16, 27))+ stat_pvalue_manual(wilcoxtemppercontopttemperature, tip.length=0, hide.ns = TRUE,  y.position = 25.2, step.increase = 0.06)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)+ scale_colour_gradientn(colours = c("#955490", "#F4797D", "#A5333E"))



p29


d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location, Country, New_Location_year) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T)) %>%filter(!(New_Location_year == "Israel_Saad"))

d_summary <- datwide3 %>%filter(!(New_Location_year == "Israel_Saad")) %>% ungroup() %>% drop_na(OptTemperature) %>%
  group_by(New_Location_year) %>%
  summarise(
    OptTemperature = mean(OptTemperature, na.rm=TRUE))

wilcoxtemppercontopttemperature<-d %>% ungroup() %>% ungroup() %>% pairwise_wilcox_test(OptTemperature ~ New_Location_year)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p29lol<- d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, New_Location_year) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% ggplot(aes(x = New_Location_year, y = OptTemperature )) + geom_jitter(alpha=0.4, aes(colour=OptTemperature) , show.legend = F, size=2, width = 0.15) + 
    geom_violin(fill="grey", show.legend = F, outlier.shape = NA, alpha=0.3) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y ="Optimal temperature (C)") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(16, 27))+ stat_pvalue_manual(wilcoxtemppercontopttemperature, tip.length=0, hide.ns = TRUE,  y.position = 25.2, step.increase = 0.06)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)+ scale_colour_gradientn(colours = c("#955490", "#F4797D", "#A5333E"))



p29lol






d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location, Country, Adm_Cluster, ID_genome_file, New_Location_year) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))%>% filter(!(New_Location_year %in% lessthan5pop)) #%>%filter(!(New_Location_year == "Israel_Saad"))

d_summary <- datwide3%>% filter(!(New_Location_year %in% lessthan5pop))  %>% ungroup() %>% drop_na(OptTemperature) %>%
  group_by(New_Location_year) %>%
  summarise(
    OptTemperature = mean(OptTemperature, na.rm=TRUE))

#%>%filter(!(New_Location_year == "Israel_Saad"))

wilcoxtemppercontopttemperature<-d %>% ungroup() %>% ungroup() %>% pairwise_wilcox_test(OptTemperature ~ New_Location_year)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 


p29pop<- d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, New_Location_year) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% ggplot(aes(x = New_Location_year, y = OptTemperature )) + geom_jitter(alpha=0.4, aes(colour=OptTemperature) , show.legend = F, size=2) + 
    geom_violin(fill="grey", show.legend = F, outlier.shape = NA, alpha=0.3) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = "Optimal temperature (C)", x="") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(16, 27))+ stat_pvalue_manual(wilcoxtemppercontopttemperature, tip.length=0, hide.ns = TRUE,  y.position = 25.2, step.increase = 0.06)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)+ scale_colour_gradientn(colours = c("#955490", "#F4797D", "#A5333E")) 


p29pop


d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))


#makes more sense to make it for eAUC12C
pall<-read_csv("snpalleleinforefalt_chr1_ammonium_12C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)

pall<-pall%>% drop_na(OptTemperature) %>% drop_na(Allele)
pall$Allele<-as.character(pall$Allele)
pall %>%
  ggplot( aes(x=OptTemperature, fill=Allele, group=Allele)) +
    geom_density( alpha=0.8) + scale_fill_manual(values=c("#7EE8FA", "#E58C8A"))


d<-datwide3  %>% ungroup()%>% filter(Condition == "12C") %>% drop_na(approx_area)
pall<-read_csv("snpalleleinforefalt_chr1_ammonium_12C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)
pall<-pall%>% drop_na(approx_area) %>% drop_na(Allele)
pall$Allele<-as.character(pall$Allele)
pall %>%
  ggplot( aes(x=approx_area, fill=Allele, group=Allele)) +
    geom_density( alpha=0.8) + scale_fill_manual(values=c("#7EE8FA", "#E58C8A"))


d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))

pall<-read_csv("snpalleleinforefalt_chr4_compass_22C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)

pall<-pall%>% drop_na(OptTemperature) %>% drop_na(Allele)
pall$Allele<-as.character(pall$Allele)
pall %>%
  ggplot( aes(x=OptTemperature, fill=Allele, group=Allele)) +
    geom_density( alpha=0.5) + scale_fill_manual(values=c("#7EE8FA", "#E58C8A"))


d<-datwide3  %>% ungroup()%>% filter(Condition == "22C") %>% drop_na(approx_area)
pall<-read_csv("snpalleleinforefalt_chr4_compass_22C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)
pall<-pall%>% drop_na(approx_area) %>% drop_na(Allele)
pall$Allele<-as.character(pall$Allele)
pall %>%
  ggplot( aes(x=approx_area, fill=Allele, group=Allele)) +
    geom_density( alpha=0.8) + scale_fill_manual(values=c("#7EE8FA", "#E58C8A"))



d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))


#makes more sense to make it for eAUC12C
pall<-read_csv("snpalleleinforefalt_chr2_unknown_12C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)

pall<-pall%>% drop_na(OptTemperature) %>% drop_na(Allele)
pall$Allele<-as.character(pall$Allele)
pall %>%
  ggplot( aes(x=OptTemperature, fill=Allele, group=Allele)) +
    geom_density( alpha=0.8) + scale_fill_manual(values=c("#7EE8FA", "#E58C8A"))


d<-datwide3  %>% ungroup()%>% filter(Condition == "12C") %>% drop_na(approx_area)
pall<-read_csv("snpalleleinforefalt_chr2_unknown_12C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)
pall<-pall%>% drop_na(approx_area) %>% drop_na(Allele)
pall$Allele<-as.character(pall$Allele)
pall %>%
  ggplot( aes(x=approx_area, fill=Allele, group=Allele)) +
    geom_density( alpha=0.8) + scale_fill_manual(values=c("#7EE8FA", "#E58C8A"))






d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% filter(Adm_Cluster=="USA")

d_summary <- datwide3 %>% filter(Adm_Cluster=="USA") %>% ungroup() %>% drop_na(OptTemperature) %>%
  group_by(climate) %>%
  summarise(
    OptTemperature = mean(OptTemperature, na.rm=TRUE)) 

wilcoxtemppercontopttemperature<-d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% ungroup() %>% pairwise_wilcox_test(OptTemperature ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 



p30<- d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% ggplot(aes(x = climate, y = OptTemperature )) + geom_jitter(alpha=0.4, aes(colour=OptTemperature) , show.legend = F, size=2) + 
    geom_violin(fill="grey", show.legend = F, outlier.shape = NA, alpha=0.3) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = bquote(Temperature[opt] (C)), title="Optimal temperature USA", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(16, 27))+ stat_pvalue_manual(wilcoxtemppercontopttemperature, tip.length=0, hide.ns = TRUE,  y.position = 25.2, step.increase = 0.06)+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)+ scale_colour_gradientn(colours = c("#955490", "#F4797D", "#A5333E"))


p30

d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% filter(Adm_Cluster=="Europe")

d_summary <- datwide3 %>% filter(Adm_Cluster=="Europe") %>% ungroup() %>% drop_na(OptTemperature) %>%
  group_by(climate) %>%
  summarise(
    OptTemperature = mean(OptTemperature, na.rm=TRUE)) 

#wilcoxtemppercontopttemperature<-d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% ungroup() %>% pairwise_wilcox_test(OptTemperature ~ climate)%>% adjust_pvalue(method = "bonferroni") %>% add_significance("p.adj")%>% add_xy_position( scales="fixed") 



p31<- d %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate) %>% summarise(OptTemperature=mean(OptTemperature, na.rm=T)) %>% ggplot(aes(x = climate, y = OptTemperature )) + geom_jitter(alpha=0.4, aes(colour=OptTemperature) , show.legend = F, size=2) + 
    geom_violin(fill="grey", show.legend = F, outlier.shape = NA, alpha=0.3) +theme_minimal() +theme(axis.title.x = element_blank(), legend.position = "bottom") +labs(y = bquote(Temperature[opt] (C)), title="Optimal temperature Europe", subtitle= "for each climate class") + theme(text = element_text(size=18)) + theme(axis.line = element_line(colour = "grey50"))+ scale_y_continuous(expand = expansion(mult = c(0, 0.05)), limits = c(16, 27))+ geom_point(aes(group = 1), data = d_summary, color="#4f4f4f", shape=17, size=2)+ scale_colour_gradientn(colours = c("#955490", "#F4797D", "#A5333E"))


p31

#+ stat_pvalue_manual(wilcoxtemppercontopttemperature, tip.length=0, hide.ns = TRUE,  y.position = 25.2, step.increase = 0.06)


d<-datwide3  %>% ungroup() %>% drop_na(approx_area) 
all_strains<-unique(d$Strain)
  
for (strain in all_strains){
  tempnum=d %>% ungroup()%>% filter(Strain == strain) 
    tempnum1= tempnum %>% filter(approx_area == max(approx_area))
    
    if (strain == all_strains[[1]]) { 
      maxapproxareatable = tempnum1
    }
    else { 
      maxapproxareatable = bind_rows(maxapproxareatable,tempnum1)
    }
}

  
  
maxapproxareatable %>% ungroup() %>% group_by(Location, Condition, climate) %>% count() %>% ungroup()%>% group_by(Location) %>%
  mutate(freq = (n / sum(n))*100)%>% ggplot(aes(x=Location, y=freq, fill=Condition)) + geom_bar(stat="identity")  +  scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E")) + coord_flip()+ theme_minimal() +facet_grid(climate~., scales = "free_y", space = "free_y", switch = "y") +
  theme(strip.placement = "outside")+ geom_text(aes(label=n), color="white", size=3, position = position_stack(vjust = 0.5)) + ggtitle("Percentage and number of strains of a given Location \nwith larger eAUC for a particular temperature")+labs(y="Proportion of strains with maximum eAUC at a condition (%)")

d<-datwide3  %>% ungroup() %>% drop_na(approx_area) 
all_strains<-unique(d$Strain)
  
for (strain in all_strains){
  tempnum=d %>% ungroup()%>% filter(Strain == strain) 
    tempnum1= tempnum %>% filter(approx_area == max(approx_area))
    
    if (strain == all_strains[[1]]) { 
      maxapproxareatable = tempnum1
    }
    else { 
      maxapproxareatable = bind_rows(maxapproxareatable,tempnum1)
    }
}

  
  
maxapproxareatable %>% filter(climate != "Cwb") %>% filter(climate != "BWk") %>% ungroup() %>% group_by(Condition, climate) %>% count() %>% ungroup()%>% group_by(climate) %>% mutate(freq = (n / sum(n))*100)%>% ggplot(aes(x=climate, y=freq, fill=Condition)) + geom_bar(stat="identity")  +  scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E")) + coord_flip()+ theme_minimal() +
  theme(strip.placement = "outside")+ geom_text(aes(label=n), color="white", size=3, position = position_stack(vjust = 0.5)) + ggtitle("Percentage and number of strains of a given climate \nwith larger eAUC for a particular temperature")+labs(y="Proportion of strains with maximum eAUC at a condition (%)")





```

## Temperature Sensitivity comparisons to how good or bad the strains are at both warm and cold temperatures

The temperature sensitivity for 15C and 25C are similar so there will be used to compare how well certain strains do at warmer temperatures vs colder temperatures, or both. 


```{r temperaturesensitivity_comparisons, message=FALSE, warning=FALSE, error=FALSE}

lofopttempsens_filtered %>% ggplot(aes(x=Tempoptsens))+ geom_density(aes(fill=Condition, color= Condition), alpha=0.5, size=1.5) +  scale_fill_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))  +  scale_color_manual(values = c("#74CCED", "#3596B5", "#955490", "#F4797D", "#A5333E"))+ labs(x="Temperature Sensitivity", y="Density")


extravarsforsenstemp<-lofopttempsens_filtered %>% dplyr::select(ID_genome_file, Location, New_Location_year, Country, Adm_Cluster, Geographic_region, climate) %>% ungroup() %>% group_by_all() %>% count() %>% dplyr::select(-n)

tempsenscomp<-widelofopttempsens_filtered %>% dplyr::select(ID_genome_file, Tempsens15C, Tempsens25C) %>% drop_na(Tempsens15C, Tempsens25C) %>% 
  mutate(class_15C = ifelse(quantile(Tempsens15C, 1, na.rm = T) < Tempsens15C, 4, ifelse(quantile(Tempsens15C, 0.75, na.rm=T) < Tempsens15C, 3, ifelse(quantile(Tempsens15C, 0.5, na.rm=T) < Tempsens15C, 2, ifelse(quantile(Tempsens15C, 0.25, na.rm=T) < Tempsens15C, 1, ifelse(quantile(Tempsens15C, 0, na.rm=T) < Tempsens15C, 0, "nothing"))))))%>% 
  mutate(class_25C = ifelse(quantile(Tempsens25C, 1, na.rm = T) < Tempsens25C, 4, ifelse(quantile(Tempsens25C, 0.75, na.rm=T) < Tempsens25C, 3, ifelse(quantile(Tempsens25C, 0.5, na.rm=T) < Tempsens25C, 2, ifelse(quantile(Tempsens25C, 0.25, na.rm=T) < Tempsens25C, 1, ifelse(quantile(Tempsens25C, 0, na.rm=T) < Tempsens25C, 0, "nothing")))))) %>% filter(class_25C != "nothing") %>% filter(class_15C != "nothing") %>% left_join(extravarsforsenstemp) 

tempsenscomp$class_15C<-as.numeric(tempsenscomp$class_15C)
tempsenscomp$class_25C<-as.numeric(tempsenscomp$class_25C)


tempsenscomp2<-tempsenscomp%>% mutate(class_diff=class_25C-class_15C) 

tempsenscomp2a<-tempsenscomp2 %>% group_by(New_Location_year) %>% count() %>% rename(num_isolates=n) %>% ungroup()

#extrainfoloc<-read_csv("./01_Source_Tables/extravarsnewlocationandyearmorecomplete.csv")

tempsenscomp3<-tempsenscomp2 %>% ungroup() %>% group_by(New_Location_year,climate, class_diff) %>% count() %>% rename(numberperdiff=n) %>% ungroup() %>% left_join(tempsenscomp2a) %>% filter(num_isolates >= 5 ) %>% mutate(percentageofscore=(numberperdiff/num_isolates)*100) 

tempsenscomp3$class_diff<-as.character(tempsenscomp3$class_diff)

tempsenscomp4<-tempsenscomp3 %>% filter(class_diff == "3") %>% dplyr::select(percentageofscore, New_Location_year) %>% arrange(percentageofscore)
tempsenscomp5<-tempsenscomp3 %>% filter(class_diff == "0") %>% dplyr::select(percentageofscore, New_Location_year) %>% arrange(percentageofscore)

tempsenscomp4a<-tempsenscomp4  %>% dplyr::select( New_Location_year)
tempsenscomp5a<-tempsenscomp5 %>% dplyr::select(New_Location_year)

tempsens5b<-anti_join(tempsenscomp5a,tempsenscomp4a)

vecoflocyear<-c("USA_Daviess_County_IN", "Czech_Republic_Prague", "Denmark_Sollested","USA_Prosper_TX", "Argentina_Barrow", "Canada_Rosthern", "Chile_Temuco",  "USA_Corvallis_OR_2015", "USA_Corvallis_OR_1990", "USA_Davis_CA", "UK_Jealotts_Hill",  "Israel_Nahal_Oz_April", "Tunisia_Beja", "Germany_Giessen", "Australia_Badgingarra", "Uruguay_Colonia", "Switzerland_Berg_am_Irchel",  "USA_Columbia_MO", "Switzerland_Eschikon_May_2016", "Switzerland_Eschikon_July_2016", "Iran_Dezful", "Australia_Wagga_Wagga")

tempsenscomp3$New_Location_year = factor(tempsenscomp3$New_Location_year, levels = c(vecoflocyear), ordered = TRUE)

vecoflocyearlegend<-c("Greatly better at cold", "Better at cold", "Slightly better at cold", "Similar at cold and warm", "Slightly better at warm", "Better at warm", "Greatly better at warm")



tempsenscomp3%>% ggplot()+
  geom_bar(aes(x = New_Location_year, y = percentageofscore, 
               fill = class_diff), stat = "identity")+scale_fill_manual(values=c("#00349d", "#0054ff", "#89b0ff", "#8a8a8a", "#ff9d9d", "#ff0000", "#890000"), labels=c(vecoflocyearlegend))+labs(x="", y="Percentage of strains (%)", fill="Comparative temperature performance") +coord_flip()+ theme_minimal() +
  theme(strip.placement = "outside")


tempsenscomp3%>% ggplot()+
  geom_bar(aes(x = New_Location_year, y = percentageofscore, 
               fill = class_diff), stat = "identity")+scale_fill_manual(values=c("#00349d", "#0054ff", "#89b0ff", "#8a8a8a", "#ff9d9d", "#ff0000", "#890000"), labels=c(vecoflocyearlegend))+labs(x="", y="Percentage of strains (%)", fill="Genetic cluster") +coord_flip()+ theme_minimal() +
  theme(strip.placement = "outside")




```

```{r}

tempsenscomp2a<-tempsenscomp2 %>% group_by(climate) %>% count() %>% rename(num_isolates=n) %>% ungroup()
tempsenscomp3<-tempsenscomp2 %>% ungroup() %>% group_by(climate, class_diff) %>% count() %>% rename(numberperdiff=n) %>% ungroup() %>% left_join(tempsenscomp2a) %>% filter(num_isolates > 5 ) %>% mutate(percentageofscore=(numberperdiff/num_isolates)*100) 

tempsenscomp3$class_diff<-as.character(tempsenscomp3$class_diff)

tempsenscomp4<-tempsenscomp3 %>% filter(class_diff == "3") %>% dplyr::select(percentageofscore, climate) %>% arrange(percentageofscore)
tempsenscomp5<-tempsenscomp3 %>% filter(class_diff == "0") %>% dplyr::select(percentageofscore, climate) %>% arrange(percentageofscore)

tempsenscomp4a<-tempsenscomp4  %>% dplyr::select( climate)
tempsenscomp5a<-tempsenscomp5 %>% dplyr::select(climate)

tempsens5b<-anti_join(tempsenscomp5a,tempsenscomp4a)



vecoflocyearlegend<-c("Greatly better at cold", "Better at cold", "Slightly better at cold", "Similar at cold and warm", "Slightly better at warm", "Better at warm", "Greatly better at warm")



tempsenscomp3%>% ggplot()+
  geom_bar(aes(x = climate, y = percentageofscore, 
               fill = class_diff), stat = "identity")+scale_fill_manual(values=c("#00349d", "#0054ff", "#89b0ff", "#8a8a8a", "#ff9d9d", "#ff0000", "#890000"), labels=c(vecoflocyearlegend))+labs(x="", y="Percentage of strains (%)", fill="Comparative temperature performance") +coord_flip()+ theme_minimal() +
  theme(strip.placement = "outside")


```

## Plots of examples of significant variants allele distributions and the optimal growing temperature values in the map of the mean temperature of the warmest quarter {.tabset}


### Significant variant in chromosome 1 co-located with ammonium transporter predicted gene



```{r ammonium, message=FALSE, warning=FALSE, error=FALSE}


newcoords1<-read.csv("01_Source_Tables/WP1_SMP_Coordandvariety_contlocextralat_20240826_v1.csv", header=TRUE) %>% rename(Strain=Isolate_full_name_phenotyping) %>% dplyr::select(ID_genome_file, Latitude, Longitude)

datwidegea<-datwidegea%>% dplyr::select(-Latitude, -Longitude)%>% left_join(newcoords1)

i=10

temp2 = raster(paste0("wc2.1__bio_tifffiles/wc2.1_10m_bio_", i, ".tif"))
temp = as.data.frame(rasterToPoints(temp2, xy = TRUE))
colnames(temp) <- c("Longitude", "Latitude", "Bioclim_var")

map1 = ggplot() +
  geom_raster(data = temp , 
              aes(x = Longitude, y = Latitude, 
                  fill = Bioclim_var)) + 
  theme_void() +
  scale_fill_viridis("inferno") +
  theme(panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))


p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))
p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))
p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), get_legend(p1), ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 

d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, Location_year, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))


missing20rem<-c("Texas_1994_ST94TX_PF4a_1", "Tunisia_2008_ST08TN_33_1_2", "ST90ORE_a12_3B_13", "ST92YE_1")  

isolateswrongstructure<-c("Ukraine_1995_ST95UR_F1c_1", "Israel_1992_ISZE2a_2", "Texas_1994_ST94TXPG7a_1", "Kenya_1994_STKE94ML1a", "Kenya_1994_STKE94MF1a")



#makes more sense to make it for eAUC12C
allele<-read_csv("snpalleleinforefalt_chr1_ammonium_12C_v1.csv") %>% rename(ID_genome_file=ID_file) %>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year)) %>% group_by_all() %>% summarize() %>% ungroup() %>% drop_na(Allele)%>% left_join(d) %>% group_by(New_Location_year) %>% mutate( number_of_isolates = n()) %>% filter(!(ID_genome_file %in% isolateswrongstructure)) %>% filter(!(ID_genome_file %in% missing20rem)) %>%
    mutate(Allele = ifelse(Allele == 0, "REF", ifelse(Allele == 1, "ALT1", "ALT2"))) %>%
    group_by(New_Location_year, Allele, Longitude, Latitude, number_of_isolates) %>%
    dplyr::count() %>% 
    pivot_wider(names_from = Allele, values_from = n, values_fill = 0) %>%
    filter(!is.na(Longitude)) %>%
    mutate(Radius =  ifelse(number_of_isolates > 10, 1.5, log(number_of_isolates))) %>% filter(number_of_isolates > 5)
map1 = map1#ggplot() + 
  #theme_void() +
  #geom_polygon(data = map_data("world"), aes(x=long, y = lat, group = group), fill="#ede7e3")  +
  #scale_size("Number of genomes", limits = c(1, max_circle))  +
  #theme(legend.position = "None", 
    #    panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))

p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*0.9,group=New_Location_year,), 
                  cols = c("REF", "ALT1"), color="grey40") +
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent")) +
  scale_fill_manual(values = c("#E58C8A","#7EE8FA"))

p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40")+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#E58C8A","#7EE8FA"))

p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40")+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#E58C8A","#7EE8FA"))


#Boxplot

  
temp = read_csv("snpalleleinforefalt_chr1_ammonium_12C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)%>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year, `Mean Temperature of Warmest Quarter`)) %>%
    filter(Allele != "NA") %>% drop_na(Strain)

p4 = temp %>%
    ggplot(aes(x = as.factor(Allele), y = OptTemperature, fill = as.factor(Allele), col = as.factor(Allele))) +
    geom_boxplot(alpha = .7, outlier.shape = NA) + 
    #labs(title = SNP_name, ylab = Bioclim_var[i]) +
    theme_bw() +
    scale_color_manual(values = c( "#E58C8A", "#7EE8FA")) +
    scale_fill_manual(values = c("#E58C8A","#7EE8FA" ))+labs(y="Optimal temperature", x="Allele") + scale_x_discrete(labels=c("Ref", "Alt"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

#Plotting all the maps together! 
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), p4 + theme(legend.position = "None"), 
                             ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 



```


### Significant variant in chromosome 4 co-located with COMPASS set1c complex predicted gene



```{r compass, message=FALSE, warning=FALSE, error=FALSE}


i=10

temp2 = raster(paste0("wc2.1__bio_tifffiles/wc2.1_10m_bio_", i, ".tif"))
temp = as.data.frame(rasterToPoints(temp2, xy = TRUE))
colnames(temp) <- c("Longitude", "Latitude", "Bioclim_var")

map1 = ggplot() +
  geom_raster(data = temp , 
              aes(x = Longitude, y = Latitude, 
                  fill = Bioclim_var)) + 
  theme_void() +
  scale_fill_viridis("inferno") +
  theme(panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))


p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))
p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))
p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), get_legend(p1), ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 

d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, New_Location_year, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))


missing20rem<-c("Texas_1994_ST94TX_PF4a_1", "Tunisia_2008_ST08TN_33_1_2", "ST90ORE_a12_3B_13", "ST92YE_1")  

isolateswrongstructure<-c("Ukraine_1995_ST95UR_F1c_1", "Israel_1992_ISZE2a_2", "Texas_1994_ST94TXPG7a_1", "Kenya_1994_STKE94ML1a", "Kenya_1994_STKE94MF1a")



#makes more sense to make it for 22C
allele<-read_csv("snpalleleinforefalt_chr4_compass_22C_v1.csv") %>% rename(ID_genome_file=ID_file) %>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year)) %>% group_by_all() %>% summarize() %>% ungroup() %>% drop_na(Allele)%>% left_join(d) %>% group_by(New_Location_year) %>% mutate( number_of_isolates = n()) %>% filter(!(ID_genome_file %in% isolateswrongstructure)) %>% filter(!(ID_genome_file %in% missing20rem)) %>%
    mutate(Allele = ifelse(Allele == 0, "REF", ifelse(Allele == 1, "ALT1", "ALT2"))) %>%
    group_by(New_Location_year, Allele, Longitude, Latitude, number_of_isolates) %>%
    dplyr::count() %>%
    pivot_wider(names_from = Allele, values_from = n, values_fill = 0) %>%
    filter(!is.na(Longitude)) %>%
    mutate(Radius =  ifelse(number_of_isolates > 10, 1.5, log(number_of_isolates))) %>% filter(number_of_isolates > 5)


map1 = map1#ggplot() + 
  #theme_void() +
  #geom_polygon(data = map_data("world"), aes(x=long, y = lat, group = group), fill="#ede7e3")  +
  #scale_size("Number of genomes", limits = c(1, max_circle))  +
  #theme(legend.position = "None", 
    #    panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))

p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*0.9,group=New_Location_year,), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8) +
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent")) +
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))

p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8)+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))

p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8)+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))


#Boxplot

  
temp = read_csv("snpalleleinforefalt_chr4_compass_22C_v1.csv") %>% rename(ID_genome_file=ID_file) %>% full_join(d)%>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year, `Mean Temperature of Warmest Quarter`)) %>%
    filter(Allele != "NA") %>% drop_na(Strain)

p4 = temp %>%
    ggplot(aes(x = as.factor(Allele), y = OptTemperature, fill = as.factor(Allele), col = as.factor(Allele))) +
    geom_boxplot(alpha = .7, outlier.shape = NA) + 
    #labs(title = SNP_name, ylab = Bioclim_var[i]) +
    theme_bw() +
    scale_color_manual(values = c( "#7EE8FA" ,"#E58C8A")) +
    scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))+labs(y="Optimal temperature", x="Allele") + scale_x_discrete(labels=c("Ref", "Alt"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

#Plotting all the maps together! 
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), p4 + theme(legend.position = "None"), 
                             ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 



#colors.gengroups<-c("grey30", "#FF6319", #Argentina & Uruguay
#             "#FCCC0A", #Australia
#             "#020057", #Europe
#             "#7cdbff", #Africa
#             "#6E3219",  #USA
#             "#1a5fff", #Middle East
#             "#C60C30", #Chile
#             "#8E258D") #Canada

#names(colors.gengroups) = c("Other", "Argentina_Uruguay", "Australia", "Europe", "Tunisia", "USA", "Middle_East", "Chile", "Canada")


#widelofarea_approx_filtered %>% ungroup() %>% group_by(Adm_Cluster, New_Location_year) %>% count() %>% ggplot(aes(x=New_Location_year, y=n, fill=Adm_Cluster, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = colors.gengroups)+labs(x="", y="Number of strains")+ geom_text(vjust=1.6, color="white", size=3)+
#  theme(axis.text.x = element_text(angle = 60, hjust = 1))





```


### Significant variant in chromosome 7 co-located with ZtIPO323_084320


```{r gea084320, message=FALSE, warning=FALSE, error=FALSE}

supplementarytab3<-read_csv("Supplementary_Table_S3_a.csv")  %>% filter(Gene== "ZtIPO323_084320") %>% filter(Feurtey_genes != "Zt09_chr_7.469")

supplementarytab3a<-supplementarytab3 %>% pivot_longer(
  cols = Algeria_1992_ALA2a:c3_14_14A2,
  names_to = c("ID_genome_file"),
  values_to = "Allele") %>% dplyr::select(ID_genome_file, Allele)


i=10

temp2 = raster(paste0("wc2.1__bio_tifffiles/wc2.1_10m_bio_", i, ".tif"))
temp = as.data.frame(rasterToPoints(temp2, xy = TRUE))
colnames(temp) <- c("Longitude", "Latitude", "Bioclim_var")

map1 = ggplot() +
  geom_raster(data = temp , 
              aes(x = Longitude, y = Latitude, 
                  fill = Bioclim_var)) + 
  theme_void() +
  scale_fill_viridis("magma") +
  theme(panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))


p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))
p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))
p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), get_legend(p1), ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 

d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, New_Location_year, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))


missing20rem<-c("Texas_1994_ST94TX_PF4a_1", "Tunisia_2008_ST08TN_33_1_2", "ST90ORE_a12_3B_13", "ST92YE_1")  

isolateswrongstructure<-c("Ukraine_1995_ST95UR_F1c_1", "Israel_1992_ISZE2a_2", "Texas_1994_ST94TXPG7a_1", "Kenya_1994_STKE94ML1a", "Kenya_1994_STKE94MF1a")



#makes more sense to make it for 22C
allele<-supplementarytab3a %>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year)) %>% group_by_all() %>% summarize() %>% ungroup() %>% drop_na(Allele)%>% left_join(d) %>% group_by(New_Location_year) %>% mutate( number_of_isolates = n()) %>% filter(!(ID_genome_file %in% isolateswrongstructure)) %>% filter(!(ID_genome_file %in% missing20rem)) %>%
    mutate(Allele = ifelse(Allele == 0, "REF", ifelse(Allele == 1, "ALT1", "ALT2"))) %>%
    group_by(New_Location_year, Allele, Longitude, Latitude, number_of_isolates) %>%
    dplyr::count() %>%
    pivot_wider(names_from = Allele, values_from = n, values_fill = 0) %>%
    filter(!is.na(Longitude)) %>%
    mutate(Radius =  ifelse(number_of_isolates > 10, 1.5, log(number_of_isolates))) 


map1 = map1#ggplot() + 
  #theme_void() +
  #geom_polygon(data = map_data("world"), aes(x=long, y = lat, group = group), fill="#ede7e3")  +
  #scale_size("Number of genomes", limits = c(1, max_circle))  +
  #theme(legend.position = "None", 
    #    panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))

p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*0.9,group=New_Location_year,), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8) +
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent")) +
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))

p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8)+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))

p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8)+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))


#Boxplot

  
temp = supplementarytab3a %>% full_join(d)%>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year, `Mean Temperature of Warmest Quarter`)) %>%
    filter(Allele != "NA") %>% drop_na(Strain)

p4 = temp %>%
    ggplot(aes(x = as.factor(Allele), y = OptTemperature, fill = as.factor(Allele), col = as.factor(Allele))) +
    geom_boxplot(alpha = .7, outlier.shape = NA) + 
    #labs(title = SNP_name, ylab = Bioclim_var[i]) +
    theme_bw() +
    scale_color_manual(values = c( "#7EE8FA" ,"#E58C8A")) +
    scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))+labs(y="Optimal temperature", x="Allele") + scale_x_discrete(labels=c("Ref", "Alt"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

#Plotting all the maps together! 
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), p4 + theme(legend.position = "None"), 
                             ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 



#colors.gengroups<-c("grey30", "#FF6319", #Argentina & Uruguay
#             "#FCCC0A", #Australia
#             "#020057", #Europe
#             "#7cdbff", #Africa
#             "#6E3219",  #USA
#             "#1a5fff", #Middle East
#             "#C60C30", #Chile
#             "#8E258D") #Canada

#names(colors.gengroups) = c("Other", "Argentina_Uruguay", "Australia", "Europe", "Tunisia", "USA", "Middle_East", "Chile", "Canada")


#widelofarea_approx_filtered %>% ungroup() %>% group_by(Adm_Cluster, New_Location_year) %>% count() %>% ggplot(aes(x=New_Location_year, y=n, fill=Adm_Cluster, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = colors.gengroups)+labs(x="", y="Number of strains")+ geom_text(vjust=1.6, color="white", size=3)+
#  theme(axis.text.x = element_text(angle = 60, hjust = 1))



```

```{r gea084320iso, message=FALSE, warning=FALSE, error=FALSE}

supplementarytab3<-read_csv("Supplementary_Table_S3_a.csv")  %>% filter(Gene== "ZtIPO323_084320") %>% filter(Feurtey_genes != "Zt09_chr_7.469")

supplementarytab3a<-supplementarytab3 %>% pivot_longer(
  cols = Algeria_1992_ALA2a:c3_14_14A2,
  names_to = c("ID_genome_file"),
  values_to = "Allele") %>% dplyr::select(ID_genome_file, Allele)


i=9

temp2 = raster(paste0("wc2.1__bio_tifffiles/wc2.1_10m_bio_", i, ".tif"))
temp = as.data.frame(rasterToPoints(temp2, xy = TRUE))
colnames(temp) <- c("Longitude", "Latitude", "Bioclim_var")

map1 = ggplot() +
  geom_raster(data = temp , 
              aes(x = Longitude, y = Latitude, 
                  fill = Bioclim_var)) + 
  theme_void() +
  scale_fill_viridis(option="inferno") +
  theme(panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))


p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))
p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))
p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), get_legend(p1), ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 

d<-datwide3  %>% ungroup() %>% drop_na(OptTemperature) %>% group_by(Strain, climate, New_Location_year, Country, Adm_Cluster, ID_genome_file) %>% summarize(OptTemperature=mean(OptTemperature, na.rm=T))


missing20rem<-c("Texas_1994_ST94TX_PF4a_1", "Tunisia_2008_ST08TN_33_1_2", "ST90ORE_a12_3B_13", "ST92YE_1")  

isolateswrongstructure<-c("Ukraine_1995_ST95UR_F1c_1", "Israel_1992_ISZE2a_2", "Texas_1994_ST94TXPG7a_1", "Kenya_1994_STKE94ML1a", "Kenya_1994_STKE94MF1a")



#makes more sense to make it for 22C
allele<-supplementarytab3a %>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year)) %>% group_by_all() %>% summarize() %>% ungroup() %>% drop_na(Allele)%>% left_join(d) %>% group_by(New_Location_year) %>% mutate( number_of_isolates = n()) %>% filter(!(ID_genome_file %in% isolateswrongstructure)) %>% filter(!(ID_genome_file %in% missing20rem)) %>%
    mutate(Allele = ifelse(Allele == 0, "REF", ifelse(Allele == 1, "ALT1", "ALT2"))) %>%
    group_by(New_Location_year, Allele, Longitude, Latitude, number_of_isolates) %>%
    dplyr::count() %>%
    pivot_wider(names_from = Allele, values_from = n, values_fill = 0) %>%
    filter(!is.na(Longitude)) %>%
    mutate(Radius =  ifelse(number_of_isolates > 10, 1.5, log(number_of_isolates))) %>% filter(number_of_isolates > 5)


map1 = map1#ggplot() + 
  #theme_void() +
  #geom_polygon(data = map_data("world"), aes(x=long, y = lat, group = group), fill="#ede7e3")  +
  #scale_size("Number of genomes", limits = c(1, max_circle))  +
  #theme(legend.position = "None", 
    #    panel.border  = element_rect(fill=NA, colour = "grey",size = 0.5, linetype = 1))

p1 = map1 + coord_cartesian(xlim=c(-20, 60), ylim=c(20, 70))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*0.9,group=New_Location_year,), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8) +
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent")) +
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))

p2 = map1 + coord_cartesian(xlim=c(-160, -40), ylim=c(-80, 80))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8)+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))

p3 = map1 + coord_cartesian(xlim=c(105, 175), ylim=c(-65, 10))+ new_scale_fill() + 
  geom_scatterpie(data = allele, mapping = aes(x=Longitude, y=Latitude, r = Radius*2.5), 
                  cols = c("REF", "ALT1"), color="grey40", alpha = .8)+
  theme(panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"))+
  scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))


#Boxplot

  
temp = supplementarytab3a %>% full_join(d)%>%left_join( datwidegea %>% dplyr::select(ID_genome_file, Latitude, Longitude, New_Location_year, `Mean Temperature of Driest Quarter`)) %>%
    filter(Allele != "NA") %>% drop_na(Strain)

p4 = temp %>%
    ggplot(aes(x = as.factor(Allele), y = OptTemperature, fill = as.factor(Allele), col = as.factor(Allele))) +
    geom_boxplot(alpha = .7, outlier.shape = NA) + 
    #labs(title = SNP_name, ylab = Bioclim_var[i]) +
    theme_bw() +
    scale_color_manual(values = c( "#7EE8FA" ,"#E58C8A")) +
    scale_fill_manual(values = c("#7EE8FA" ,"#E58C8A"))+labs(y="Optimal temperature", x="Allele") + scale_x_discrete(labels=c("Ref", "Alt"))+ theme(axis.text.x = element_text(angle = 45, hjust = 1))

#Plotting all the maps together! 
aus_map = cowplot::plot_grid(p3 + theme(legend.position = "None"), p4 + theme(legend.position = "None"), 
                             ncol = 2, rel_widths = c(1.1, 1))
ligne = cowplot::plot_grid(p1 + theme(legend.position = "None"), aus_map, ncol = 1,  rel_heights = c(1, 0.7))
cowplot::plot_grid(p2 + theme(legend.position = "None"), ligne, ncol = 2, rel_widths = c(0.9, 1)) 



#colors.gengroups<-c("grey30", "#FF6319", #Argentina & Uruguay
#             "#FCCC0A", #Australia
#             "#020057", #Europe
#             "#7cdbff", #Africa
#             "#6E3219",  #USA
#             "#1a5fff", #Middle East
#             "#C60C30", #Chile
#             "#8E258D") #Canada

#names(colors.gengroups) = c("Other", "Argentina_Uruguay", "Australia", "Europe", "Tunisia", "USA", "Middle_East", "Chile", "Canada")


#widelofarea_approx_filtered %>% ungroup() %>% group_by(Adm_Cluster, New_Location_year) %>% count() %>% ggplot(aes(x=New_Location_year, y=n, fill=Adm_Cluster, label=n))+geom_bar(stat='identity')+theme_minimal()+ theme(legend.position="none", axis.line = element_line(colour = "grey50"))+ scale_fill_manual(values = colors.gengroups)+labs(x="", y="Number of strains")+ geom_text(vjust=1.6, color="white", size=3)+
#  theme(axis.text.x = element_text(angle = 60, hjust = 1))



```


# References
